--- gnome-session/main.c	2007-01-22 20:15:57.000000000 -0600
+++ gnome-session/main.c	2007-01-22 20:17:57.000000000 -0600
@@ -572,7 +572,6 @@
   GError *err;
   int status;
   char *display_str;
-  char **versions;
   GConfClient *gconf_client;
   GOptionContext *goption_context;
   gboolean dbus_daemon_owner;
@@ -580,20 +579,6 @@
   if (getenv ("GSM_VERBOSE_DEBUG"))
     gsm_set_verbose (TRUE);
 
-  /* Help eradicate the critical warnings in unstable releases of GNOME */
-  versions = g_strsplit (VERSION, ".", 3);
-  if (versions && versions [0] && versions [1])
-    {
-      int major;
-      major = atoi (versions [1]);
-      if ((major % 2) != 0)
-	{
-          g_setenv ("G_DEBUG", "fatal_criticals", FALSE);
-	  g_log_set_always_fatal (G_LOG_LEVEL_CRITICAL);
-	}
-    }
-  g_strfreev (versions);
-      
   set_lang();
   set_gtk1_theme_rcfile ();
 
