From c13d63ffa55cf03bc8fa37f19961714d7386a69a Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Sun, 14 Aug 2011 14:37:41 -0400
Subject: [PATCH] Drop dsextras.py (it's installed by pygobject-2.28.x)

---
 Makefile.am |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3b8b6fe..e696cea 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -56,11 +56,6 @@ INCLUDES = -I$(top_srcdir)/gobject $(PYTHON_INCLUDES) $(GLIB_CFLAGS)
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pygobject-$(PLATFORM_VERSION).pc
 
-# python modules
-pkgpyexecdir = $(pyexecdir)/gtk-2.0
-pkgpyexec_PYTHON = dsextras.py
-pkgpyexec_LTLIBRARIES =
-
 # python
 pyexec_LTLIBRARIES =
 
-- 
1.7.6

