From 9243b978cf984b11be3695e4f9cc6dbbe57efb68 Mon Sep 17 00:00:00 2001
From: Shaun McCance <shaunm@gnome.org>
Date: Fri, 22 Jul 2011 14:50:32 +0000
Subject: Don't try to search with NULL URI, bug #655124

---
diff --git a/libyelp/yelp-location-entry.c b/libyelp/yelp-location-entry.c
index 493dab2..a7ebfa9 100644
--- a/libyelp/yelp-location-entry.c
+++ b/libyelp/yelp-location-entry.c
@@ -622,6 +622,8 @@ location_entry_search_activated  (YelpLocationEntry *entry)
     YelpLocationEntryPrivate *priv = GET_PRIV (entry);
 
     g_object_get (priv->view, "yelp-uri", &base, NULL);
+    if (base == NULL)
+        return;
     uri = yelp_uri_new_search (base,
                                gtk_entry_get_text (GTK_ENTRY (priv->text_entry)));
     g_object_unref (base);
--
cgit v0.9
