diff -Naur system-tools-backends-2.4.1/Init/Services.pm system-tools-backends-2.4.1.new/Init/Services.pm
--- Init/Services.pm	2007-09-26 05:13:27.000000000 +0530
+++ Init/Services.pm	2007-11-06 05:19:51.000000000 +0530
@@ -647,7 +647,7 @@
 
 sub get_gentoo_runlevels
 {
-  my($raw_output) = Utils::File::run_backtick("rc-status -nc -l");
+  my($raw_output) = Utils::File::run_backtick("rc-status --nocolor -l");
   my(@runlevels);
 
   return undef if (!$raw_output);
@@ -659,7 +659,7 @@
 sub get_gentoo_services_for_runlevel
 {
   my($runlevel) = @_;
-  my($raw_output) = Utils::File::run_backtick("rc-status -nc $runlevel");
+  my($raw_output) = Utils::File::run_backtick("rc-status --nocolor $runlevel");
   my(@raw_lines) = split(/\n/,$raw_output);
   my($line, $service);
   my(%services);
