From 65736d8de622a793fc1201d71d2bfc25f6b2400b Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Mon, 29 Oct 2012 00:42:38 +0100
Subject: [PATCH 1/4] Out-of-tree build fixes

---
 src/Makefile.am | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 9f7c7f8..387defa 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -165,7 +165,7 @@ endif
 if HAVE_HB_OLD
 SUBDIRS += hb-old
 HBCFLAGS += -I$(srcdir)/hb-old
-HBLIBS   += hb-old/libhb-old.la
+HBLIBS   += $(builddir)/hb-old/libhb-old.la
 HBSOURCES += hb-old.cc
 endif
 DIST_SUBDIRS += hb-old
@@ -173,7 +173,7 @@ DIST_SUBDIRS += hb-old
 if HAVE_ICU_LE
 SUBDIRS += hb-icu-le
 HBCFLAGS += -I$(srcdir)/hb-icu-le
-HBLIBS   += hb-icu-le/libhb-icu-le.la
+HBLIBS   += $(builddir)/hb-icu-le/libhb-icu-le.la
 HBSOURCES += hb-icu-le.cc
 endif
 DIST_SUBDIRS += hb-icu-le
@@ -181,7 +181,7 @@ DIST_SUBDIRS += hb-icu-le
 if HAVE_UCDN
 SUBDIRS += hb-ucdn
 HBCFLAGS += -I$(srcdir)/hb-ucdn
-HBLIBS   += hb-ucdn/libhb-ucdn.la
+HBLIBS   += $(builddir)/hb-ucdn/libhb-ucdn.la
 HBSOURCES += hb-ucdn.cc
 endif
 DIST_SUBDIRS += hb-ucdn
@@ -251,11 +251,11 @@ bin_PROGRAMS =
 
 main_SOURCES = main.cc
 main_CPPFLAGS = $(HBCFLAGS)
-main_LDADD = libharfbuzz.la $(HBLIBS)
+main_LDADD = $(builddir)/libharfbuzz.la $(HBLIBS)
 
 test_would_substitute_SOURCES = test-would-substitute.cc
 test_would_substitute_CPPFLAGS = $(HBCFLAGS) $(FREETYPE_CFLAGS)
-test_would_substitute_LDADD = libharfbuzz.la $(HBLIBS) $(FREETYPE_LIBS)
+test_would_substitute_LDADD = $(builddir)/libharfbuzz.la $(HBLIBS) $(FREETYPE_LIBS)
 
 dist_check_SCRIPTS = \
 	check-c-linkage-decls.sh \
-- 
1.7.12.4

