From 26fe8720c9699d37757cc92064a4975b9ac021cf Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Mon, 29 Oct 2012 22:04:46 +0100
Subject: [PATCH 3/4] Make icu support optional

---
 configure.ac | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index ae462ad..bcd3991 100644
--- a/configure.ac
+++ b/configure.ac
@@ -169,19 +169,23 @@ AM_CONDITIONAL(HAVE_CAIRO_FT, $have_cairo_ft)
 
 dnl ==========================================================================
 
+AC_ARG_ENABLE([icu],
+    AS_HELP_STRING([--disable-icu], [Disable icu support]),
+    ,[enable_icu=yes])
+
 PKG_CHECK_MODULES(ICU, icu-uc, have_icu=true, have_icu=false)
-if $have_icu; then
+if test $enable_icu = "yes" && $have_icu; then
 	AC_DEFINE(HAVE_ICU, 1, [Have ICU library])
 fi
-AM_CONDITIONAL(HAVE_ICU, $have_icu)
+AM_CONDITIONAL(HAVE_ICU, test $enable_icu = "yes" && $have_icu)
 
 dnl ==========================================================================
 
 PKG_CHECK_MODULES(ICU_LE, icu-le icu-uc, have_icu_le=true, have_icu_le=false)
-if $have_icu_le; then
+if test $enable_icu = "yes" && $have_icu_le; then
 	AC_DEFINE(HAVE_ICU_LE, 1, [Have ICU Layout Engine library])
 fi
-AM_CONDITIONAL(HAVE_ICU_LE, $have_icu_le)
+AM_CONDITIONAL(HAVE_ICU_LE, test $enable_icu = "yes" && $have_icu_le)
 
 dnl ==========================================================================
 
-- 
1.7.12.4

