diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN libgnomeui-2.19.1.orig/libgnomeui/gnome-druid.h libgnomeui-2.19.1/libgnomeui/gnome-druid.h
--- libgnomeui-2.19.1.orig/libgnomeui/gnome-druid.h	2007-06-26 18:59:09.000000000 -0400
+++ libgnomeui-2.19.1/libgnomeui/gnome-druid.h	2007-08-27 16:29:46.000000000 -0400
@@ -24,8 +24,6 @@
 #ifndef __GNOME_DRUID_H__
 #define __GNOME_DRUID_H__
 
-#ifndef GNOME_DISABLE_DEPRECATED
-
 #include <gtk/gtk.h>
 #include "gnome-druid-page.h"
 
@@ -96,6 +94,4 @@ void       gnome_druid_construct_with_wi
 
 G_END_DECLS
 
-#endif /* GNOME_DISABLE_DEPRECATED */
-
 #endif /* __GNOME_DRUID_H__ */
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page-edge.h libgnomeui-2.19.1/libgnomeui/gnome-druid-page-edge.h
--- libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page-edge.h	2007-06-26 18:59:09.000000000 -0400
+++ libgnomeui-2.19.1/libgnomeui/gnome-druid-page-edge.h	2007-08-27 16:29:46.000000000 -0400
@@ -24,8 +24,6 @@
 #ifndef __GNOME_DRUID_PAGE_EDGE_H__
 #define __GNOME_DRUID_PAGE_EDGE_H__
 
-#ifndef GNOME_DISABLE_DEPRECATED
-
 #include <gtk/gtk.h>
 #include <libgnomecanvas/gnome-canvas.h>
 #include "gnome-druid-page.h"
@@ -158,6 +156,4 @@ void       gnome_druid_page_edge_set_top
 
 G_END_DECLS
 
-#endif /* GNOME_DISABLE_DEPRECATED */
-
 #endif /* __GNOME_DRUID_PAGE_EDGE_H__ */
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page.h libgnomeui-2.19.1/libgnomeui/gnome-druid-page.h
--- libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page.h	2007-06-26 18:59:09.000000000 -0400
+++ libgnomeui-2.19.1/libgnomeui/gnome-druid-page.h	2007-08-27 16:29:46.000000000 -0400
@@ -23,8 +23,6 @@
 #ifndef __GNOME_DRUID_PAGE_H__
 #define __GNOME_DRUID_PAGE_H__
 
-#ifndef GNOME_DISABLE_DEPRECATED
-
 #include <gtk/gtk.h>
 
 #include <libgnomecanvas/gnome-canvas.h>
@@ -84,8 +82,6 @@ void       gnome_druid_page_finish      
 
 G_END_DECLS
 
-#endif /* GNOME_DISABLE_DEPRECATED */
-
 #endif /* __GNOME_DRUID_PAGE_H__ */
 
 
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page-standard.h libgnomeui-2.19.1/libgnomeui/gnome-druid-page-standard.h
--- libgnomeui-2.19.1.orig/libgnomeui/gnome-druid-page-standard.h	2007-06-26 18:59:09.000000000 -0400
+++ libgnomeui-2.19.1/libgnomeui/gnome-druid-page-standard.h	2007-08-27 16:29:46.000000000 -0400
@@ -24,8 +24,6 @@
 #ifndef __GNOME_DRUID_PAGE_STANDARD_H__
 #define __GNOME_DRUID_PAGE_STANDARD_H__
 
-#ifndef GNOME_DISABLE_DEPRECATED
-
 #include <gtk/gtk.h>
 #include "gnome-druid-page.h"
 
@@ -83,10 +81,14 @@ struct _GnomeDruidPageStandardClass
 	gpointer padding2;
 };
 
+#ifndef GNOME_DISABLE_DEPRECATED
+
 #define gnome_druid_page_standard_set_bg_color      gnome_druid_page_standard_set_background
 #define gnome_druid_page_standard_set_logo_bg_color gnome_druid_page_standard_set_logo_background
 #define gnome_druid_page_standard_set_title_color   gnome_druid_page_standard_set_title_foreground
 
+#endif
+
 
 GType      gnome_druid_page_standard_get_type                (void) G_GNUC_CONST;
 GtkWidget *gnome_druid_page_standard_new                     (void);
@@ -117,7 +119,5 @@ void       gnome_druid_page_standard_app
 
 G_END_DECLS
 
-#endif /* GNOME_DISABLE_DEPRECATED */
-
 #endif /* __GNOME_DRUID_PAGE_STANDARD_H__ */
 
