From e2b44ece90711752dfe970f64b290d7921119399 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Sun, 11 Sep 2011 13:40:00 +0000
Subject: Fix clock to actually update every minute

https://bugzilla.gnome.org/show_bug.cgi?id=658746
---
diff --git a/src/gs-window-x11.c b/src/gs-window-x11.c
index a0de35a..6956874 100644
--- a/src/gs-window-x11.c
+++ b/src/gs-window-x11.c
@@ -2190,6 +2190,14 @@ update_clock (GSWindow *window)
         g_free (markup);
 }
 
+static void
+on_clock_changed (GnomeWallClock *clock,
+                  GParamSpec     *pspec,
+                  gpointer        user_data)
+{
+        update_clock (GS_WINDOW (user_data));
+}
+
 static char *
 get_user_display_name (void)
 {
@@ -2339,6 +2347,7 @@ gs_window_init (GSWindow *window)
         create_info_bar (window);
 
         window->priv->clock_tracker = g_object_new (GNOME_TYPE_WALL_CLOCK, NULL);
+        g_signal_connect (window->priv->clock_tracker, "notify::clock", G_CALLBACK (on_clock_changed), window);
         update_clock (window);
 
         force_no_pixmap_background (window->priv->drawing_area);
--
cgit v0.9.0.2
