diff -urN libsoup-2.44.0.old/configure.ac libsoup-2.44.0/configure.ac
--- libsoup-2.44.0.old/configure.ac	2013-09-24 02:21:30.000000000 +0400
+++ libsoup-2.44.0/configure.ac	2013-10-07 14:03:54.524427669 +0400
@@ -78,19 +78,25 @@
 dnl *** Checks for glib ***
 dnl ***********************
 
-GLIB_REQUIRED=2.36.0
-AM_PATH_GLIB_2_0($GLIB_REQUIRED,,,gobject gio)
-if test "$GLIB_LIBS" = ""; then
-   AC_MSG_ERROR(GLIB $GLIB_REQUIRED or later is required to build libsoup)
+PKG_PROG_PKG_CONFIG
+enable_ssl=no
+AC_ARG_WITH(libsoup-system,
+   AS_HELP_STRING([--with-libsoup-system], [Use libsoup system library to build(default=no)]),
+   [with_libsoup_system=$withval], [with_libsoup_system=no])
+if test "$with_libsoup_system" = "no"; then
+    GLIB_REQUIRED=2.36.0
+    AM_PATH_GLIB_2_0($GLIB_REQUIRED,,,gobject gio)
+    if test "$GLIB_LIBS" = ""; then
+        AC_MSG_ERROR(GLIB $GLIB_REQUIRED or later is required to build libsoup)
+    fi
+    GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_36"
+    GLIB_MAKEFILE='$(top_srcdir)/Makefile.glib'
+    AC_SUBST(GLIB_MAKEFILE)
+    PKG_CHECK_MODULES(XML, libxml-2.0)
+    AC_SUBST(XML_CFLAGS)
+    AC_SUBST(XML_LIBS)
 fi
-GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_36"
-
-GLIB_MAKEFILE='$(top_srcdir)/Makefile.glib'
-AC_SUBST(GLIB_MAKEFILE)
-
-PKG_CHECK_MODULES(XML, libxml-2.0)
-AC_SUBST(XML_CFLAGS)
-AC_SUBST(XML_LIBS)
+AM_CONDITIONAL(BUILD_LIBSOUP, test $with_libsoup_system = no)
 
 PKG_CHECK_MODULES(SQLITE, sqlite3)
 AC_SUBST(SQLITE_CFLAGS)
@@ -139,6 +145,10 @@
 
 if test $with_gnome != no; then
 	AC_DEFINE(HAVE_GNOME, 1, [Defined if GNOME support is enabled])
+
+   if test $with_libsoup_system != no; then
+       PKG_CHECK_MODULES(LIBSOUP, libsoup-$SOUP_API_VERSION = $VERSION)
+   fi
 fi
 AC_SUBST(HAVE_GNOME)
 
diff -urN libsoup-2.44.0.old/libsoup/Makefile.am libsoup-2.44.0/libsoup/Makefile.am
--- libsoup-2.44.0.old/libsoup/Makefile.am	2013-09-23 17:08:20.000000000 +0400
+++ libsoup-2.44.0/libsoup/Makefile.am	2013-10-07 14:30:52.693289768 +0400
@@ -1,14 +1,19 @@
 ## Process this file with automake to produce Makefile.in
 
 EXTRA_DIST =
-
+BUILT_SOURCES =
 include $(GLIB_MAKEFILE)
 
 if OS_WIN32
 LIBWS2_32 = -lws2_32
 endif
 
-AM_CPPFLAGS = 				\
+AM_CPPFLAGS =
+lib_LTLIBRARIES =
+CLEANFILES =
+
+if BUILD_LIBSOUP
+AM_CPPFLAGS += 				\
 	-DG_LOG_DOMAIN=\"libsoup\" 	\
 	-DLOCALEDIR=\"$(localedir)\"	\
 	-I$(top_srcdir)			\
@@ -76,7 +81,7 @@
 nodist_libsoupinclude_HEADERS =	\
 	soup-version.h
 
-lib_LTLIBRARIES = libsoup-2.4.la
+lib_LTLIBRARIES += libsoup-2.4.la
 
 libsoup_2_4_la_LDFLAGS =	\
 	-version-info $(SOUP_CURRENT):$(SOUP_REVISION):$(SOUP_AGE) \
@@ -184,18 +189,23 @@
 	soup-version.c			\
 	soup-xmlrpc.c
 
-# TLD rules
-EXTRA_DIST += tld-parser.py
-
+BUILT_SOURCES += tld_data.inc
 TLD_DATA_FILE=$(top_srcdir)/data/effective_tld_names.dat
 
 tld_data.inc: tld-parser.py $(TLD_DATA_FILE)
-	$(srcdir)/tld-parser.py $(TLD_DATA_FILE) tld_data.inc
+	$(AM_V_GEN) python $(srcdir)/tld-parser.py $(TLD_DATA_FILE) tld_data.inc
+
+# TLD rules
+EXTRA_DIST += tld-parser.py
+
+endif
 
 if BUILD_LIBSOUP_GNOME
 
 libsoupgnomeincludedir = $(includedir)/libsoup-gnome-2.4/libsoup
 
+AM_CPPFLAGS += $(LIBSOUP_CFLAGS)
+
 libsoupgnomeinclude_HEADERS =	\
 	soup-cookie-jar-sqlite.h\
 	soup-gnome.h		\
@@ -203,17 +213,26 @@
 
 lib_LTLIBRARIES += libsoup-gnome-2.4.la
 
+if BUILD_LIBSOUP
+libsoup_gnome_2_4_la_LDFLAGS = $(libsoup_2_4_la_LDFLAGS)
+else
 libsoup_gnome_2_4_la_LDFLAGS =	\
 	-version-info $(SOUP_CURRENT):$(SOUP_REVISION):$(SOUP_AGE) \
 	-no-undefined \
 	-export-symbols $(srcdir)/libsoup-gnome-2.4.sym
+endif
 
 EXTRA_DIST += libsoup-gnome-2.4.sym
 
 libsoup_gnome_2_4_la_LIBADD =		\
-	libsoup-2.4.la			\
 	$(GLIB_LIBS)
 
+if BUILD_LIBSOUP
+libsoup_gnome_2_4_la_LIBADD += libsoup-2.4.la
+else
+libsoup_gnome_2_4_la_LIBADD += $(LIBSOUP_LIBS)
+endif
+
 libsoup_gnome_2_4_la_SOURCES =		\
 	soup-cookie-jar-sqlite.c	\
 	soup-gnome-features.c		\
@@ -224,10 +243,10 @@
 
 endif
 
+#GLIB_GENERATED = soup-marshal.c soup-marshal.h
 GLIB_GENERATED = soup-enum-types.c soup-enum-types.h
-BUILT_SOURCES = \
-	$(GLIB_GENERATED)   \
-	tld_data.inc
+BUILT_SOURCES += \
+	$(GLIB_GENERATED)
 
 soup_enum_types_sources = $(libsoupinclude_HEADERS) $(libsoupgnomeinclude_HEADERS)
 soup_enum_types_MKENUMS_C_FLAGS = --fhead "\#define LIBSOUP_USE_UNSTABLE_REQUEST_API"
@@ -241,6 +260,7 @@
 INTROSPECTION_COMPILER_ARGS = --includedir=.
 
 if HAVE_INTROSPECTION
+if BUILD_LIBSOUP
 
 # Core library
 gi_soup_files = \
@@ -263,20 +283,28 @@
 
 INTROSPECTION_GIRS += Soup-2.4.gir
 
+endif
+
 if BUILD_LIBSOUP_GNOME
 
+if BUILD_LIBSOUP
+SOUP_GIR = Soup-2.4.gir
+else
+SOUP_GIR =
+endif
+
 # GNOME extensions
 gi_soup_gnome_files = $(filter-out soup-gnome.h,\
 		          $(libsoupgnomeinclude_HEADERS) \
 	                  $(filter-out %.h, $(libsoup_gnome_2_4_la_SOURCES)))
-SoupGNOME-2.4.gir: libsoup-gnome-2.4.la Soup-2.4.gir
+SoupGNOME-2.4.gir: libsoup-gnome-2.4.la $(SOUP_GIR)
+SoupGNOME_2_4_gir_INCLUDES = Soup-2.4
 SoupGNOME_2_4_gir_SCANNERFLAGS =			\
 	--identifier-prefix=Soup			\
 	--symbol-prefix=soup				\
-	--c-include "libsoup/soup-gnome.h"		\
-	--include-uninstalled=$(builddir)/Soup-2.4.gir
+	--c-include "libsoup/soup-gnome.h"
 SoupGNOME_2_4_gir_CFLAGS = $(AM_CPPFLAGS)
-SoupGNOME_2_4_gir_LIBS = libsoup-gnome-2.4.la libsoup-2.4.la
+SoupGNOME_2_4_gir_LIBS = libsoup-gnome-2.4.la
 SoupGNOME_2_4_gir_FILES = $(addprefix $(srcdir)/,$(gi_soup_gnome_files))
 SoupGNOME_2_4_gir_EXPORT_PACKAGES = libsoup-gnome-2.4
 
@@ -290,6 +318,6 @@
 typelibdir = $(libdir)/girepository-1.0
 typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
-CLEANFILES = $(gir_DATA) $(typelib_DATA) $(BUILT_SOURCES)
+CLEANFILES += $(gir_DATA) $(typelib_DATA) $(BUILT_SOURCES)
 
 endif
diff -urN libsoup-2.44.0.old/Makefile.am libsoup-2.44.0/Makefile.am
--- libsoup-2.44.0.old/Makefile.am	2013-08-24 20:48:49.000000000 +0400
+++ libsoup-2.44.0/Makefile.am	2013-10-07 14:00:11.714446657 +0400
@@ -1,7 +1,11 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = libsoup po tests examples docs
+SUBDIRS = libsoup
+
+if BUILD_LIBSOUP
+SUBDIRS += po tests docs
+endif
 
 EXTRA_DIST =				\
 	data/effective_tld_names.dat	\
@@ -16,7 +20,11 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 
-pkgconfig_DATA = libsoup-2.4.pc
+pkgconfig_DATA =
+
+if BUILD_LIBSOUP
+pkgconfig_DATA += libsoup-2.4.pc
+endif
 
 if BUILD_LIBSOUP_GNOME
 pkgconfig_DATA += libsoup-gnome-2.4.pc
