diff --git a/src/Makefile.am b/src/Makefile.am
index 37f8444..2f72fa6 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -279,14 +279,14 @@ GVNC_INTROSPECTION_SRCS = \
 			$(srcdir)/vnccolormap.h $(srcdir)/vnccolormap.c \
 			$(srcdir)/vnccursor.h $(srcdir)/vnccursor.c \
 			$(srcdir)/vncconnection.h $(srcdir)/vncconnection.c \
-			$(builddir)/vncconnectionenums.h $(builddir)/vncconnectionenums.c \
+			$(srcdir)/vncconnectionenums.h $(srcdir)/vncconnectionenums.c \
 			$(srcdir)/vncutil.h $(srcdir)/vncutil.c
 
 GTK_VNC_INTROSPECTION_SRCS = \
 			$(srcdir)/vnccairoframebuffer.h $(srcdir)/vnccairoframebuffer.c \
 			$(srcdir)/vncdisplay.h $(srcdir)/vncdisplay.c \
 			$(srcdir)/vncgrabsequence.h $(srcdir)/vncgrabsequence.c \
-			$(builddir)/vncdisplayenums.h $(builddir)/vncdisplayenums.c
+			$(srcdir)/vncdisplayenums.h $(srcdir)/vncdisplayenums.c
 
 if HAVE_GTK_2
 GTK_VNC_INTROSPECTION_SRCS += \
