--- librsvg-2.31.0.orig/configure.in
+++ librsvg-2.31.0/configure.in
@@ -240,9 +240,20 @@ have_gtk_3=no
 GTK2_BINARY_VERSION=
 GTK3_BINARY_VERSION=
 
+AC_ARG_WITH([gtk2],
+  [AS_HELP_STRING([--with-gtk2],[Build against gtk-2 (default=auto)])],
+  [],[with_gtk2=auto])
+AC_ARG_WITH([gtk3],
+  [AS_HELP_STRING([--with-gtk3],[Build against gtk-3 (default=auto)])],
+  [],[with_gtk3=auto])
+
 if test "x$enable_gtk_theme" = "xyes" -o "x$enable_pixbuf_loader" = "xyes"; then
-  PKG_CHECK_MODULES([GTK2],[gtk+-2.0 >= $GTK2_REQUIRED],[have_gtk_2=yes],[have_gtk_2=no])
-  PKG_CHECK_MODULES([GTK3],[gtk+-3.0 >= $GTK3_REQUIRED],[have_gtk_3=yes],[have_gtk_3=no])
+  AS_IF([test "x$with_gtk2" = "xno"],
+    [have_gtk_2=no],
+    [PKG_CHECK_MODULES([GTK2],[gtk+-2.0 >= $GTK2_REQUIRED],[have_gtk_2=yes],[have_gtk_2=no])])
+  AS_IF([test "x$with_gtk3" = "xno"],
+    [have_gtk_3=no],
+    [PKG_CHECK_MODULES([GTK3],[gtk+-3.0 >= $GTK3_REQUIRED],[have_gtk_3=yes],[have_gtk_3=no])])
 
   if test "$have_gtk_2" = "yes"; then
      GTK2_BINARY_VERSION="`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`"
