diff --git a/configure.ac b/configure.ac
index 1111111..2222222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -124,6 +124,19 @@ AS_IF(test "$major" -lt "$MINIMUM_RUST_MAJOR" -o "$minor" -lt "$MINIMUM_RUST_MIN
     AC_MSG_ERROR([librsvg needs at rustc version at least $MINIMUM_RUST_MAJOR.$MINIMUM_RUST_MINOR])
 )
 
+# See: https://github.com/rust-lang/rust/issues/58277
+AC_MSG_CHECKING([if clzsi2.o workaround is needed])
+case "$major.$minor" in
+  1.34|1.35)
+    clzsi2_hack=yes
+    ;;
+  *)
+    clzsi2_hack=no
+    ;;
+esac
+AC_MSG_RESULT([$clzsi2_hack])
+AM_CONDITIONAL(CLZSI2_HACK, test "$clzsi2_hack" = "yes")
+
 dnl ===========================================================================
 
 #AC_CANONICAL_HOST
diff --git a/Makefile.am b/Makefile.am
index 1111111..2222222 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -131,6 +131,12 @@ if CROSS_COMPILING
 CARGO_TARGET_ARGS=--target=$(RUST_TARGET)
 endif
 
+if CLZSI2_HACK
+DELETE_MODULES=clzsi2.o
+else
+DELETE_MODULES=
+endif
+
 CARGO_VERBOSE = $(cargo_verbose_$(V))
 cargo_verbose_ = $(cargo_verbose_$(AM_DEFAULT_VERBOSITY))
 cargo_verbose_0 =
@@ -177,7 +183,7 @@ librsvg_internals.la: $(librsvg_internals_la_SOURCES)
 	LIBRSVG_BUILD_DIR=$(LIBRSVG_BUILD_DIR)					\
 	LIBRSVG_TARGET_DIR=$(LIBRSVG_TARGET_DIR)					\
 	$(CARGO) --locked build $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) --features "c-library" \
-	&& $(AR) d $(RUST_LIB) clzsi2.o # HACK: https://github.com/rust-lang/rust/issues/58277
+	&& $(AR) d $(RUST_LIB) $(DELETE_MODULES)
 
 librsvg_@RSVG_API_MAJOR_VERSION@_la_CPPFLAGS = \
 	-I$(top_srcdir) 			\

