--- libsoup-2.25.3/configure.in	2008-12-16 04:53:57.000000000 +0530
+++ libsoup-2.25.3.new/configure.in	2009-01-25 15:43:43.891319072 +0530
@@ -179,6 +179,11 @@
 	    :, [if test $os_win32 = yes; then with_gnome=no; else with_gnome=yes; fi])
 AC_MSG_RESULT($with_gnome)
 
+AC_ARG_WITH(libproxy,
+	    AS_HELP_STRING([--without-libproxy],
+			   [Do not build libsoup-gnome with libproxy support]),, with_libproxy=yes)
+AC_MSG_RESULT($with_libproxy)
+
 AM_CONDITIONAL(BUILD_LIBSOUP_GNOME, test $with_gnome != no)
 
 if test $with_gnome != no; then
@@ -186,7 +191,7 @@
 
 	PKG_CHECK_MODULES(LIBPROXY, libproxy-1.0,
 			  have_libproxy=yes, have_libproxy=no)
-	if test $have_libproxy = yes; then
+	if test $have_libproxy = yes -a $with_libproxy != no; then
 		AC_DEFINE(HAVE_LIBPROXY, 1, [Defined if libproxy support is enabled])
 	else
 		PKG_CHECK_MODULES(GCONF, gconf-2.0, , [AC_MSG_ERROR(dnl
@@ -376,3 +381,16 @@
 	docs/Makefile
 	docs/reference/Makefile
 	])
+
+echo "
+	Configuration
+	Debug:		$enable_debug
+	SSL:		$enable_ssl
+	libsoup-gnome:	$with_gnome
+	libproxy:	$with_libproxy
+
+	Tests
+	apache:		$APACHE_HTTPD
+	php:		${have_php:-"no"}
+	curl:		$CURL
+"
