diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN gtk-doc-1.9.orig/Makefile.am gtk-doc-1.9/Makefile.am
--- gtk-doc-1.9.orig/Makefile.am	2007-09-24 15:16:20.000000000 -0400
+++ gtk-doc-1.9/Makefile.am	2008-03-24 10:29:24.000000000 -0400
@@ -35,16 +35,12 @@ gtkdocdata_DATA = \
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = gtk-doc.pc
 
-aclocaldir = $(datadir)/aclocal
-aclocal_DATA = gtk-doc.m4
-
 sgmldir = $(datadir)/sgml/gtk-doc
 sgml_DATA = gtk-doc.cat
 
 EXTRA_DIST = \
   MAINTAINERS     \
   gtk-doc.pc.in   \
-  gtk-doc.m4      \
   gtk-doc.make    \
   gtk-doc.notmpl.make \
   gtk-doc.spec.in \
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN gtk-doc-1.9.orig/Makefile.in gtk-doc-1.9/Makefile.in
--- gtk-doc-1.9.orig/Makefile.in	2007-09-24 15:18:36.000000000 -0400
+++ gtk-doc-1.9/Makefile.in	2008-03-24 10:29:22.000000000 -0400
@@ -260,14 +260,11 @@ gtkdocdata_DATA = \
 
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = gtk-doc.pc
-aclocaldir = $(datadir)/aclocal
-aclocal_DATA = gtk-doc.m4
 sgmldir = $(datadir)/sgml/gtk-doc
 sgml_DATA = gtk-doc.cat
 EXTRA_DIST = \
   MAINTAINERS     \
   gtk-doc.pc.in   \
-  gtk-doc.m4      \
   gtk-doc.make    \
   gtk-doc.notmpl.make \
   gtk-doc.spec.in \
