From b09ddd0f5c9de463ad0d1ebd329a9d16b084e192 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 1 Oct 2008 22:27:24 +0200
Subject: [PATCH] allow shutdown without policykit or RBAC intervention.

---
 src/ck-manager.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/ck-manager.c b/src/ck-manager.c
index bcb9350..54a78e6 100644
--- a/src/ck-manager.c
+++ b/src/ck-manager.c
@@ -1349,7 +1349,6 @@ ck_manager_restart (CkManager             *manager,
         }
 #else
         g_warning ("Compiled without PolicyKit or RBAC support!");
-        goto out;
 #endif
 
         g_debug ("ConsoleKit preforming Restart: %s", action);
@@ -1409,7 +1408,6 @@ ck_manager_stop (CkManager             *manager,
         }
 #else
         g_warning ("Compiled without PolicyKit or RBAC support!");
-        goto out;
 #endif
 
         g_debug ("Stopping system");
-- 
1.6.0.2

