From 612b9e9faf014f9b9ff8fbf58176769583f69d5f Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Tue, 30 Aug 2011 11:52:25 +0000
Subject: Fix batch import for loginDialog

Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Signed-off-by: Adel Gadllah <adel.gadllah@gmail.com>
---
diff --git a/js/gdm/loginDialog.js b/js/gdm/loginDialog.js
index 36799e3..3ba8aca 100644
--- a/js/gdm/loginDialog.js
+++ b/js/gdm/loginDialog.js
@@ -32,7 +32,7 @@ const Shell = imports.gi.Shell;
 const St = imports.gi.St;
 const GdmGreeter = imports.gi.GdmGreeter;
 
-const Batch = imports.misc.batch;
+const Batch = imports.gdm.batch;
 const Lightbox = imports.ui.lightbox;
 const Main = imports.ui.main;
 const ModalDialog = imports.ui.modalDialog;
--
cgit v0.9.0.2
