--- src/main.c
+++ src/main.c
@@ -458,3 +458,17 @@ main (int argc, char **argv)
 
   return meta_run ();
 }
+
+#if HAVE_BLUETOOTH
+/* HACK:
+   Add a non-static function that calls into libgnome-bluetooth-applet.so,
+   to avoid the linker being too smart and removing the dependency.
+   This function is never actually called.
+*/
+extern GType bluetooth_applet_get_type(void);
+void _shell_link_to_bluetooth(void);
+
+void _shell_link_to_bluetooth(void) {
+  bluetooth_applet_get_type();
+}
+#endif
