--- gtk-doc.make	2008-01-14 21:07:04.000000000 -0800
+++ gtk-doc.make	2008-01-14 21:15:30.000000000 -0800
@@ -139,8 +139,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 	
 
--- doc/Makefile.in	2008-01-14 21:59:37.000000000 -0800
+++ doc/Makefile.in	2008-01-14 22:01:06.000000000 -0800
@@ -569,8 +569,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 
 uninstall-local:
