From f23bb0a5aac613c385272ee448afd27ce50b9ee4 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 15 Nov 2010 17:47:54 +0000
Subject: Update for new GTK+ 3.x and libgnome-media-profiles

---
diff --git a/data/sound-juicer.ui b/data/sound-juicer.ui
index 12de20c..50042c7 100644
--- a/data/sound-juicer.ui
+++ b/data/sound-juicer.ui
@@ -615,7 +615,6 @@ audio-volume-medium</property>
     <property name="border_width">5</property>
     <property name="title" translatable="yes">Multiple Albums Found</property>
     <property name="type_hint">dialog</property>
-    <property name="has_separator">False</property>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox2">
         <property name="visible">True</property>
@@ -735,7 +734,6 @@ audio-volume-medium</property>
     <property name="title" translatable="yes">Preferences</property>
     <property name="resizable">False</property>
     <property name="type_hint">dialog</property>
-    <property name="has_separator">False</property>
     <signal name="delete_event" handler="gtk_widget_hide_on_delete"/>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox4">
diff --git a/libjuicer/sj-extractor.c b/libjuicer/sj-extractor.c
index dab6ec0..52f80d2 100644
--- a/libjuicer/sj-extractor.c
+++ b/libjuicer/sj-extractor.c
@@ -30,7 +30,7 @@
 #include <glib-object.h>
 #include <gst/gst.h>
 #include <gst/tag/tag.h>
-#include <profiles/gnome-media-profiles.h>
+#include <libgnome-media-profiles/gnome-media-profiles.h>
 #include "sj-extractor.h"
 #include "sj-structures.h"
 #include "sj-error.h"
diff --git a/libjuicer/sj-extractor.h b/libjuicer/sj-extractor.h
index d9d8717..bbe23fc 100644
--- a/libjuicer/sj-extractor.h
+++ b/libjuicer/sj-extractor.h
@@ -26,7 +26,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gio/gio.h>
-#include <profiles/audio-profile.h>
+#include <libgnome-media-profiles/audio-profile.h>
 #include "sj-structures.h"
 
 G_BEGIN_DECLS
diff --git a/src/sj-main.c b/src/sj-main.c
index 351fb66..f9d3d71 100644
--- a/src/sj-main.c
+++ b/src/sj-main.c
@@ -33,7 +33,7 @@
 #include <gconf/gconf-client.h>
 #include <brasero-medium-selection.h>
 #include <brasero-volume.h>
-#include <profiles/gnome-media-profiles.h>
+#include <libgnome-media-profiles/gnome-media-profiles.h>
 #include <gst/gst.h>
 
 #include "bacon-message-connection.h"
@@ -886,10 +886,10 @@ static void reread_cd (gboolean ignore_no_media)
 
   /* Set watch cursor */
   if (realized) {
-    cursor = gdk_cursor_new_for_display (gdk_drawable_get_display (window), GDK_WATCH);
+    cursor = gdk_cursor_new_for_display (gtk_widget_get_display (GTK_WIDGET (window)), GDK_WATCH);
     gdk_window_set_cursor (window, cursor);
     gdk_cursor_unref (cursor);
-    gdk_display_sync (gdk_drawable_get_display (window));
+    gdk_display_sync (gtk_widget_get_display (GTK_WIDGET (window)));
   }
 
   /* Set statusbar message */
diff --git a/src/sj-prefs.c b/src/sj-prefs.c
index 698b4aa..aeb2bc5 100644
--- a/src/sj-prefs.c
+++ b/src/sj-prefs.c
@@ -25,7 +25,7 @@
 #include <string.h>
 #include <gtk/gtk.h>
 #include <gconf/gconf-client.h>
-#include <profiles/gnome-media-profiles.h>
+#include <libgnome-media-profiles/gnome-media-profiles.h>
 #include <brasero-drive-selection.h>
 
 #include "sj-util.h"
--
cgit v0.8.3.4
