From 83398a1e2d1f1045353884b44c681e035b561023 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 15 Nov 2010 17:43:37 +0000
Subject: build: Remove ability to build against GTK+ 2.x

---
diff --git a/configure.in b/configure.in
index 8727f14..9e5f0c8 100644
--- a/configure.in
+++ b/configure.in
@@ -34,30 +34,7 @@ GNOME_CXX_WARNINGS
 GNOME_DEBUG_CHECK
 GNOME_MAINTAINER_MODE_DEFINES
 
-AC_MSG_CHECKING([which gtk+ version to compile against])
-AC_ARG_WITH([gtk],
-  [AS_HELP_STRING([--with-gtk=2.0|3.0],[which gtk+ version to compile against (default: 2.0)])],
-  [case "$with_gtk" in
-     2.0|3.0) ;;
-     *) AC_MSG_ERROR([invalid gtk version specified]) ;;
-   esac],
-  [with_gtk=2.0])
-AC_MSG_RESULT([$with_gtk])
-
-case "$with_gtk" in
-  2.0) GTK_API_VERSION=2.0
-       GTK_REQUIRED=2.20.0
-       LIBBRASERO_MEDIA_PC="libbrasero-media >= 2.26"
-       LIBCANBERRA_GTK_PC=libcanberra-gtk
-       GNOME_MEDIA_PROFILES_PC="gnome-media-profiles >= 2.11.91"
-       ;;
-  3.0) GTK_API_VERSION=3.0
-       GTK_REQUIRED=2.90.0
-       LIBBRASERO_MEDIA_PC=libbrasero-media3
-       LIBCANBERRA_GTK_PC=libcanberra-gtk3
-       GNOME_MEDIA_PROFILES_PC=gnome-media-profiles-3.0
-       ;;
-esac
+GTK_REQUIRED=2.90.0
 
 AC_CHECK_FUNC(socket,,[AC_CHECK_LIB(socket,socket)])
 
@@ -70,7 +47,7 @@ AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
 # Find libbrasero-media
-PKG_CHECK_MODULES(BURN, $LIBBRASERO_MEDIA_PC)
+PKG_CHECK_MODULES(BURN, libbrasero-media3)
 AC_SUBST(BURN_CFLAGS)
 AC_SUBST(BURN_LIBS)
 
@@ -81,7 +58,7 @@ LIBS="$LIBS $BURN_LIBS"
 # LIBS="$oldlibs"
 
 # Find the UI libraries
-PKG_CHECK_MODULES(UI, gtk+-$GTK_API_VERSION >= $GTK_REQUIRED gconf-2.0 gio-2.0 gmodule-export-2.0 dbus-glib-1 $LIBCANBERRA_GTK_PC)
+PKG_CHECK_MODULES(UI, gtk+-3.0 >= $GTK_REQUIRED gconf-2.0 gio-2.0 gmodule-export-2.0 dbus-glib-1 libcanberra-gtk3)
 AC_SUBST(UI_CFLAGS)
 AC_SUBST(UI_LIBS)
 
@@ -96,7 +73,7 @@ AM_GST_ELEMENT_CHECK(wavenc,,AC_MSG_WARN([The 'wavenc' element was not found. Th
 AM_GST_ELEMENT_CHECK(giosink,,AC_MSG_WARN([The 'giosink' element was not found. This will cause Sound Juicer to fail at runtime.]))
 
 # Find GNOME Media Profiles
-PKG_CHECK_MODULES(MEDIA_PROFILES, $GNOME_MEDIA_PROFILES_PC)
+PKG_CHECK_MODULES(MEDIA_PROFILES, libgnome-media-profiles-3.0)
 AC_SUBST(MEDIA_PROFILES_CFLAGS)
 AC_SUBST(MEDIA_PROFILES_LIBS)
 
--
cgit v0.8.3.4
