--- libmenu/menu-monitor-gio.c.orig	2008-01-15 21:11:58.000000000 -0800
+++ libmenu/menu-monitor-gio.c	2008-01-15 21:12:24.000000000 -0800
@@ -83,10 +83,10 @@
 
   if (menu_monitor_get_is_directory (monitor))
       file_monitor = g_file_monitor_directory (file, G_FILE_MONITOR_NONE,
-                                               NULL, NULL);
+                                               NULL);
   else
       file_monitor = g_file_monitor_file (file, G_FILE_MONITOR_NONE,
-                                          NULL, NULL);
+                                          NULL);
 
   g_object_unref (G_OBJECT (file));
 
