diff --exclude-from=/home/dang/.diffrc -up -ruN pkg-config-0.21.orig/check/check-requires-private pkg-config-0.21/check/check-requires-private
--- pkg-config-0.21.orig/check/check-requires-private	2006-08-16 13:05:30.000000000 -0400
+++ pkg-config-0.21/check/check-requires-private	2007-01-03 20:49:56.000000000 -0500
@@ -9,14 +9,14 @@ set -e
 
 . ${srcdir}/common
 
-# expect cflags from requires-test and public-dep
+# expect cflags from requires-test, public-dep, and private-dep
 ARGS="--cflags requires-test"
-RESULT="-I/requires-test/include -I/public-dep/include"
+RESULT="-I/requires-test/include -I/private-dep/include -I/public-dep/include"
 run_test
 
-# still expect those cflags for static linking case
+# still expect cflags from requires-test, public-dep, and private-dep
 ARGS="--static --cflags requires-test"
-RESULT="-I/requires-test/include -I/public-dep/include"
+RESULT="-I/requires-test/include -I/private-dep/include -I/public-dep/include"
 run_test
 
 # expect libs for just requires-test and public-dep
