From 1f27a7fb4ccc8b119e60c8281a71f1e3309fd959 Mon Sep 17 00:00:00 2001
From: Raul Gutierrez Segales <rgs@collabora.co.uk>
Date: Thu, 01 Sep 2011 16:19:58 +0000
Subject: Introspection: e_source_set_proproperty's property_value param can be null

---
diff --git a/libedataserver/e-source.c b/libedataserver/e-source.c
index bbcaad4..971ec5a 100644
--- a/libedataserver/e-source.c
+++ b/libedataserver/e-source.c
@@ -1135,7 +1135,7 @@ e_source_get_duped_property (ESource *source,
  * e_source_set_property:
  * @source: an #ESource
  * @property_name: a custom property name
- * @property_value: a new value for the property, or %NULL
+ * @property_value: (allow-none): a new value for the property, or %NULL
  *
  * Create a new custom #ESource property or replaces an existing one.  If
  * @property_value is %NULL, the property is removed from @source.  This
--
cgit v0.9.0.2
