From a3ba2ff92fd84d89d0fd97dce77cfad71bef1aa8 Mon Sep 17 00:00:00 2001
From: Raul Gutierrez Segales <rgs@collabora.co.uk>
Date: Wed, 31 Aug 2011 09:49:44 +0000
Subject: Introspection: add the correct header to libecal's GIR file

---
diff --git a/calendar/libecal/Makefile.am b/calendar/libecal/Makefile.am
index 1ae10ad..de74a1e 100644
--- a/calendar/libecal/Makefile.am
+++ b/calendar/libecal/Makefile.am
@@ -72,7 +72,8 @@ DISTCLEANFILES = $(pkgconfig_DATA)
 
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
-INTROSPECTION_SCANNER_ARGS = --add-include-path=$(top_builddir)/libedataserver --identifier-prefix E --pkg-export libecal-1.2
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(top_builddir)/libedataserver --identifier-prefix E --pkg-export libecal-1.2 \
+	--c-include="libecal/e-cal.h"
 INTROSPECTION_COMPILER_ARGS = --includedir=$(top_builddir)/libedataserver
 
 if HAVE_INTROSPECTION
--
cgit v0.9.0.2
