diff --git a/calendar/libecal/Makefile.am b/calendar/libecal/Makefile.am
index 47831ba..e3320f9 100644
--- a/calendar/libecal/Makefile.am
+++ b/calendar/libecal/Makefile.am
@@ -31,8 +31,8 @@ libecal_1_2_la_SOURCES =			\
 	e-cal-view-private.h
 
 libecal_1_2_la_LIBADD = 					\
-	$(top_builddir)/calendar/libegdbus/libegdbus-cal.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la 	\
+	$(top_builddir)/calendar/libegdbus/libegdbus-cal.la	\
 	$(LIBICAL_LIBS)						\
 	$(EVOLUTION_CALENDAR_LIBS)
 
diff --git a/calendar/libedata-cal/Makefile.am b/calendar/libedata-cal/Makefile.am
index 8b2f5f0..c9dc25b 100644
--- a/calendar/libedata-cal/Makefile.am
+++ b/calendar/libedata-cal/Makefile.am
@@ -36,8 +36,8 @@ libedata_cal_1_2_la_SOURCES =		\
 	e-data-cal-view.c
 
 libedata_cal_1_2_la_LIBADD =					\
-	$(top_builddir)/calendar/libegdbus/libegdbus-cal.la	\
 	$(top_builddir)/calendar/libecal/libecal-1.2.la		\
+	$(top_builddir)/calendar/libegdbus/libegdbus-cal.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la 	\
 	$(top_builddir)/libebackend/libebackend-1.2.la 		\
 	$(LIBICAL_LIBS)						\