diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/addressbook/backends/Makefile.am evolution-data-server-2.21.4/addressbook/backends/Makefile.am
--- evolution-data-server-2.21.4.orig/addressbook/backends/Makefile.am	2007-10-23 05:57:41.000000000 -0400
+++ evolution-data-server-2.21.4/addressbook/backends/Makefile.am	2007-12-17 13:25:00.000000000 -0500
@@ -4,4 +4,4 @@ else
 LDAP_SUBDIR =
 endif
 
-SUBDIRS = file vcf $(LDAP_SUBDIR) groupwise 
+SUBDIRS = file vcf $(LDAP_SUBDIR)
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/calendar/backends/Makefile.am evolution-data-server-2.21.4/calendar/backends/Makefile.am
--- evolution-data-server-2.21.4.orig/calendar/backends/Makefile.am	2007-10-24 02:39:45.000000000 -0400
+++ evolution-data-server-2.21.4/calendar/backends/Makefile.am	2007-12-17 13:33:56.000000000 -0500
@@ -1 +1 @@
-SUBDIRS = file groupwise http contacts weather caldav google
+SUBDIRS = file http contacts weather caldav
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/camel/providers/Makefile.am evolution-data-server-2.21.4/camel/providers/Makefile.am
--- evolution-data-server-2.21.4.orig/camel/providers/Makefile.am	2007-10-23 05:57:04.000000000 -0400
+++ evolution-data-server-2.21.4/camel/providers/Makefile.am	2007-12-17 13:24:50.000000000 -0500
@@ -17,6 +17,6 @@ else
 SENDMAIL_DIR=sendmail
 endif
 
-SUBDIRS = pop3 $(SENDMAIL_DIR) smtp imap $(IMAP4_DIR) $(NNTP_DIR) local $(IMAPP_DIR) groupwise hula
+SUBDIRS = pop3 $(SENDMAIL_DIR) smtp imap $(IMAP4_DIR) $(NNTP_DIR) local $(IMAPP_DIR) hula
 
 
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/Makefile.am evolution-data-server-2.21.4/Makefile.am
--- evolution-data-server-2.21.4.orig/Makefile.am	2007-10-23 05:59:44.000000000 -0400
+++ evolution-data-server-2.21.4/Makefile.am	2007-12-17 13:33:44.000000000 -0500
@@ -7,9 +7,12 @@ endif
 
 if ENABLE_CALENDAR
 CALENDAR_DIR = calendar
+CALENDAR_LATER = calendar/backends/groupwise calendar/backends/google
 endif
 
-SUBDIRS = $(LIBDB) win32 libedataserver servers camel addressbook $(CALENDAR_DIR) libedataserverui src docs art po
+LATER = camel/providers/groupwise addressbook/backends/groupwise $(CALENDAR_LATER)
+
+SUBDIRS = $(LIBDB) win32 libedataserver camel addressbook $(CALENDAR_DIR) libedataserverui servers $(LATER) src docs art po
 DIST_SUBDIRS= win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
 
 changelogs =			\
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/servers/exchange/storage/Makefile.am evolution-data-server-2.21.4/servers/exchange/storage/Makefile.am
--- evolution-data-server-2.21.4.orig/servers/exchange/storage/Makefile.am	2007-10-23 05:57:26.000000000 -0400
+++ evolution-data-server-2.21.4/servers/exchange/storage/Makefile.am	2007-12-17 13:20:50.000000000 -0500
@@ -63,6 +63,7 @@ libexchange_storage_1_2_la_LIBADD =				\
 	$(top_builddir)/servers/exchange/lib/libexchange.la	\
 	$(top_builddir)/servers/exchange/xntlm/libxntlm.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la	\
+	$(top_builddir)/libedataserverui/libedataserverui-1.2.la	\
 	$(E_DATA_SERVER_LIBS)					\
 	$(E_DATA_SERVER_UI_LIBS)				\
 	$(LDAP_LIBS)						\
