diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.23.6.orig/addressbook/backends/Makefile.am evolution-data-server-2.23.6/addressbook/backends/Makefile.am
--- evolution-data-server-2.23.6.orig/addressbook/backends/Makefile.am	2008-08-04 00:28:27.000000000 -0400
+++ evolution-data-server-2.23.6/addressbook/backends/Makefile.am	2008-08-05 12:26:15.000000000 -0400
@@ -4,4 +4,4 @@ else
 LDAP_SUBDIR =
 endif
 
-SUBDIRS = file vcf $(LDAP_SUBDIR) google groupwise webdav
+SUBDIRS = file vcf $(LDAP_SUBDIR)
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.23.6.orig/calendar/backends/Makefile.am evolution-data-server-2.23.6/calendar/backends/Makefile.am
--- evolution-data-server-2.23.6.orig/calendar/backends/Makefile.am	2008-04-11 07:10:31.000000000 -0400
+++ evolution-data-server-2.23.6/calendar/backends/Makefile.am	2008-08-05 12:25:51.000000000 -0400
@@ -1 +1 @@
-SUBDIRS = file groupwise http contacts weather caldav google
+SUBDIRS = file http contacts weather caldav
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.23.6.orig/camel/providers/Makefile.am evolution-data-server-2.23.6/camel/providers/Makefile.am
--- evolution-data-server-2.23.6.orig/camel/providers/Makefile.am	2008-08-01 02:34:59.000000000 -0400
+++ evolution-data-server-2.23.6/camel/providers/Makefile.am	2008-08-05 12:26:59.000000000 -0400
@@ -21,6 +21,6 @@ else
 SENDMAIL_DIR=sendmail
 endif
 
-SUBDIRS = pop3 $(SENDMAIL_DIR) smtp imap $(IMAP4_DIR) $(NNTP_DIR) local $(IMAPP_DIR) groupwise $(HULA_DIR)
+SUBDIRS = pop3 $(SENDMAIL_DIR) smtp imap $(IMAP4_DIR) $(NNTP_DIR) local $(IMAPP_DIR) $(HULA_DIR)
 
 
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.23.6.orig/Makefile.am evolution-data-server-2.23.6/Makefile.am
--- evolution-data-server-2.23.6.orig/Makefile.am	2008-08-01 02:35:37.000000000 -0400
+++ evolution-data-server-2.23.6/Makefile.am	2008-08-05 12:28:22.000000000 -0400
@@ -7,9 +7,12 @@ endif
 
 if ENABLE_CALENDAR
 CALENDAR_DIR = calendar
+CALENDAR_LATER = calendar/backends/groupwise calendar/backends/google
 endif
 
-SUBDIRS = $(LIBDB) win32 libedataserver libebackend servers camel addressbook $(CALENDAR_DIR) libedataserverui src docs art po
+LATER = camel/providers/groupwise addressbook/backends/groupwise addressbook/backends/google addressbook/backends/webdav $(CALENDAR_LATER)
+
+SUBDIRS = $(LIBDB) win32 libedataserver libebackend camel addressbook $(CALENDAR_DIR) libedataserverui servers $(LATER) src docs art po
 DIST_SUBDIRS= win32 libedataserver libebackend servers camel addressbook calendar libedataserverui src docs art po	
 changelogs =			\
 	ChangeLog
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.23.6.orig/servers/exchange/storage/Makefile.am evolution-data-server-2.23.6/servers/exchange/storage/Makefile.am
--- evolution-data-server-2.23.6.orig/servers/exchange/storage/Makefile.am	2008-04-11 07:10:16.000000000 -0400
+++ evolution-data-server-2.23.6/servers/exchange/storage/Makefile.am	2008-08-05 12:25:51.000000000 -0400
@@ -64,6 +64,7 @@ libexchange_storage_1_2_la_LIBADD =				\
 	$(top_builddir)/servers/exchange/lib/libexchange.la	\
 	$(top_builddir)/servers/exchange/xntlm/libxntlm.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la	\
+	$(top_builddir)/libedataserverui/libedataserverui-1.2.la	\
 	$(E_DATA_SERVER_LIBS)					\
 	$(E_DATA_SERVER_UI_LIBS)				\
 	$(LDAP_LIBS)						\
