From af60abe51121f6ab98d0c67d762e4e1da43a3e71 Mon Sep 17 00:00:00 2001
From: Akhil Laddha <lakhil@novell.com>
Date: Wed, 06 Jul 2011 04:47:04 +0000
Subject: Bug#654054 Fix eds build failure

---
diff --git a/addressbook/libedata-book/Makefile.am b/addressbook/libedata-book/Makefile.am
index bd04300..7506b63 100644
--- a/addressbook/libedata-book/Makefile.am
+++ b/addressbook/libedata-book/Makefile.am
@@ -71,7 +71,7 @@ e_book_backend_sqlitedb_test_CPPFLAGS = \
 	$(libedata_book_1_2_la_CPPFLAGS)
 e_book_backend_sqlitedb_test_SOURCES =	e-book-backend-sqlitedb-test.c
 e_book_backend_sqlitedb_test_LDADD = \
-	$(top_builddir)/addressbook/libedata-book/libedata-book-1.2.la \
+	libedata-book-1.2.la \
 	$(libedata_book_1_2_la_LIBADD)
 
 factory_PROGRAMS = e-addressbook-factory
@@ -95,7 +95,7 @@ e_addressbook_factory_SOURCES = \
 	e-data-book-migrate-basedir.c
 
 e_addressbook_factory_LDADD = \
-	$(top_builddir)/addressbook/libedata-book/libedata-book-1.2.la \
+	libedata-book-1.2.la \
 	$(top_builddir)/libedataserver/libedataserver-1.2.la \
 	$(top_builddir)/libebackend/libebackend-1.2.la \
 	$(EVOLUTION_ADDRESSBOOK_LIBS) \
diff --git a/calendar/libedata-cal/Makefile.am b/calendar/libedata-cal/Makefile.am
index 1145bd9..5bbd4e6 100644
--- a/calendar/libedata-cal/Makefile.am
+++ b/calendar/libedata-cal/Makefile.am
@@ -111,7 +111,7 @@ e_calendar_factory_CPPFLAGS = \
 e_calendar_factory_LDADD =						\
 	$(top_builddir)/calendar/libecal/libecal-1.2.la			\
 	$(top_builddir)/calendar/libegdbus/libegdbus-cal.la		\
-	$(top_builddir)/calendar/libedata-cal/libedata-cal-1.2.la	\
+	libedata-cal-1.2.la						\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la 		\
 	$(top_builddir)/libebackend/libebackend-1.2.la
 
--
cgit v0.9
