diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-data-server-2.21.4.orig/gtk-doc.make evolution-data-server-2.21.4/gtk-doc.make
--- evolution-data-server-2.21.4.orig/gtk-doc.make	2007-12-16 22:56:57.000000000 -0500
+++ evolution-data-server-2.21.4/gtk-doc.make	2007-12-17 10:51:40.000000000 -0500
@@ -139,8 +139,7 @@ install-data-local:
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
-	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) || true ; \
 	fi
 	
 
