diff -Nur nautilus-sendto-1.1.1/src/plugins/pidgin/pidgin.c nautilus-sendto-1.1.1_patched/src/plugins/pidgin/pidgin.c
--- nautilus-sendto-1.1.1/src/plugins/pidgin/pidgin.c	2009-01-07 14:17:14.000000000 +0200
+++ nautilus-sendto-1.1.1_patched/src/plugins/pidgin/pidgin.c	2009-02-22 09:42:01.000000000 +0200
@@ -145,7 +145,7 @@
 		for (i = 0; i < accounts; ++i) {
 			dat = g_ptr_array_index (contacts_group, i);
 
-			if (strcmp(dat->prt, "prpl-msn")==0)
+			if (strcmp(dat->prt, "prpl-msn")==0 || strcmp(dat->prt, "prpl-msn-pecan")==0)
 				icon = msn;
 			else if (strcmp(dat->prt,"prpl-jabber")==0)
 				icon = jabber;
