Index: configure.in
===================================================================
--- configure.in	(révision 876)
+++ configure.in	(révision 878)
@@ -47,7 +47,7 @@
 	
 PKG_CHECK_MODULES(VINO_CAPPLET, glib-2.0 >= 2.12.0 gtk+-2.0 >= 2.10.0 gconf-2.0 libglade-2.0 libgnomeui-2.0 dbus-glib-1)
 
-PKG_CHECK_MODULES(VINO_TOOLS, glib-2.0 >= 2.12.0 gconf-2.0)
+PKG_CHECK_MODULES(VINO_TOOLS, glib-2.0 >= 2.12.0 gconf-2.0 gobject-2.0 >= 2.12.0 gnome-keyring-1)
 
 dnl --enable-libnotify=(yes|no|auto)
 LIBNOTIFY_VERSION=0.4.4
Index: ChangeLog
===================================================================
--- ChangeLog	(révision 876)
+++ ChangeLog	(révision 878)
@@ -1,3 +1,12 @@
+2008-07-25  Matthias Clasen  <mclasen@redhat.com>
+
+	* capplet/vino-preferences.glade: Use a standard icon name.
+	Closes #544639.
+
+2008-07-24  Halton Huo <halton.huo@sun.com>
+
+	* configure.in: Fix build error in folder tools. Closes #544277.
+
 2008-07-22  Jonh Wendell  <jwendell@gnome.org>
 
 	* configure.in: post-release bump version to 2.23.6
Index: capplet/vino-preferences.glade
===================================================================
--- capplet/vino-preferences.glade	(révision 876)
+++ capplet/vino-preferences.glade	(révision 878)
@@ -394,7 +394,7 @@
                             <property name="yalign">0</property>
                             <property name="stock">gtk-missing-image</property>
                             <property name="icon_size">6</property>
-                            <property name="icon_name">gnome-fs-network</property>
+                            <property name="icon_name">network-workgroup</property>
                           </widget>
                           <packing>
                             <property name="expand">False</property>
