From a2cb0e91f58b22832f4efc51acf1d1dd3efb6969 Mon Sep 17 00:00:00 2001
From: Maxim Britov <maxim@office.modum.by>
Date: Tue, 6 Oct 2009 19:16:51 +0200
Subject: [PATCH] Fix missing $(PANEL_CFLAGS) for clock applet

---
 applets/clock/Makefile.am |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/applets/clock/Makefile.am b/applets/clock/Makefile.am
index da5e0d9..a9b95f9 100644
--- a/applets/clock/Makefile.am
+++ b/applets/clock/Makefile.am
@@ -8,6 +8,7 @@ INCLUDES =							\
 	$(WARN_CFLAGS)						\
 	$(CLOCK_CFLAGS)						\
 	$(POLKIT_CFLAGS)					\
+	$(PANEL_CFLAGS)						\
 	$(LIBPANEL_APPLET_CFLAGS)				\
 	-DDATADIR=\""$(datadir)"\"				\
 	-DBUILDERDIR=\""$(uidir)"\"				\
-- 
1.6.5.rc2

