Index: applets/clock/calendar-window.c
===================================================================
--- applets/clock/calendar-window.c	(revision 10739)
+++ applets/clock/calendar-window.c	(working copy)
@@ -1534,7 +1534,9 @@
                 g_signal_connect_swapped (button, "clicked", callback, calwin);
         }
 
+#ifdef HAVE_LIBECAL
 	connect_expander_with_gconf (calwin, expander, key);
+#endif
 
         return vbox;
 }
