---
 Allow building without sandbox violations.

diff --git a/libgda-report/RML/trml2html/Makefile.am b/libgda-report/RML/trml2html/Makefile.am
index 9b1e390..dd050cc 100644
--- a/libgda-report/RML/trml2html/Makefile.am
+++ b/libgda-report/RML/trml2html/Makefile.am
@@ -3,10 +3,10 @@ files = __init__.py  \
 	utils.py
 
 install-exec-local:
-	-mkdir -p $(bindir)/gda_trml2html
-	-cd $(top_srcdir)/libgda-report/RML/trml2html && $(INSTALL_PROGRAM) $(files) $(bindir)/gda_trml2html
+	-mkdir -p $(DESTDIR)$(bindir)/gda_trml2html
+	-cd $(top_srcdir)/libgda-report/RML/trml2html && $(INSTALL_PROGRAM) $(files) $(DESTDIR)$(bindir)/gda_trml2html
 
 uninstall-local:
-	-rm -rf $(bindir)/gda_trml2html
+	-rm -rf $(DESTDIR)$(bindir)/gda_trml2html
 
 EXTRA_DIST = $(files)
diff --git a/libgda-report/RML/trml2pdf/Makefile.am b/libgda-report/RML/trml2pdf/Makefile.am
index 7c241fe..aeab872 100644
--- a/libgda-report/RML/trml2pdf/Makefile.am
+++ b/libgda-report/RML/trml2pdf/Makefile.am
@@ -4,10 +4,10 @@ files = color.py \
 	utils.py
 
 install-exec-local:
-	-mkdir -p $(bindir)/gda_trml2pdf
-	-cd $(top_srcdir)/libgda-report/RML/trml2pdf && $(INSTALL_PROGRAM) $(files) $(bindir)/gda_trml2pdf
+	-mkdir -p $(DESTDIR)$(bindir)/gda_trml2pdf
+	-cd $(top_srcdir)/libgda-report/RML/trml2pdf && $(INSTALL_PROGRAM) $(files) $(DESTDIR)$(bindir)/gda_trml2pdf
 
 uninstall-local:
-	-rm -rf $(bindir)/gda_trml2pdf
+	-rm -rf $(DESTDIR)$(bindir)/gda_trml2pdf
 
 EXTRA_DIST = $(files)
