From f39858f84155d0346d31a2607997257a54e246ff Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 29 Sep 2010 23:45:46 +0200
Subject: [PATCH] fix build order for xml_embedded.h

---
 libgda/sqlite/Makefile.am |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/libgda/sqlite/Makefile.am b/libgda/sqlite/Makefile.am
index 1d2df2c..ecce04b 100644
--- a/libgda/sqlite/Makefile.am
+++ b/libgda/sqlite/Makefile.am
@@ -58,9 +58,9 @@ sqlitesources = \
 	gda-sqlite.h \
 	keywords_hash.h \
 	gda-symbols-util.h \
-	gda-symbols-util.c
-
-$(libgda_sqlite_la_OBJECTS): xml_embedded.h keywords_hash.c
+	gda-symbols-util.c \
+	xml_embedded.h \
+	keywords_hash.c
 
 libgda_sqlite_la_SOURCES = $(sqlitesources)
 libgda_sqlite_la_CFLAGS = -DPNAME=\""SQLite"\" -DCLASS_PREFIX=\""GdaSqlite"\" -DSEARCH_LIB_PATH=\""$(SQLITE_PATH)"\"
@@ -85,4 +85,5 @@ gdainclude_HEADERS=$(sqliteheaders)
 
 EXTRA_DIST = gen_emb_string.c mkkeywordhash.c keywords.list
 
-CLEANFILES = gen_emb_string$(EXEEXT_FOR_BUILD) xml_embedded.h mkkeywordhash$(EXEEXT_FOR_BUILD) keywords_hash.c
+CLEANFILES = gen_emb_string$(EXEEXT_FOR_BUILD) mkkeywordhash$(EXEEXT_FOR_BUILD) $(BUILT_SOURCES)
+BUILT_SOURCES = xml_embedded.h keywords_hash.c
-- 
1.7.3

