diff -Naur libproxy-0.4.6/.cproject libproxy-0.4.6.new/.cproject
--- libproxy-0.4.6/.cproject	2010-09-02 01:53:34.000000000 +0530
+++ libproxy-0.4.6.new/.cproject	2011-01-24 22:57:36.767000067 +0530
@@ -167,7 +167,7 @@
 <pathentry include="/usr/include/NetworkManager" kind="inc" path="" system="true"/>
 <pathentry include="/usr/include/dbus-1.0" kind="inc" path="" system="true"/>
 <pathentry include="/usr/lib64/dbus-1.0/include" kind="inc" path="" system="true"/>
-<pathentry include="/usr/include/webkit-1.0" kind="inc" path="" system="true"/>
+<pathentry include="/usr/include/webkit-3.0" kind="inc" path="" system="true"/>
 <pathentry include="/usr/include/glib-2.0" kind="inc" path="" system="true"/>
 <pathentry include="/usr/lib64/glib-2.0/include" kind="inc" path="" system="true"/>
 <pathentry include="/usr/include/gtk-2.0" kind="inc" path="" system="true"/>
diff -Naur libproxy-0.4.6/libproxy/cmake/modules/pacrunner_webkit.cmk libproxy-0.4.6.new/libproxy/cmake/modules/pacrunner_webkit.cmk
--- libproxy-0.4.6/libproxy/cmake/modules/pacrunner_webkit.cmk	2010-09-02 01:53:34.000000000 +0530
+++ libproxy-0.4.6.new/libproxy/cmake/modules/pacrunner_webkit.cmk	2011-01-24 22:57:46.410000066 +0530
@@ -12,5 +12,5 @@
     set(WEBKIT_FOUND 1)
   endif()
 else()
-  px_check_modules(WEBKIT webkit-1.0)
-endif()
\ No newline at end of file
+  px_check_modules(WEBKIT webkitgtk-3.0)
+endif()
