From 09f3a2a4b6d997b5fd9b42add27de465dd529d43 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Sun, 26 Feb 2012 18:15:50 -0500
Subject: [PATCH] Make vala dependency non-automagic

---
 configure.ac |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2e21d48..40cfe36 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,7 +13,15 @@ AM_SILENT_RULES([yes])
 
 # Check for programs
 AC_PROG_CC
-AM_PROG_VALAC([0.15.1])
+
+AC_ARG_ENABLE(vala,
+              AC_HELP_STRING([--disable-vala],
+                             [Disable vala support]))
+AM_CONDITIONAL(ENABLE_VALA, test "x$enable_vala" != "xno")
+
+if test "x$enable_vala" != "xno"; then
+  AM_PROG_VALAC([0.15.1])
+fi
 
 # Use GSettings
 GLIB_GSETTINGS
-- 
1.7.8.4

