diff -uNr file-roller-2.10.0/src/fr-command-rpm.c file-roller-2.10.0.patched/src/fr-command-rpm.c
--- file-roller-2.10.0/src/fr-command-rpm.c	2005-03-07 13:29:37.000000000 +0100
+++ file-roller-2.10.0.patched/src/fr-command-rpm.c	2005-03-29 12:40:38.000000000 +0200
@@ -131,7 +131,7 @@
 				      list__process_line,
 				      comm);
 
-	fr_process_begin_command (comm->process, "rpm2cpio");
+	fr_process_begin_command (comm->process, "rpm2cpio-file-roller");
 	fr_process_add_arg (comm->process, comm->e_filename);
 	fr_process_add_arg (comm->process, "| cpio --list --force-local --verbose");
 	fr_process_end_command (comm->process);
@@ -150,7 +150,7 @@
 {
 	GList *scan;
 
-	fr_process_begin_command (comm->process, "rpm2cpio");
+	fr_process_begin_command (comm->process, "rpm2cpio-file-roller");
 	if (dest_dir != NULL)
                 fr_process_set_working_dir (comm->process, dest_dir);
 	fr_process_add_arg (comm->process, comm->e_filename);
@@ -240,7 +240,7 @@
 {
 	FRCommand *comm;
 
-	if (!is_program_in_path("rpm2cpio")) {
+	if (!is_program_in_path("rpm2cpio-file-roller")) {
 		return NULL;
 	}
 
