--- a/libempathy-gtk/empathy-avatar-chooser.c	2009-10-19 14:08:26.000000000 +0200
+++ b/libempathy-gtk/empathy-avatar-chooser.c	2009-10-21 23:57:04.000000000 +0200
@@ -811,8 +811,8 @@
 						NULL, NULL);
 
 		if (handled) {
-			/* this in turn calls empathy_avatar_new(), which assumes
-			 * ownership of data.
+			/* this in turn calls empathy_avatar_new (), which
+			 * assumes ownership of data.
 			 */
 			avatar_chooser_set_image_from_data (chooser, data,
 							    bytes_read,
