From 1915baa92e72242450b6042603ad0e20de072c17 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dominique-gnomezilla@leuenberger.net>
Date: Sun, 19 Jun 2011 14:46:15 +0000
Subject: libview: fix build with --enable-introspection

Fixes bug #643382.
---
diff --git a/libview/Makefile.am b/libview/Makefile.am
index 7766c4b..2e8671e 100644
--- a/libview/Makefile.am
+++ b/libview/Makefile.am
@@ -136,6 +136,7 @@ EvinceView-$(EV_API_VERSION).gir: libevview3.la Makefile $(INST_H_FILES) $(filte
 	--output $@ \
 	--pkg evince-document-$(EV_API_VERSION) \
 	--pkg evince-view-$(EV_API_VERSION) \
+	-L$(top_builddir)/libdocument/ \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	-DEVINCE_COMPILATION \
--
cgit v0.9.0.2
