diff --exclude-from=/home/dang/.diffrc -up -ruN evince-0.6.1.orig/configure.ac evince-0.6.1/configure.ac
--- evince-0.6.1.orig/configure.ac	2006-10-09 14:58:09.000000000 -0400
+++ evince-0.6.1/configure.ac	2006-12-17 11:41:10.000000000 -0500
@@ -109,6 +109,9 @@ AM_CONDITIONAL(WITH_GTK_PRINT, test x$en
 
 
 dnl ========= Check for DBUS
+AC_ARG_ENABLE(dbus,
+ 	[AC_HELP_STRING([--enable-dbus], [Compile with support for dbus])],enable_dbus="$enableval",enable_dbus=yes)
+if test "x$enable_dbus" != "xno"; then
 PKG_CHECK_MODULES([DBUS], [dbus-glib-1 >= $DBUS_GLIB_REQUIRED],
 	[enable_dbus=yes],[enable_dbus=no])
 AC_SUBST([DBUS_CFLAGS])
@@ -130,6 +133,7 @@ if test "x$enable_dbus" = "xyes" ; then
 	DBUS_VERSION=`$PKG_CONFIG --modversion dbus-glib-1 | sed 's/0\.\([[0-9]]*\)\.\{0,1\}\([[0-9]]*\)/\1/'`
 	AC_DEFINE_UNQUOTED(DBUS_VERSION, $DBUS_VERSION, [DBUS version.])
 fi
+fi
 
 SHELL_CFLAGS="$SHELL_CORE_CFLAGS $DBUS_CFLAGS"
 SHELL_LIBS="$SHELL_CORE_LIBS $DBUS_LIBS -lz"
