From 69445fce2109ae420ce05a9bf4a1fcf63641fcad Mon Sep 17 00:00:00 2001
From: Hib Eris <hib@hiberis.nl>
Date: Tue, 08 Jun 2010 08:38:28 +0000
Subject: [impress] Fix build

---
diff --git a/backend/impress/impress-document.c b/backend/impress/impress-document.c
index 9b55a45..6942951 100644
--- a/backend/impress/impress-document.c
+++ b/backend/impress/impress-document.c
@@ -59,7 +59,7 @@ struct _ImpressDocument
 
 typedef struct _ImpressDocumentClass ImpressDocumentClass;
 
-static void impress_document_document_thumbnails_iface_init (EvDocumentThumbnailsIface *iface);
+static void impress_document_document_thumbnails_iface_init (EvDocumentThumbnailsInterface *iface);
 
 EV_BACKEND_REGISTER_WITH_CODE (ImpressDocument, impress_document,
 		         {
@@ -516,7 +516,7 @@ impress_document_thumbnails_get_dimensions (EvDocumentThumbnails *document,
 }
 
 static void
-impress_document_document_thumbnails_iface_init (EvDocumentThumbnailsIface *iface)
+impress_document_document_thumbnails_iface_init (EvDocumentThumbnailsInterface *iface)
 {
   iface->get_thumbnail = impress_document_thumbnails_get_thumbnail;
   iface->get_dimensions = impress_document_thumbnails_get_dimensions;
--
cgit v0.8.3.1
