From 9dd59f719ba098ef93aacf2e5e3d0974cfcd5b2c Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <carlosgc@gnome.org>
Date: Tue, 29 Jun 2010 06:52:13 +0000
Subject: [pdf] Fix build when text_layout is not available

Fixes bug #623080.
---
diff --git a/backend/pdf/ev-poppler.cc b/backend/pdf/ev-poppler.cc
index 91f5c2e..6e56423 100644
--- a/backend/pdf/ev-poppler.cc
+++ b/backend/pdf/ev-poppler.cc
@@ -2029,6 +2029,7 @@ pdf_document_text_get_text (EvDocumentText  *selection,
 				      &r);
 }
 
+#ifdef HAVE_POPPLER_PAGE_GET_TEXT_LAYOUT
 static gboolean
 pdf_document_text_get_text_layout (EvDocumentText  *selection,
 				   EvPage          *page,
@@ -2043,6 +2044,7 @@ pdf_document_text_get_text_layout (EvDocumentText  *selection,
 
 	return poppler_page_get_text_layout (poppler_page, (PopplerRectangle **)areas, n_areas);
 }
+#endif
 
 static void
 pdf_document_text_iface_init (EvDocumentTextInterface *iface)
--
cgit v0.8.3.1
