--- configure.ac.orig	2008-03-11 19:09:44.000000000 -0700
+++ configure.ac	2008-03-11 19:10:37.000000000 -0700
@@ -29,7 +29,11 @@
 GNOME_CXX_WARNINGS
 GNOME_DEBUG_CHECK
 
+dnl Crazy hack for dropping dep on gtk-doc
+AC_DEFUN(GTK_DOC_CHECK, [])
 GTK_DOC_CHECK(1.0)
+AM_CONDITIONAL(ENABLE_GTK_DOC, 0, [])
+AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, 0, [])
 
 GETTEXT_PACKAGE=evince
 AC_SUBST(GETTEXT_PACKAGE)
--- help/Makefile.old.am	2008-03-15 12:54:08.000000000 +0100
+++ help/Makefile.am	2008-03-15 12:54:40.000000000 +0100
@@ -1,4 +1,9 @@
-SUBDIRS = reference
+
+SUBDIRS =
+
+if ENABLE_GTK_DOC
+SUBDIRS += reference
+endif
 
 include $(top_srcdir)/gnome-doc-utils.make
 dist-hook: doc-dist-hook
