--- gtk-doc.make	2008-01-27 00:04:08.000000000 -0800
+++ gtk-doc.make	2008-01-27 00:05:20.000000000 -0800
@@ -139,8 +139,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 	
 
--- docs/reference/Makefile.in	2008-01-27 00:05:28.000000000 -0800
+++ docs/reference/Makefile.in	2008-01-27 00:05:56.000000000 -0800
@@ -636,8 +636,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 
 uninstall-local:
