diff --git a/plugins/filebrowser/meson.build b/plugins/filebrowser/meson.build
index 708f7f1d49e01d78f40666e060f1c0fd0b12ee5f..94a4ab0a66126ebf14a75e752e6bb57e86d6c812 100644
--- a/plugins/filebrowser/meson.build
+++ b/plugins/filebrowser/meson.build
@@ -59,6 +59,7 @@ libfilebrowser_enums_c = custom_target('libfilebrowser_enums_c',
 
 libfilebrowser_sources += [
   libfilebrowser_enums_c,
+  libfilebrowser_type_enums.get(1),
 ]
 
 subdir('resources')
