diff -NurdB gedit-2.16.2/docs/reference/gedit.types gedit-2.16.2-patched/docs/reference/gedit.types
--- gedit-2.16.2/docs/reference/gedit.types     2006-10-29 18:32:53.000000000 +0000
+++ gedit-2.16.2-patched/docs/reference/gedit.types     2006-12-21 10:04:31.000000000 +0000
@@ -10,7 +10,6 @@
 #include "gedit-print.h"
 #include "gedit-print-job-preview.h"
 #include "gedit-progress-message-area.h"
-#include "gedit-python-module.h"
 #include "gedit-statusbar.h"
 #include "gedit-tab.h"
 #include "gedit-view.h"
@@ -27,7 +26,6 @@
 gedit_print_job_get_type
 gedit_print_job_preview_get_type
 gedit_progress_message_area_get_type
-gedit_python_module_get_type
 gedit_statusbar_get_type
 gedit_tab_get_type
 gedit_tooltips_get_type
