From 485704419b11685e4753024dabec53d1668aaa9d Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 24 Aug 2012 20:47:00 +0000
Subject: Adjust for webkit 1.9.90 API change.

---
diff --git a/widgets/misc/e-web-view.c b/widgets/misc/e-web-view.c
index 802e402..6612876 100644
--- a/widgets/misc/e-web-view.c
+++ b/widgets/misc/e-web-view.c
@@ -390,8 +390,13 @@ replace_text (WebKitDOMNode *node,
 			}
 
 			split_offset = offset - node_data + text_length;
+			#if WEBKIT_CHECK_VERSION (1, 9, 90)
+			replacement_node =
+				webkit_dom_node_clone_node (replacement, TRUE, NULL);
+			#else
 			replacement_node =
 				webkit_dom_node_clone_node (replacement, TRUE);
+			#endif
 
 			data_length = webkit_dom_character_data_get_length (
 					(WebKitDOMCharacterData *) node);
--
cgit v0.9.0.2
