From bd7a0270f0f315197c20bb54397f0ae2bac0715d Mon Sep 17 00:00:00 2001
From: Vibha Yadav <yvibha@novell.com>
Date: Mon, 31 Jan 2011 12:58:50 +0000
Subject: Bug 641003 - Build failure in mail-capplet

---
diff --git a/capplet/settings/mail-capplet-shell.c b/capplet/settings/mail-capplet-shell.c
index 7a9a4fa..c14af1b 100644
--- a/capplet/settings/mail-capplet-shell.c
+++ b/capplet/settings/mail-capplet-shell.c
@@ -26,6 +26,8 @@
 #ifdef HAVE_CONFIG_H
 #  include <config.h>
 #endif
+
+#include <gtk/gtkx.h>
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
 #include <gconf/gconf-client.h>
--
cgit v0.8.3.1
