diff -burN evolution-3.10.1/calendar/alarm-notify/Makefile.am evolution-3.10.1.new/calendar/alarm-notify/Makefile.am
--- evolution-3.10.1/calendar/alarm-notify/Makefile.am	2013-09-21 08:53:21.000000000 -0500
+++ evolution-3.10.1.new/calendar/alarm-notify/Makefile.am	2013-10-24 23:39:38.907870822 -0500
@@ -44,6 +44,7 @@
 evolution_alarm_notify_LDADD =						\
 	$(top_builddir)/e-util/libevolution-util.la			\
 	$(top_builddir)/composer/libevolution-mail-composer.la		\
+	$(top_builddir)/em-format/libevolution-mail-formatter.la	\
 	$(top_builddir)/calendar/gui/libevolution-calendar.la		\
 	$(top_builddir)/calendar/importers/libevolution-calendar-importers.la \
 	$(top_builddir)/addressbook/gui/contact-editor/libecontacteditor.la \
