diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-2.23.3.1.orig/mail/em-folder-view.c evolution-2.23.3.1/mail/em-folder-view.c
--- evolution-2.23.3.1.orig/mail/em-folder-view.c	2008-06-02 03:51:43.000000000 -0400
+++ evolution-2.23.3.1/mail/em-folder-view.c	2008-06-03 11:16:19.000000000 -0400
@@ -175,7 +175,6 @@ static const EMFolderViewEnable emfv_ena
 	{ "MessageFilterJunk",        EM_POPUP_SELECT_MANY },
 	{ "MessageCopy",              EM_POPUP_SELECT_MANY },
 	{ "MessageDelete",            EM_POPUP_SELECT_MANY|EM_POPUP_SELECT_DELETE },
-	{ "MessageDeleteKey",         EM_POPUP_SELECT_MANY},	
 	{ "MessageForward",           EM_POPUP_SELECT_MANY },
 	{ "MessageForwardAttached",   EM_POPUP_SELECT_MANY },
 	{ "MessageForwardInline",     EM_POPUP_SELECT_ONE },
@@ -2063,7 +2062,6 @@ static BonoboUIVerb emfv_message_verbs[]
 	BONOBO_UI_UNSAFE_VERB ("MessageFilterJunk", emfv_message_filter_junk),
 	BONOBO_UI_UNSAFE_VERB ("MessageCopy", emfv_message_copy),
 	BONOBO_UI_UNSAFE_VERB ("MessageDelete", emfv_message_delete),
-	BONOBO_UI_UNSAFE_VERB ("MessageDeleteKey", emfv_message_delete),
 	BONOBO_UI_UNSAFE_VERB ("MessageForward", emfv_message_forward),
 	BONOBO_UI_UNSAFE_VERB ("MessageForwardAttached", emfv_message_forward_attached),
 	BONOBO_UI_UNSAFE_VERB ("MessageForwardInline", emfv_message_forward_inline),
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN evolution-2.23.3.1.orig/ui/evolution-mail-message.xml evolution-2.23.3.1/ui/evolution-mail-message.xml
--- evolution-2.23.3.1.orig/ui/evolution-mail-message.xml	2008-06-02 03:51:43.000000000 -0400
+++ evolution-2.23.3.1/ui/evolution-mail-message.xml	2008-06-03 11:16:19.000000000 -0400
@@ -58,6 +58,7 @@
 
     <cmd name="MessageDelete"
      _tip="Mark the selected messages for deletion"
+     accel="*Control*d"
      pixtype="pixbuf"/>
 
     <cmd name="MessageFollowUpFlag"
@@ -243,9 +244,8 @@
     <accel verb="MailNextUnread"     name="*Control*period"/>
     <accel verb="MailNextUnread"     name="period"/>
     <accel verb="MailNextUnread"     name="bracketright"/>
-    <accel verb="MessageDeleteKey"   name="Delete"/>
-    <accel verb="MessageDeleteKey"   name="KP_Delete"/>
-    <accel verb="MessageDeleteKey"   name="*Control*d"/>
+    <accel verb="MessageDelete"      name="Delete"/>
+    <accel verb="MessageDelete"      name="KP_Delete"/>
   </keybindings>
 
   <menu>
