 configure.in |   63 ++++++++++++++++++++++++++++++++++++---------------------
 1 files changed, 40 insertions(+), 23 deletions(-)

diff --git a/configure.in b/configure.in
index 22de1c4..1b0cfa3 100644
--- a/configure.in
+++ b/configure.in
@@ -1773,9 +1773,9 @@ dnl Add any new plugins here
 plugins_base_always="calendar-file calendar-http $CALENDAR_WEATHER itip-formatter plugin-manager default-source addressbook-file startup-wizard mark-all-read groupwise-features groupwise-account-setup mail-account-disable publish-calendar caldav imap-features google-account-setup webdav-account-setup"
 
 plugins_base="$plugins_base_always $SA_JUNK_PLUGIN $BF_JUNK_PLUGIN $EXCHANGE_PLUGIN $MONO_PLUGIN " 
-all_plugins_base="$plugins_base_always sa-junk-plugin bogo-junk-plugin exchange-operations mono"
+all_plugins_base="$plugins_base_always sa-junk-plugin bogo-junk-plugin mono"
 
-plugins_standard_always="bbdb subject-thread save-calendar select-one-source copy-tool mail-to-task audio-inline mailing-list-actions default-mailer import-ics-attachments prefer-plain mail-notification attachment-reminder face backup-restore email-custom-header templates pst-import" 
+plugins_standard_always="bbdb subject-thread save-calendar select-one-source copy-tool mail-to-task mailing-list-actions default-mailer import-ics-attachments prefer-plain mail-notification attachment-reminder face backup-restore email-custom-header templates" 
 
 plugins_standard="$plugins_standard_always"
 all_plugins_standard="$plugins_standard"
@@ -1835,8 +1835,13 @@ AC_SUBST(all_plugins_base)
 AC_SUBST(all_plugins_standard)
 AC_SUBST(all_plugins_experimental)
 
-if echo ${plugins_enabled} | grep "audio-inline" > /dev/null
+AC_ARG_ENABLE([audio-inline],
+	AC_HELP_STRING([--disable-audio-inline], [Disable audio-inline plugin @<:@default=yes@:>@]),
+	[enable_audio_inline="$enableval"], [enable_audio_inline=yes])
+
+if test "x$enable_audio_inline" = "xyes"
 then
+	plugins_standard_always="$plugins_standard_always audio-inline"
 	if ${PKG_CONFIG} --exists gstreamer-0.10
 	then
 		dnl *********************
@@ -1852,31 +1857,37 @@ then
 	fi
 fi
 
-if ${PKG_CONFIG} --exists dbus-glib-1 ; then
+AC_ARG_ENABLE([dbus],
+	AC_HELP_STRING([--disable-dbus], [Disable mail notification DBus messages plugin @<:@default=yes@:>@]),
+	[enable_dbus="$enableval"], [enable_dbus=yes])
+
+if test "x$enable_dbus" = "xyes"; then
 	dnl **************************************************
 	dnl * Mail Notification plugin's DBus messages
 	dnl **************************************************
-	PKG_CHECK_MODULES(NMN, dbus-glib-1)
-	AC_SUBST(NMN_CFLAGS)
-	AC_SUBST(NMN_LIBS)
-
-	# Get the version of the DBus API, so we can hack around API changes until the API stabilises:
-	# multiply by 1000 to convert decimal to integer; so e.g. 0.31 become 310
-	# since preprocessor values must be integral
-	FOUND_DBUS_VERSION="`$PKG_CONFIG --modversion dbus-1 | $AWK '{print 1000 * $1}'`"
-	AC_SUBST(FOUND_DBUS_VERSION)
-	AC_DEFINE(HAVE_DBUS,1,[Define if you have DBUS support])
-	AC_SUBST(HAVE_DBUS)
-	AM_CONDITIONAL(ENABLE_DBUS, true)
-	
-else
-	echo "warning: dbus-glib-1 was not found, Mail notification plugins's dbus message support not built"
-	AM_CONDITIONAL(ENABLE_DBUS, false)
+	if $PKG_CONFIG --exists dbus-glib-1; then
+	   PKG_CHECK_MODULES(NMN, dbus-glib-1)
+	   AC_SUBST(NMN_CFLAGS)
+	   AC_SUBST(NMN_LIBS)
+
+	   # Get the version of the DBus API, so we can hack around API changes until the API stabilises:
+	   # multiply by 1000 to convert decimal to integer; so e.g. 0.31 become 310
+	   # since preprocessor values must be integral
+	   FOUND_DBUS_VERSION="`$PKG_CONFIG --modversion dbus-1 | $AWK '{print 1000 * $1}'`"
+	   AC_SUBST(FOUND_DBUS_VERSION)
+	   AC_DEFINE(HAVE_DBUS,1,[Define if you have DBUS support])
+	   AC_SUBST(HAVE_DBUS)
+	else
+		echo "warning: dbus-glib-1 was not found, Mail notification plugins's dbus message support not built"
+	fi
 
 fi
 
-if echo ${plugins_enabled} | grep "exchange-operations" > /dev/null ; then
-	PKG_CHECK_MODULES(LIBEXCHANGESTORAGE, libexchange-storage-$EDS_PACKAGE >= eds_minimum_version, have_libexchange="yes", have_libexchange="no")
+AM_CONDITIONAL(ENABLE_DBUS, test x$enable_dbus = xyes)
+
+if test "x$enable_exchange" = "xyes"; then
+	PKG_CHECK_MODULES(LIBEXCHANGESTORAGE, libexchange-storage-$EDS_PACKAGE >= eds_minimum_version, 
+	have_libexchange="yes", have_libexchange="no")
 	if test "x$have_libexchange" = "xyes"; then
 		dnl **************************************************
 		dnl * Exchange Operations plugin
@@ -1890,8 +1901,14 @@ if echo ${plugins_enabled} | grep "exchange-operations" > /dev/null ; then
 	fi
 fi
 
-if echo ${plugins_enabled} | grep "pst-import" > /dev/null
+AC_ARG_ENABLE([pst-import],
+	AC_HELP_STRING([--disable-pst-import], 
+	[Disable pst import plugin @<:@default=yes@:>@]),
+	[enable_pst="$enableval"], [enable_pst=yes])
+
+if test "x$enable_pst" = "xyes"
 then
+	plugins_standard_always="$plugins_standard_always pst-import"
 	if ${PKG_CONFIG} --exists libpst
 	then
 		dnl *********************
