diff -burN evolution-3.10.0/addressbook/gui/widgets/eab-contact-formatter.c evolution-3.10.0.new/addressbook/gui/widgets/eab-contact-formatter.c
--- evolution-3.10.0/addressbook/gui/widgets/eab-contact-formatter.c	2013-09-14 11:46:59.000000000 -0500
+++ evolution-3.10.0.new/addressbook/gui/widgets/eab-contact-formatter.c	2013-10-06 13:45:58.898422225 -0500
@@ -1158,7 +1158,7 @@
 	gboolean hidden;
 
 	document = user_data;
-	id = webkit_dom_html_element_get_id (WEBKIT_DOM_HTML_ELEMENT (event_target));
+	id = webkit_dom_element_get_id (WEBKIT_DOM_ELEMENT (event_target));
 
 	list_id = g_strconcat ("list-", id, NULL);
 	list = webkit_dom_document_get_element_by_id (document, list_id);
diff -burN evolution-3.10.0/e-util/e-web-view.c evolution-3.10.0.new/e-util/e-web-view.c
--- evolution-3.10.0/e-util/e-web-view.c	2013-09-15 07:06:45.000000000 -0500
+++ evolution-3.10.0.new/e-util/e-web-view.c	2013-10-06 13:47:14.962421691 -0500
@@ -3305,8 +3305,8 @@
 	if (!style_element) {
 		/* Create new <style> element */
 		style_element = webkit_dom_document_create_element (document, "style", NULL);
-		webkit_dom_html_element_set_id (
-			WEBKIT_DOM_HTML_ELEMENT (style_element),
+		webkit_dom_element_set_id (
+			WEBKIT_DOM_ELEMENT (style_element),
 			style_sheet_id);
 		webkit_dom_html_style_element_set_media (
 			WEBKIT_DOM_HTML_STYLE_ELEMENT (style_element),
diff -burN evolution-3.10.0/modules/itip-formatter/itip-view.c evolution-3.10.0.new/modules/itip-formatter/itip-view.c
--- evolution-3.10.0/modules/itip-formatter/itip-view.c	2013-09-09 07:34:55.000000000 -0500
+++ evolution-3.10.0.new/modules/itip-formatter/itip-view.c	2013-10-06 13:48:03.604421349 -0500
@@ -796,7 +796,7 @@
                         ItipView *view)
 {
 	WebKitDOMElement *check2;
-	gchar *id = webkit_dom_html_element_get_id (WEBKIT_DOM_HTML_ELEMENT (check1));
+	gchar *id = webkit_dom_element_get_id (WEBKIT_DOM_ELEMENT (check1));
 
 	if (g_strcmp0 (id, CHECKBOX_INHERIT_ALARM)) {
 		check2 = webkit_dom_document_get_element_by_id (
@@ -905,7 +905,7 @@
 		WEBKIT_DOM_HTML_TABLE_ELEMENT (table), -1, NULL);
 
 	id = g_strdup_printf ("%s_row_%d", table_id, item->id);
-	webkit_dom_html_element_set_id (row, id);
+	webkit_dom_element_set_id (WEBKIT_DOM_ELEMENT (row), id);
 	g_free (id);
 
 	switch (item->type) {
