diff -ru evolution-2.22.0.orig/e-util/e-profile-event.c evolution-2.22.0/e-util/e-profile-event.c
--- evolution-2.22.0.orig/e-util/e-profile-event.c	2008-03-15 12:16:20.362962216 -0400
+++ evolution-2.22.0/e-util/e-profile-event.c	2008-03-15 12:20:27.281962492 -0400
@@ -125,7 +125,7 @@
 }
 
 #ifdef ENABLE_PROFILING
-static void
+void
 e_profile_event_emit(const char *id, const char *uid, guint32 flags)
 {
 	EProfileEvent *epe = e_profile_event_peek();
