From ea52ab83fe121796a12a090ce81ba114d2c368ed Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Mon, 15 Mar 2010 14:38:41 +0000
Subject: Bug 612792 - Crash when trying to delete WebDav addressbook

---
diff --git a/modules/addressbook/e-book-shell-view-actions.c b/modules/addressbook/e-book-shell-view-actions.c
index dbfdc99..0e4b58f 100644
--- a/modules/addressbook/e-book-shell-view-actions.c
+++ b/modules/addressbook/e-book-shell-view-actions.c
@@ -71,7 +71,7 @@ action_address_book_delete_cb (GtkAction *action,
 	response = e_alert_run_dialog_for_args (
 		GTK_WINDOW (shell_window),
 		"addressbook:ask-delete-addressbook",
-		e_source_peek_name (source));
+		e_source_peek_name (source), NULL);
 
 	if (response != GTK_RESPONSE_YES)
 		return;
diff --git a/modules/calendar/e-cal-shell-view-actions.c b/modules/calendar/e-cal-shell-view-actions.c
index 552e0f9..5eaa9a4 100644
--- a/modules/calendar/e-cal-shell-view-actions.c
+++ b/modules/calendar/e-cal-shell-view-actions.c
@@ -91,7 +91,7 @@ action_calendar_delete_cb (GtkAction *action,
 	response = e_alert_run_dialog_for_args (
 		GTK_WINDOW (shell_window),
 		"calendar:prompt-delete-calendar",
-		e_source_peek_name (source));
+		e_source_peek_name (source), NULL);
 	if (response != GTK_RESPONSE_YES)
 		return;
 
diff --git a/modules/calendar/e-memo-shell-view-actions.c b/modules/calendar/e-memo-shell-view-actions.c
index 515118f..a9b7fea 100644
--- a/modules/calendar/e-memo-shell-view-actions.c
+++ b/modules/calendar/e-memo-shell-view-actions.c
@@ -159,7 +159,7 @@ action_memo_list_delete_cb (GtkAction *action,
 	response = e_alert_run_dialog_for_args (
 		GTK_WINDOW (shell_window),
 		"calendar:prompt-delete-memo-list",
-		e_source_peek_name (source));
+		e_source_peek_name (source), NULL);
 	if (response != GTK_RESPONSE_YES)
 		return;
 
diff --git a/modules/calendar/e-task-shell-view-actions.c b/modules/calendar/e-task-shell-view-actions.c
index 6452e57..0fd218c 100644
--- a/modules/calendar/e-task-shell-view-actions.c
+++ b/modules/calendar/e-task-shell-view-actions.c
@@ -182,7 +182,7 @@ action_task_list_delete_cb (GtkAction *action,
 	response = e_alert_run_dialog_for_args (
 		GTK_WINDOW (shell_window),
 		"calendar:prompt-delete-task-list",
-		e_source_peek_name (source));
+		e_source_peek_name (source), NULL);
 	if (response != GTK_RESPONSE_YES)
 		return;
 
--
cgit v0.8.3.1
