From 51a648e46080e6a23007370e859c717719482ddf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <michel@daenzer.net>
Date: Wed, 24 Mar 2010 16:51:31 +0100
Subject: Fix NULL pointer dereference in em_format_snoop_type().
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michel Dänzer <michel@daenzer.net>
---
 em-format/em-format.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/em-format/em-format.c b/em-format/em-format.c
index e5856a0..45d9cf2 100644
--- a/em-format/em-format.c
+++ b/em-format/em-format.c
@@ -2015,10 +2015,11 @@ em_format_snoop_type (CamelMimePart *part)
 
 	dw = camel_medium_get_content_object((CamelMedium *)part);
 	if (!camel_data_wrapper_is_offline(dw)) {
-		CamelStreamMem *mem = (CamelStreamMem *)camel_stream_mem_new();
+		GByteArray *buffer = g_byte_array_new ();
+		CamelStreamMem *mem = (CamelStreamMem *)camel_stream_mem_new_with_byte_array(buffer);
 
 		if (camel_data_wrapper_decode_to_stream(dw, (CamelStream *)mem) > 0) {
-			gchar *ct = g_content_type_guess (filename, mem->buffer->data, mem->buffer->len, NULL);
+			gchar *ct = g_content_type_guess (filename, buffer->data, buffer->len, NULL);
 
 			if (ct)
 				magic_type = g_content_type_get_mime_type (ct);
-- 
1.7.0.3

