diff -Naur hamster-applet-2.24.0.orig/configure.ac hamster-applet-2.24.0/configure.ac
--- hamster-applet-2.24.0.orig/configure.ac	2008-09-22 22:20:03.000000000 +0530
+++ hamster-applet-2.24.0/configure.ac	2008-09-26 15:46:37.000000000 +0530
@@ -100,20 +100,38 @@
 AC_MSG_RESULT($PYGTK_DEFSDIR)
 
 
-AC_MSG_CHECKING([for pysqlite2 module])
-if AC_RUN_LOG([DISPLAY= $PYTHON -c '
+if test "x$PYTHON_VERSION" = "x2.5"; then
+    AC_MSG_CHECKING([for python sqlite module])
+    if AC_RUN_LOG([DISPLAY= $PYTHON -c '
+try:
+    import sqlite3
+except ImportError, e:
+    if str(e).find("sqlite3") >= 0:
+        raise
+except:
+    pass
+    ']); then
+      AC_MSG_RESULT([yes])
+    else
+      AC_MSG_RESULT([no])
+      AC_MSG_ERROR([Python 2.5: inbuilt sqlite is required to build hamster])
+    fi
+else
+    AC_MSG_CHECKING([for pysqlite2 module])
+    if AC_RUN_LOG([DISPLAY= $PYTHON -c '
 try:
     import pysqlite2
 except ImportError, e:
     if str(e).find("pysqlite2") >= 0:
-          raise
+        raise
 except:
     pass
-']); then
-  AC_MSG_RESULT([yes])
-else
-  AC_MSG_RESULT([no])
-  AC_MSG_ERROR([pysqlite2 Python module required to build hamster])
+    ']); then
+      AC_MSG_RESULT([yes])
+    else
+      AC_MSG_RESULT([no])
+      AC_MSG_ERROR([Python 2.4: pysqlite2 module required to build hamster])
+    fi
 fi
 
 
diff -Naur hamster-applet-2.24.0.orig/hamster/db.py hamster-applet-2.24.0/hamster/db.py
--- hamster-applet-2.24.0.orig/hamster/db.py	2008-09-22 22:20:02.000000000 +0530
+++ hamster-applet-2.24.0/hamster/db.py	2008-09-26 16:05:58.000000000 +0530
@@ -21,7 +21,14 @@
 
 """separate file for database operations"""
 
-from pysqlite2 import dbapi2 as sqlite
+try:
+    import sqlite3 as sqlite
+except ImportError:
+    try:
+        from pysqlite2 import dbapi2 as sqlite
+    except ImportError:
+        print "Error: Neither sqlite3 nor pysqlite2 found"
+        raise
 import os, time
 import datetime
 import hamster
