Assumes $(PYGOBJECT_FIXXREF) is executable

---
--- docs/Makefile.am
+++ docs/Makefile.am
@@ -82,7 +82,7 @@
 	fi
 
 install-data-hook:
-	@$(PYGOBJECT_FIXXREF) -i $(PYGOBJECT_PYGDOCS) $(DESTDIR)$(TARGET_DIR)
+	python $(PYGOBJECT_FIXXREF) -i $(PYGOBJECT_PYGDOCS) $(DESTDIR)$(TARGET_DIR)
 
 uninstall-local:
 	rm -rf $(DESTDIR)$(TARGET_DIR)/*
--- docs/Makefile.in
+++ docs/Makefile.in
@@ -82,7 +82,7 @@
 	fi
 
 install-data-hook:
-	@$(PYGOBJECT_FIXXREF) -i $(PYGOBJECT_PYGDOCS) $(DESTDIR)$(TARGET_DIR)
+	python $(PYGOBJECT_FIXXREF) -i $(PYGOBJECT_PYGDOCS) $(DESTDIR)$(TARGET_DIR)
 
 uninstall-local:
 	rm -rf $(DESTDIR)$(TARGET_DIR)/*
