From 814e578545eaa089a107679e669dd7cb80d56782 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Thu, 7 May 2009 22:44:41 +0200
Subject: [PATCH] Fix automagic libpanelappletmm dependency.

---
 configure.ac |   54 ++++++++++++++++++++++++++++++------------------------
 1 files changed, 30 insertions(+), 24 deletions(-)

diff --git a/configure.ac b/configure.ac
index a743d8d..c20a03a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -86,30 +86,36 @@ dnl check for the ext2 uuid library. TODO replace it with something more portabl
 AC_CHECK_HEADER(uuid/uuid.h,,[AC_MSG_ERROR("uuid header is missing")])
 AC_CHECK_LIB(uuid,uuid_unparse_lower, ,[AC_MSG_ERROR("libuuid library is missing")])
 
-PKG_CHECK_EXISTS(libpanelappletmm-2.6 >= $LIBPANELAPPLETMM_VERSION,
-    [
-     AC_DEFINE(HAVE_PANELAPPLETMM, 1, [Define to 1 if panelapplet is available])
-     have_panelapplet="yes"
-     PKG_CHECK_MODULES(LIBPANELAPPLETMM, [libpanelappletmm-2.6 >= $LIBPANELAPPLETMM_VERSION])
-    ],
-    [have_panelapplet="no"]
-)
-
-AM_CONDITIONAL(HAVE_PANELAPPLETMM, test $have_panelapplet == "yes")
-
-#
-# If libpanelapplet available, use it to find .server install location.
-#
-if test "x$prefix" = "x$ac_default_prefix"; then
-   PANELAPPLET_LIBDIR=`$PKG_CONFIG --variable=libdir libpanelapplet-2.0`
-fi
-if test "x$PANELAPPLET_LIBDIR" != "x"; then
-   PANELAPPLET_SERVER_DIR="$PANELAPPLET_LIBDIR/bonobo/servers"
-fi
-# Set PANELAPPLET_SERVER_DIR default if it's not already set
-if test "x$PANELAPPLET_SERVER_DIR" = "x" ; then
-        PANELAPPLET_SERVER_DIR="$libdir/bonobo/servers"
+AC_ARG_ENABLE([applet],
+	AS_HELP_STRING([--disable-applet],[Disable panel applet]),
+	[],[enable_applet=yes])
+
+if test $enable_applet = "yes"; then
+	PKG_CHECK_EXISTS(libpanelappletmm-2.6 >= $LIBPANELAPPLETMM_VERSION,
+	  [
+		AC_DEFINE(HAVE_PANELAPPLETMM, 1, [Define to 1 if panelapplet is available])
+		have_panelapplet="yes"
+		PKG_CHECK_MODULES(LIBPANELAPPLETMM, [libpanelappletmm-2.6 >= $LIBPANELAPPLETMM_VERSION])
+	  ], [
+		have_panelapplet="no"]
+	)
+
+	#
+	# If libpanelapplet available, use it to find .server install location.
+	#
+	if test "x$prefix" = "x$ac_default_prefix"; then
+		PANELAPPLET_LIBDIR=`$PKG_CONFIG --variable=libdir libpanelapplet-2.0`
+	fi
+	if test "x$PANELAPPLET_LIBDIR" != "x"; then
+		PANELAPPLET_SERVER_DIR="$PANELAPPLET_LIBDIR/bonobo/servers"
+	fi
+	# Set PANELAPPLET_SERVER_DIR default if it's not already set
+	if test "x$PANELAPPLET_SERVER_DIR" = "x" ; then
+		PANELAPPLET_SERVER_DIR="$libdir/bonobo/servers"
+	fi
 fi
+
+AM_CONDITIONAL(HAVE_PANELAPPLETMM, test $have_panelapplet = "yes")
 AC_SUBST(PANELAPPLET_SERVER_DIR)
 
 #
@@ -227,4 +233,4 @@ Options:
   Spellchecking support:       ${FIXED_GTKSPELL}
   GNOME Panel Applet support:  ${have_panelapplet}
 
-"
\ No newline at end of file
+"
-- 
1.6.2.3

