From 3a0683d6b4804f9cd4924713dcab8b80f5864354 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dominique-gnomezilla@leuenberger.net>
Date: Thu, 16 Jun 2011 12:55:00 +0000
Subject: Fix linking with -Wl,--as-needed

Resolves https://bugzilla.gnome.org/show_bug.cgi?id=652722

Signed-off-by: Richard Hughes <richard@hughsie.com>
---
diff --git a/tools/Makefile.am b/tools/Makefile.am
index 1edc90c..9621137 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -28,10 +28,10 @@ gcm_dump_profile_SOURCES =				\
 	gcm-dump-profile.c
 
 gcm_dump_profile_LDADD =				\
+	$(top_builddir)/src/libgcmshared.a		\
 	$(GLIB_LIBS)					\
 	$(COLORD_LIBS)					\
 	$(LCMS_LIBS)					\
-	$(top_builddir)/src/libgcmshared.a		\
 	-lm
 
 gcm_dump_profile_CFLAGS =				\
--
cgit v0.9
