diff --git a/src/cc-color-panel.c b/src/cc-color-panel.c
index 34ce173..f4307cc 100644
--- a/src/cc-color-panel.c
+++ b/src/cc-color-panel.c
@@ -2150,7 +2150,7 @@ cc_color_panel_is_color_profiles_extra_installed (CcColorPanel *panel)
 	GError *error = NULL;
 
 #ifndef HAVE_PACKAGEKIT
-	g_warning ("cannot query %s: this package was not compiled with --enable-packagekit", package_name);
+	g_warning ("cannot query %s: this package was not compiled with --enable-packagekit", GCM_PREFS_PACKAGE_NAME_COLOR_PROFILES_EXTRA);
 	return;
 #endif
 
