diff -Naur gdm-2.26.0/common/gdm-settings-keys.h gdm-2.26.0.new/common/gdm-settings-keys.h
--- gdm-2.26.0/common/gdm-settings-keys.h	2009-02-20 08:15:13.000000000 +0530
+++ gdm-2.26.0.new/common/gdm-settings-keys.h	2009-03-23 10:00:08.156374929 +0530
@@ -27,6 +27,7 @@
 
 #define GDM_KEY_USER "daemon/User"
 #define GDM_KEY_GROUP "daemon/Group"
+#define GDM_KEY_VT "daemon/VT"
 #define GDM_KEY_AUTO_LOGIN_ENABLE "daemon/AutomaticLoginEnable"
 #define GDM_KEY_AUTO_LOGIN_USER "daemon/AutomaticLogin"
 #define GDM_KEY_TIMED_LOGIN_ENABLE "daemon/TimedLoginEnable"
diff -Naur gdm-2.26.0/daemon/gdm-server.c gdm-2.26.0.new/daemon/gdm-server.c
--- gdm-2.26.0/daemon/gdm-server.c	2009-02-20 08:15:13.000000000 +0530
+++ gdm-2.26.0.new/daemon/gdm-server.c	2009-03-23 09:59:27.173374855 +0530
@@ -44,6 +44,10 @@
 #include "gdm-common.h"
 #include "gdm-signal-handler.h"
 
+#include "gdm-settings.h"
+#include "gdm-settings-direct.h"
+#include "gdm-settings-keys.h"
+
 #include "gdm-server.h"
 
 extern char **environ;
@@ -79,6 +83,7 @@
         char    *parent_display_name;
         char    *parent_auth_file;
         char    *chosen_hostname;
+        char    *vt;
 
         guint    child_watch_id;
 };
@@ -676,7 +681,7 @@
         gboolean res;
 
         /* fork X server process */
-        res = gdm_server_spawn (server, NULL);
+        res = gdm_server_spawn (server, server->priv->vt);
 
         return res;
 }
@@ -928,12 +933,33 @@
 gdm_server_init (GdmServer *server)
 {
 
+        int vt;
+        gboolean has_vt;
+        GdmSettings *settings = NULL;
         server->priv = GDM_SERVER_GET_PRIVATE (server);
 
+
+        settings = gdm_settings_new();
+        if (settings == NULL) {
+                g_assert ("Unable to initialize settings");
+        }
+
+        if (! gdm_settings_direct_init (settings, GDMCONFDIR "/gdm.schemas", "/")) {
+                g_assert ("Unable to initialize settings");
+        }
+
+        has_vt =  gdm_settings_direct_get_int (GDM_KEY_VT, &vt);
+
+        g_object_unref (settings);
+
         server->priv->pid = -1;
         server->priv->command = g_strdup (X_SERVER " -br -verbose");
         server->priv->log_dir = g_strdup (LOGDIR);
 
+        if (has_vt) {
+                server->priv->vt = g_strdup_printf ("vt %d", vt);
+        }
+
         add_ready_handler (server);
 }
 
diff -Naur gdm-2.26.0/data/gdm.schemas.in.in gdm-2.26.0.new/data/gdm.schemas.in.in
--- gdm-2.26.0/data/gdm.schemas.in.in	2009-02-20 08:15:14.000000000 +0530
+++ gdm-2.26.0.new/data/gdm.schemas.in.in	2009-03-23 09:59:41.893374593 +0530
@@ -11,6 +11,11 @@
       <signature>s</signature>
       <default>ff02::1</default>
     </schema>
+    <schema>
+      <key>daemon/VT</key>
+      <signature>i</signature>
+      <default>7</default>
+    </schema>
 
     <schema>
       <key>daemon/User</key>
