From b669aa14d85a0d011fd25093f8e4f4da531ee679 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 26 Jul 2011 01:10:36 +0000
Subject: gdm-slave: fix memory leak

https://bugzilla.gnome.org/show_bug.cgi?id=655380
---
diff --git a/daemon/gdm-slave.c b/daemon/gdm-slave.c
index 2dc0323..fad7c96 100644
--- a/daemon/gdm-slave.c
+++ b/daemon/gdm-slave.c
@@ -1432,6 +1432,7 @@ session_unlock (GdmSlave   *slave,
 
         if (dbus_error_is_set (&error)) {
                 g_debug ("GdmSlave: ConsoleKit %s raised:\n %s\n\n", error.name, error.message);
+                dbus_error_free (&error);
                 return FALSE;
         }
 
--
cgit v0.9.0.2
