From b8f1c67f7db3f86f631420b057ae08fc5f053819 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 26 Jul 2011 01:11:20 +0000
Subject: gdm-slave: fix double dbus_error_init()

https://bugzilla.gnome.org/show_bug.cgi?id=655380
---
diff --git a/daemon/gdm-slave.c b/daemon/gdm-slave.c
index fad7c96..e7553ad 100644
--- a/daemon/gdm-slave.c
+++ b/daemon/gdm-slave.c
@@ -1376,8 +1376,6 @@ activate_session_id (GdmSlave   *slave,
                 goto out;
         }
 
-
-        dbus_error_init (&local_error);
         reply = dbus_connection_send_with_reply_and_block (dbus_g_connection_get_connection (slave->priv->connection),
                                                            message,
                                                            -1,
--
cgit v0.9.0.2
