--- docs/reference/Makefile.am
+++ docs/reference/Makefile.am
@@ -9,7 +9,7 @@
 # of using the various options.
 
 # The name of the module, e.g. 'glib'.
-DOC_MODULE=gdl
+DOC_MODULE=gdl-3
 
 # The top-level SGML file. You can change this if you want to.
 DOC_MAIN_SGML_FILE=$(DOC_MODULE)-docs.sgml
--- docs/reference/Makefile.in
+++ docs/reference/Makefile.in
@@ -256,7 +256,7 @@
 # of using the various options.
 
 # The name of the module, e.g. 'glib'.
-DOC_MODULE = gdl
+DOC_MODULE = gdl-3
 
 # The top-level SGML file. You can change this if you want to.
 DOC_MAIN_SGML_FILE = $(DOC_MODULE)-docs.sgml
--- gdl/Makefile.am
+++ gdl/Makefile.am
@@ -80,12 +80,12 @@
 	libgdltypebuiltins.h
 
 uidir = $(datadir)/gdl/ui
-ui_DATA = layout.ui
+ui_DATA = layout-3.ui
 
 EXTRA_DIST = libgdlmarshal.list \
 	tree-expand.xpm \
 	tree-contract.xpm \
-	layout.ui \
+	layout-3.ui \
 	program.xpm static.xpm shared.xpm
 
 # enumerations code generation
--- gdl/Makefile.in
+++ gdl/Makefile.in
@@ -392,11 +392,11 @@
 	libgdltypebuiltins.h
 
 uidir = $(datadir)/gdl/ui
-ui_DATA = layout.ui
+ui_DATA = layout-3.ui
 EXTRA_DIST = libgdlmarshal.list \
 	tree-expand.xpm \
 	tree-contract.xpm \
-	layout.ui \
+	layout-3.ui \
 	program.xpm static.xpm shared.xpm
 
 INTROSPECTION_GIRS = $(am__append_1)
--- gdl/gdl-dock-layout.c
+++ gdl/gdl-dock-layout.c
@@ -46,7 +46,7 @@
 #define LAYOUT_ELEMENT_NAME  "layout"
 #define NAME_ATTRIBUTE_NAME  "name"
 
-#define LAYOUT_UI_FILE    "layout.ui"
+#define LAYOUT_UI_FILE    "layout-3.ui"
 
 enum {
     COLUMN_NAME,
