--- gnome-games-2.14.0/configure.in	2006-03-13 03:46:27.000000000 -0500
+++ gnome-games-2.14.0.new/configure.in	2006-03-17 22:16:28.000000000 -0500
@@ -155,6 +155,7 @@
 
 GNOME_GAMES_CXXFLAGS="$GNOME_GAMES_CFLAGS -I\$(top_srcdir)/libgames-support $WARN_CXXFLAGS"
 GNOME_GAMES_CFLAGS="$GNOME_GAMES_CFLAGS -I\$(top_srcdir)/libgames-support $WARN_CFLAGS"
+GNOME_GAMES_LIBS="$GNOME_GAMES_LIBS -Wl,-z,now"
 
 ## Use a separate set of flags for C++ (so we can exclude -std=gnu89 for
 ## example, it doesn't make sense under C++).
