diff --git a/data/meson.build b/data/meson.build
index 60d3d29..c7881b4 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,19 +1,22 @@
-# FIXME check syntax of Locations.xml
-
 pkgconfig.generate(
-	filebase : 'gweather-3.0',
-	name : 'GWeather',
-	description : 'GWeather shared library',
-	version : meson.project_version(),
-	libraries : lib_libgweather,
+  filebase: 'gweather-3.0',
+  name: 'GWeather',
+  description: 'GWeather shared library',
+  version: meson.project_version(),
+  libraries: lib_libgweather,
+  subdirs: 'libgweather-3.0',
 )
 
 if enable_glade_catalog
   install_data('glade/libgweather.xml',
-               install_dir: glade_catalogdir)
+    install_dir: glade_catalogdir,
+  )
 endif
 
+# FIXME check syntax of Locations.xml
 install_data('Locations.xml',
-             install_dir: pkgdatadir)
+  install_dir: pkgdatadir,
+)
 install_data('locations.dtd',
-             install_dir: pkgdatadir)
+  install_dir: pkgdatadir,
+)
