From f2a45390d982c08bf1871a33756e537430de3831 Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Tue, 09 Mar 2010 15:50:58 +0000
Subject: Bug 612168 - Crash while pasting over a selection

---
diff --git a/components/editor/gtkhtml-editor.c b/components/editor/gtkhtml-editor.c
index 42cd102..8f10855 100644
--- a/components/editor/gtkhtml-editor.c
+++ b/components/editor/gtkhtml-editor.c
@@ -1637,7 +1637,7 @@ gtkhtml_editor_insert_text (GtkhtmlEditor *editor,
 
 	html = gtkhtml_editor_get_html (editor);
 
-	html_engine_insert_text (html->engine, plain_text, -1);
+	html_engine_paste_text (html->engine, plain_text, -1);
 }
 
 gboolean
--
cgit v0.8.3.1
