From 477ddd90c29aba07a9ab2319a6fe9e6b6acef5c1 Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip@tecnocode.co.uk>
Date: Mon, 04 Jul 2011 22:15:56 +0000
Subject: youtube: Add a missing case to a switch statement

---
diff --git a/src/media/youtube/grl-youtube.c b/src/media/youtube/grl-youtube.c
index 7a08683..2bdaf01 100644
--- a/src/media/youtube/grl-youtube.c
+++ b/src/media/youtube/grl-youtube.c
@@ -1430,6 +1430,7 @@ grl_youtube_source_browse (GrlMediaSource *source,
     case YOUTUBE_MEDIA_TYPE_CATEGORY:
       produce_from_category (os);
       break;
+    case YOUTUBE_MEDIA_TYPE_VIDEO:
     default:
       g_assert_not_reached ();
       break;
--
cgit v0.9
