From e4c40d02fc1b6627b66682a28cdbb4cbd71aad7e Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip@tecnocode.co.uk>
Date: Mon, 04 Jul 2011 21:56:05 +0000
Subject: youtube: Fix variable shadowing a parameter

---
diff --git a/src/media/youtube/grl-youtube.c b/src/media/youtube/grl-youtube.c
index 66609dd..050a80d 100644
--- a/src/media/youtube/grl-youtube.c
+++ b/src/media/youtube/grl-youtube.c
@@ -604,9 +604,9 @@ build_media_from_entry (GrlYoutubeSource *source,
 	gdata_youtube_video_look_up_content (video,
 					     "application/x-shockwave-flash");
       if (youtube_content != NULL) {
-	GDataMediaContent *content = GDATA_MEDIA_CONTENT (youtube_content);
-	grl_media_set_external_player (media,
-				       gdata_media_content_get_uri (content));
+        const gchar *uri =
+          gdata_media_content_get_uri (GDATA_MEDIA_CONTENT (youtube_content));
+	grl_media_set_external_player (media, uri);
       }
     }
     iter = g_list_next (iter);
--
cgit v0.9
