From 9c1ccee8cadddeb6dd5d61e90ca65b3e951e326f Mon Sep 17 00:00:00 2001
From: Lionel Landwerlin <lionel.g.landwerlin@linux.intel.com>
Date: Sun, 03 Jul 2011 01:46:53 +0000
Subject: local-metadata: fix memory leak on picture resolve

A reference was still hold on a GFile object.

Signed-off-by: Lionel Landwerlin <lionel.g.landwerlin@linux.intel.com>
---
diff --git a/src/metadata/local-metadata/grl-local-metadata.c b/src/metadata/local-metadata/grl-local-metadata.c
index 0ba1fcc..d1e3208 100644
--- a/src/metadata/local-metadata/grl-local-metadata.c
+++ b/src/metadata/local-metadata/grl-local-metadata.c
@@ -570,6 +570,7 @@ resolve_image (GrlMetadataSource *source,
     g_file_query_info_async (file, G_FILE_ATTRIBUTE_THUMBNAIL_PATH,
                              G_FILE_QUERY_INFO_NONE, G_PRIORITY_DEFAULT, cancellable,
                              (GAsyncReadyCallback)got_file_info, rs);
+    g_object_unref (file);
   }
 }
 
--
cgit v0.9
