commit 3f96295996ede06f9b0be681a23bb3765469865b
Author: Sebastian Dröge <sebastian.droege@collabora.co.uk>
Date:   Wed Jan 13 10:39:05 2010 +0100

    Fix syntax error in the GConf schema

diff --git a/data/epiphany.schemas.in b/data/epiphany.schemas.in
index bdfa8c2..0175a6f 100644
--- a/data/epiphany.schemas.in
+++ b/data/epiphany.schemas.in
@@ -16,7 +16,7 @@
         <applyto>/apps/epiphany/general/url_search</applyto>
         <owner>epiphany</owner>
         <type>string</type>
-        <default>http://www.google.com/search?q=%s&ie=UTF-8&oe=UTF-8</default>
+        <default>http://www.google.com/search?q=%s&amp;ie=UTF-8&amp;oe=UTF-8</default>
         <locale name="C">
         <short>URL Search</short>
         <long>Search string for keywords entered in the URL bar.</long>
