From 68bbd6ce05ee700ea1c77daae4c063ab7ae5b853 Mon Sep 17 00:00:00 2001
From: Priit Laes <plaes@plaes.org>
Date: Sun, 27 Sep 2009 14:02:31 +0000
Subject: src/Makefile.am: Fix parallel-build issue when introspection enabled.

---
diff --git a/src/Makefile.am b/src/Makefile.am
index 456d63b..9b7872a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -315,7 +315,7 @@ EPHY_GIR_H_FILES = \
 	$(top_srcdir)/src/ephy-window.h \
 	$(NULL)
 
-Epiphany-$(EPIPHANY_API_VERSION).gir: $(G_IR_SCANNER) $(EPHY_GIR_H_FILES)
+Epiphany-$(EPIPHANY_API_VERSION).gir: $(G_IR_SCANNER) $(EPHY_GIR_H_FILES) epiphany
 	$(AM_V_GEN) PKG_CONFIG_PATH=$(top_builddir)/data:$$PKG_CONFIG_PATH \
 	$(G_IR_SCANNER) -v \
 		--namespace Epiphany \
--
cgit v0.8.2
