--- embed/mozilla/mozilla-x509-cert.cpp.orig	2008-02-25 20:17:48.000000000 -0800
+++ embed/mozilla/mozilla-x509-cert.cpp	2008-02-25 20:18:46.000000000 -0800
@@ -213,7 +213,7 @@
 MozillaX509Cert *
 mozilla_x509_cert_new (nsIX509Cert *moz_cert)
 {
-	return g_object_new (MOZILLA_TYPE_X509_CERT,
-                             "mozilla-cert", moz_cert,
-                             NULL);
+	return MOZILLA_X509_CERT (g_object_new (MOZILLA_TYPE_X509_CERT,
+						"mozilla-cert", moz_cert,
+						NULL));
 }
