--- src/ephy-cert-manager-dialog.c.orig	2008-02-25 20:24:37.000000000 -0800
+++ src/ephy-cert-manager-dialog.c	2008-02-25 20:25:25.000000000 -0800
@@ -109,6 +109,8 @@
 	N_COLUMNS
 };
 
+G_DEFINE_TYPE (CertsManagerDialog, certs_manager_dialog, EPHY_TYPE_DIALOG)
+
 static void
 init_tree_view (CertsManagerDialog *dialog, GtkTreeView *tree_view)
 {
@@ -479,8 +481,6 @@
 
 /* public functions */
 
-G_DEFINE_TYPE (CertsManagerDialog, certs_manager_dialog, EPHY_TYPE_DIALOG)
-
 EphyDialog *
 certs_manager_dialog_new (void)
 {
