Index: capplets/common/ChangeLog
===================================================================
--- capplets/common/ChangeLog	(révision 8854)
+++ capplets/common/ChangeLog	(révision 8855)
@@ -1,3 +1,8 @@
+2008-08-12  Jens Granseuer  <jensgr@gmx.net>
+
+	* gnome-theme-apply.c: (gnome_meta_theme_set): make sure we have a
+	notification theme before trying to save it (bug #547448)
+
 2008-08-03  Jens Granseuer  <jensgr@gmx.net>
 
 	* gnome-theme-apply.c: (gnome_meta_theme_set): if the metatheme loaded
Index: capplets/common/gnome-theme-apply.c
===================================================================
--- capplets/common/gnome-theme-apply.c	(révision 8854)
+++ capplets/common/gnome-theme-apply.c	(révision 8855)
@@ -100,12 +100,15 @@
   g_free (old_key);
 
   /* set the notification theme */
-  old_key = gconf_client_get_string (client, NOTIFICATION_THEME_KEY, NULL);
-  if (compare (old_key, meta_theme_info->notification_theme_name))
+  if (meta_theme_info->notification_theme_name != NULL)
     {
-      gconf_client_set_string (client, NOTIFICATION_THEME_KEY, meta_theme_info->notification_theme_name, NULL);
+      old_key = gconf_client_get_string (client, NOTIFICATION_THEME_KEY, NULL);
+      if (compare (old_key, meta_theme_info->notification_theme_name))
+        {
+          gconf_client_set_string (client, NOTIFICATION_THEME_KEY, meta_theme_info->notification_theme_name, NULL);
+        }
+      g_free (old_key);
     }
-  g_free (old_key);
 
   /* Set the cursor theme key */
 #ifdef HAVE_XCURSOR
