From 6ee7b75e0f87bc39baa262b270c73f7de699c406 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 11 Jun 2011 11:40:39 +0000
Subject: user-accounts: Fix deleting users on 32-bit systems

https://bugzilla.gnome.org/show_bug.cgi?id=652073
---
diff --git a/panels/user-accounts/um-user-manager.c b/panels/user-accounts/um-user-manager.c
index 6bf8b42..442809a 100644
--- a/panels/user-accounts/um-user-manager.c
+++ b/panels/user-accounts/um-user-manager.c
@@ -577,7 +577,7 @@ um_user_manager_delete_user (UmUserManager       *manager,
                                  delete_user_done,
                                  data,
                                  async_user_op_data_free,
-                                 G_TYPE_INT64, um_user_get_uid (user),
+                                 G_TYPE_INT64, (gint64) um_user_get_uid (user),
                                  G_TYPE_BOOLEAN, remove_files,
                                  G_TYPE_INVALID);
 }
--
cgit v0.9
