From e9bd833af23596529f5007409c75b79227118d8a Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@gentoo.org>
Date: Wed, 23 Mar 2011 12:25:55 +0530
Subject: [PATCH] shell: fix shell launcher .desktop file name in gnomecc.menu

---
 shell/gnomecc.menu |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/shell/gnomecc.menu b/shell/gnomecc.menu
index edf829a..8f1c34d 100644
--- a/shell/gnomecc.menu
+++ b/shell/gnomecc.menu
@@ -39,7 +39,7 @@
 
   <!-- Avoid the shell having a launcher for itself -->
   <Exclude>
-    <Filename>gnome-control-center.desktop</Filename>
+    <Filename>gnome-shell.desktop</Filename>
   </Exclude>
 
   <!-- Personal category -->
-- 
1.7.3.4

