From 49633b3f4511c510a460dad41d76f6d36023740b Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 4 Oct 2009 20:30:26 +0200
Subject: [PATCH] Add missing -lm to create_branching_keyboard link

GNOME bug #589120.
---
 gok/Makefile.am |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/gok/Makefile.am b/gok/Makefile.am
index 85099c1..c041197 100644
--- a/gok/Makefile.am
+++ b/gok/Makefile.am
@@ -120,5 +120,6 @@ test_gok_wordcomplete_SOURCES = \
 keyboard_geometry_SOURCES = keyboard-geometry.c
 
 create_branching_keyboard_SOURCES = create-branching-keyboard.c
+create_branching_keyboard_LDADD = $(LDADD) -lm
 
 -include $(top_srcdir)/git.mk
-- 
1.6.5.rc1

