From ef500ff0511b8450838738b7ea5cad73c9c2a715 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Wed, 18 Aug 2010 13:33:17 +0000
Subject: Revert "Remove remaining single-include guards"

This reverts commit 5c195a2e2f8dac4d28ecdae9004d3d80d462604a.

Bug #622940.
---
diff --git a/gdk-pixbuf/gdk-pixbuf-enum-types.h.template b/gdk-pixbuf/gdk-pixbuf-enum-types.h.template
index 3989980..4ecb716 100644
--- a/gdk-pixbuf/gdk-pixbuf-enum-types.h.template
+++ b/gdk-pixbuf/gdk-pixbuf-enum-types.h.template
@@ -1,5 +1,5 @@
 /*** BEGIN file-header ***/
-#if !defined (GDK_PIXBUF_H_INSIDE) && !defined (GDK_PIXBUF_COMPILATION)
+#if defined(GDK_PIXBUF_DISABLE_SINGLE_INCLUDES) && !defined (GDK_PIXBUF_H_INSIDE) && !defined (GDK_PIXBUF_COMPILATION)
 #error "Only <gdk-pixbuf/gdk-pixbuf.h> can be included directly."
 #endif
 
diff --git a/gdk-pixbuf/gdk-pixbuf-features.h.in b/gdk-pixbuf/gdk-pixbuf-features.h.in
index ec5f647..ea1674d 100644
--- a/gdk-pixbuf/gdk-pixbuf-features.h.in
+++ b/gdk-pixbuf/gdk-pixbuf-features.h.in
@@ -1,4 +1,4 @@
-#if !defined (GDK_PIXBUF_H_INSIDE) && !defined (GDK_PIXBUF_COMPILATION)
+#if defined(GDK_PIXBUF_DISABLE_SINGLE_INCLUDES) && !defined (GDK_PIXBUF_H_INSIDE) && !defined (GDK_PIXBUF_COMPILATION)
 #error "Only <gdk-pixbuf/gdk-pixbuf.h> can be included directly."
 #endif
 
--
cgit v0.8.3.1
