Index: configure.in
===================================================================
--- configure.in	(révision 2495)
+++ configure.in	(copie de travail)
@@ -177,36 +182,55 @@
 
 AC_CHECK_FUNCS(getuid sigaction fsync fchmod fdwalk)
 
+dnl **************************************************
+dnl LDAP support.
+dnl **************************************************
 
 LDAP_LIBS=
-AC_CHECK_HEADER(ldap.h, [
-    AC_CHECK_LIB(ldap, ldap_init, AC_CHECK_LIB(lber, ber_free, [LDAP_LIBS="-lldap -llber"]))
+
+AC_ARG_WITH(openldap,
+	    AC_HELP_STRING([--with-openldap],
+			   [Enable LDAP backend @<:@default=auto@:>@]),
+	    ,with_openldap=auto)
+
+if test x"$with_openldap" != xno; then
+  AC_CHECK_HEADER(ldap.h, [
+      AC_CHECK_LIB(ldap, ldap_init, AC_CHECK_LIB(lber, ber_free, [LDAP_LIBS="-lldap -llber"]))
   ])
-AC_SUBST(LDAP_LIBS)
-AM_CONDITIONAL(LDAP_SUPPORT, test -n "$LDAP_LIBS")
+  AC_SUBST(LDAP_LIBS)
 
-dnl **************************************************
-dnl LDAP support.
-dnl **************************************************
-if test "$os_win32" != yes; then
+  if test "$os_win32" != yes; then
+  SAVE_CFLAGS="$CFLAGS"
+  SAVE_LIBS="$LIBS"
+  LDAP_CFLAGS="-DLDAP_DEPRECATED"
+  CFLAGS="$CFLAGS $LDAP_CFLAGS"
+  LIBS="$LIBS $LDAP_LIBS"
+  AC_CHECK_FUNCS(ldap_ntlm_bind)
+  CFLAGS="$SAVE_CFLAGS"
+  LIBS="$SAVE_LIBS"
+
+  else # Win32
+  LDAP_CFLAGS="-DLDAP_DEPRECATED"
+  LDAP_LIBS="-lwldap32"
+  AC_SUBST(LDAP_CFLAGS)
+  AC_SUBST(LDAP_LIBS)
+  AC_DEFINE(HAVE_LDAP,1,[Define if you have LDAP support])
+  msg_ldap="yes"
+  fi # Win32
+
+  if test -z "$LDAP_LIBS"; then
+    if test x"$with_openldap" = xyes; then
+      AC_MSG_ERROR([*** LDAP support explicitely requested but not found ***])
+    else
+      AC_MSG_WARN([*** LDAP support not found ***])
+    fi
+  fi
+
+else
+  AC_DEFINE(HAVE_LDAP,0,[Define if you have LDAP support])
+fi
 
-SAVE_CFLAGS="$CFLAGS"
-SAVE_LIBS="$LIBS"
-LDAP_CFLAGS="-DLDAP_DEPRECATED"
-CFLAGS="$CFLAGS $LDAP_CFLAGS"
-LIBS="$LIBS $LDAP_LIBS"
-AC_CHECK_FUNCS(ldap_ntlm_bind)
-CFLAGS="$SAVE_CFLAGS"
-LIBS="$SAVE_LIBS"
-
-else # Win32
-LDAP_CFLAGS="-DLDAP_DEPRECATED"
-LDAP_LIBS="-lwldap32"
-AC_SUBST(LDAP_CFLAGS)
-AC_SUBST(LDAP_LIBS)
-AC_DEFINE(HAVE_LDAP,1,[Define if you have LDAP support])
-msg_ldap="yes"
-fi # Win32
+AM_CONDITIONAL(LDAP_SUPPORT, test -n "$LDAP_LIBS")
 
 IT_PROG_INTLTOOL([0.35.0])
 
