--- gtk-doc.make.orig	2008-01-08 11:47:49.000000000 -0800
+++ gtk-doc.make	2008-01-08 11:48:25.000000000 -0800
@@ -139,8 +139,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase > /dev/null && \
-            gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
+	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 	
 
