--- configure.in.orig	2008-01-22 09:33:40.000000000 -0800
+++ configure.in	2008-01-22 09:35:21.000000000 -0800
@@ -127,7 +127,11 @@
   fi
 fi
 
+dnl Crazy hack for dropping dep on gtk-doc
+AC_DEFUN(GTK_DOC_CHECK, [])
 GTK_DOC_CHECK([1.0])
+AM_CONDITIONAL(ENABLE_GTK_DOC, 0, [])
+AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, 0, [])
 
 AC_ARG_ENABLE(gtk, [  --enable-gtk          Enable GTK+ support (for gconf-sanity-check) [default=auto]], enable_gtk="$enableval", enable_gtk=auto)
 
--- Makefile.am.orig	2008-01-22 09:35:24.000000000 -0800
+++ Makefile.am	2008-01-22 09:35:44.000000000 -0800
@@ -1,7 +1,11 @@
 
-SUBDIRS = gconf backends po doc examples
+SUBDIRS = gconf backends po examples
 DIST_SUBDIRS=tests $(SUBDIRS)
 
+if ENABLE_GTK_DOC
+SUBDIRS += doc 
+endif
+
 EXTRA_DIST =                \
 	TODO                \
 	gtk-doc.make        \
