diff -Naur DeviceKit-power-007.orig/src/dkp-supply.c DeviceKit-power-007/src/dkp-supply.c
--- DeviceKit-power-007.orig/src/dkp-supply.c	2009-03-09 15:01:43.000000000 +0530
+++ DeviceKit-power-007/src/dkp-supply.c	2009-04-22 20:19:00.396401215 +0530
@@ -262,9 +262,9 @@
 	gboolean is_present;
 	gdouble energy;
 	gdouble energy_full;
-	gdouble energy_full_design;
+	gdouble energy_full_design = 0.0;
 	gdouble energy_rate;
-	gdouble capacity;
+	gdouble capacity = 0.0;
 	gdouble percentage;
 	gdouble voltage;
 	guint64 time_to_empty;
