 configure.ac |   24 +++++++++++++++++++-----
 1 files changed, 19 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index b8d43d5..d114bd7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -56,16 +56,30 @@ dnl ---------------------------------------------------------------------------
 dnl - Extra verbose warning switches, disable if needed
 dnl ---------------------------------------------------------------------------
 if test "$GCC" = "yes"; then
-	CPPFLAGS="$CPPFLAGS -Werror -Wcast-align -Wno-uninitialized"
-	CPPFLAGS="$CPPFLAGS -Wall -Wformat-security"
+	CPPFLAGS="$CPPFLAGS -Wall"
 #	CPPFLAGS="$CPPFLAGS -Wall"
 fi
 
-if test "x$GCC" = "xyes"; then
+if test "x${USE_MAINTAINER_MODE}" = "xyes" -a "x$GCC" = "xyes"; then
   changequote(,)dnl
   case " $CFLAGS " in
-  *[\ \	]-Wall[\ \	]*) ;;
-  *) CFLAGS="$CFLAGS -Wall" ;;
+  *[\ \	]-Werror[\ \	]*) ;;
+  *) CFLAGS="$CFLAGS -Werror" ;;
+  esac
+
+  case " $CFLAGS " in
+  *[\ \	]-Wcast-align[\ \	]*) ;;
+  *) CFLAGS="$CFLAGS -Wcast-align" ;;
+  esac
+
+  case " $CFLAGS " in
+  *[\ \	]-Wno-uninitialized[\ \	]*) ;;
+  *) CFLAGS="$CFLAGS -Wno-uninitialized" ;;
+  esac
+
+  case " $CFLAGS " in
+  *[\ \	]-Wcast-align[\ \	]*) ;;
+  *) CFLAGS="$CFLAGS -Wformat-security" ;;
   esac
 
   case " $CFLAGS " in
