diff -Nura deskbar-applet-2.24.1.old/deskbar/core/keybinder/Makefile.am deskbar-applet-2.24.1/deskbar/core/keybinder/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/core/keybinder/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/core/keybinder/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(DESKBAR_CFLAGS)
 
-_keybinderdir = $(pythondir)/deskbar/core/keybinder
+_keybinderdir = $(pyexecdir)/deskbar/core/keybinder
 _keybinder_LTLIBRARIES = _keybinder.la
 _keybinder_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_keybinder
 _keybinder_la_LIBADD = $(DESKBAR_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/core/Makefile.am deskbar-applet-2.24.1/deskbar/core/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/core/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/core/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,6 +1,6 @@
 SUBDIRS = keybinder updater _userdirs
 	
-deskbar_coredir = $(pythondir)/deskbar/core
+deskbar_coredir = $(pyexecdir)/deskbar/core
 deskbar_core_PYTHON = \
 	__init__.py \
 	BrowserMatch.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/core/updater/Makefile.am deskbar-applet-2.24.1/deskbar/core/updater/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/core/updater/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/core/updater/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,4 +1,4 @@
-updaterdir = $(pythondir)/deskbar/core/updater
+updaterdir = $(pyexecdir)/deskbar/core/updater
 updater_PYTHON = \
 	__init__.py \
 	Capuchin.py
diff -Nura deskbar-applet-2.24.1.old/deskbar/core/_userdirs/Makefile.am deskbar-applet-2.24.1/deskbar/core/_userdirs/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/core/_userdirs/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/core/_userdirs/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(DESKBAR_CFLAGS) $(GLIB_CFLAGS)
 
-__userdirsdir = $(pythondir)/deskbar/core/_userdirs
+__userdirsdir = $(pyexecdir)/deskbar/core/_userdirs
 __userdirs_LTLIBRARIES = __userdirs.la
 __userdirs_la_LDFLAGS = -module -avoid-version -export-symbols-regex init__userdirs
 __userdirs_la_LIBADD = $(DESKBAR_LIBS) $(GLIB_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/handlers/actions/Makefile.am deskbar-applet-2.24.1/deskbar/handlers/actions/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/handlers/actions/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/handlers/actions/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -1,4 +1,4 @@
-deskbar_actionsdir = $(pythondir)/deskbar/handlers/actions
+deskbar_actionsdir = $(pyexecdir)/deskbar/handlers/actions
 deskbar_actions_PYTHON = \
 	__init__.py \
 	ActionsFactory.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/handlers/evolution/Makefile.am deskbar-applet-2.24.1/deskbar/handlers/evolution/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/handlers/evolution/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/handlers/evolution/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(EVOLUTION_CFLAGS)
 
-_evolutiondir = $(pythondir)/deskbar/handlers/evolution
+_evolutiondir = $(pyexecdir)/deskbar/handlers/evolution
 _evolution_LTLIBRARIES = _evolution.la
 _evolution_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_evolution
 _evolution_la_LIBADD = $(EVOLUTION_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/handlers/gdmclient/Makefile.am deskbar-applet-2.24.1/deskbar/handlers/gdmclient/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/handlers/gdmclient/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/handlers/gdmclient/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(DESKBAR_CFLAGS)
 
-_gdmclientdir = $(pythondir)/deskbar/handlers/gdmclient
+_gdmclientdir = $(pyexecdir)/deskbar/handlers/gdmclient
 _gdmclient_LTLIBRARIES = _gdmclient.la
 _gdmclient_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_gdmclient
 _gdmclient_la_LIBADD = $(DESKBAR_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/handlers/Makefile.am deskbar-applet-2.24.1/deskbar/handlers/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/handlers/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/handlers/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -4,7 +4,7 @@
 SUBDIRS += evolution
 endif
 
-deskbar_handlers_modulesdir = $(pythondir)/deskbar/handlers
+deskbar_handlers_modulesdir = $(pyexecdir)/deskbar/handlers
 deskbar_handlers_modules_PYTHON = __init__.py
 
 deskbar_handlersdir = $(libdir)/deskbar-applet/modules-2.20-compatible
diff -Nura deskbar-applet-2.24.1.old/deskbar/interfaces/Makefile.am deskbar-applet-2.24.1/deskbar/interfaces/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/interfaces/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/interfaces/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,4 +1,4 @@
-deskbar_interfacesdir = $(pythondir)/deskbar/interfaces
+deskbar_interfacesdir = $(pyexecdir)/deskbar/interfaces
 deskbar_interfaces_PYTHON = \
 	__init__.py \
 	Action.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/Makefile.am deskbar-applet-2.24.1/deskbar/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -6,7 +6,7 @@
 deskbarbindir = $(libdir)/deskbar-applet
 deskbarbin_SCRIPTS = deskbar-applet
 
-deskbardir = $(pythondir)/deskbar
+deskbardir = $(pyexecdir)/deskbar
 deskbar_PYTHON = \
 	defs.py \
 	gtkexcepthook.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/osutils/Makefile.am deskbar-applet-2.24.1/deskbar/osutils/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/osutils/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/osutils/Makefile.am	2008-10-24 08:53:31.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(DESKBAR_CFLAGS)
 
-_osutilsdir = $(pythondir)/deskbar/osutils
+_osutilsdir = $(pyexecdir)/deskbar/osutils
 _osutils_LTLIBRARIES = _osutils.la
 _osutils_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_osutils
 _osutils_la_LIBADD = $(DESKBAR_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/ui/cuemiac/Makefile.am deskbar-applet-2.24.1/deskbar/ui/cuemiac/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/ui/cuemiac/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/ui/cuemiac/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,4 +1,4 @@
-deskbar_cuemiacdir = $(pythondir)/deskbar/ui/cuemiac
+deskbar_cuemiacdir = $(pyexecdir)/deskbar/ui/cuemiac
 deskbar_cuemiac_PYTHON = \
 	__init__.py \
 	CuemiacActionsTreeView.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/ui/iconentry/Makefile.am deskbar-applet-2.24.1/deskbar/ui/iconentry/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/ui/iconentry/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/ui/iconentry/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,6 +1,6 @@
 INCLUDES = $(PYTHON_INCLUDES) $(DESKBAR_CFLAGS)
 
-_iconentrydir = $(pythondir)/deskbar/ui/iconentry
+_iconentrydir = $(pyexecdir)/deskbar/ui/iconentry
 _iconentry_LTLIBRARIES = _iconentry.la
 _iconentry_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_iconentry
 _iconentry_la_LIBADD = $(DESKBAR_LIBS)
diff -Nura deskbar-applet-2.24.1.old/deskbar/ui/Makefile.am deskbar-applet-2.24.1/deskbar/ui/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/ui/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/ui/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,6 +1,6 @@
 SUBDIRS = cuemiac iconentry preferences
 
-deskbar_uidir = $(pythondir)/deskbar/ui
+deskbar_uidir = $(pyexecdir)/deskbar/ui
 deskbar_ui_PYTHON = \
 	__init__.py \
 	About.py \
diff -Nura deskbar-applet-2.24.1.old/deskbar/ui/preferences/Makefile.am deskbar-applet-2.24.1/deskbar/ui/preferences/Makefile.am
--- deskbar-applet-2.24.1.old/deskbar/ui/preferences/Makefile.am	2008-10-24 08:50:46.000000000 +0200
+++ deskbar-applet-2.24.1/deskbar/ui/preferences/Makefile.am	2008-10-24 08:53:30.000000000 +0200
@@ -1,4 +1,4 @@
-deskbar_preferencesdir = $(pythondir)/deskbar/ui/preferences
+deskbar_preferencesdir = $(pyexecdir)/deskbar/ui/preferences
 deskbar_preferences_PYTHON = \
 	__init__.py \
 	AccelEntry.py \
