From 3388135658e8a8f56da7fd01015490c2c97958ee Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@gentoo.org>
Date: Thu, 2 Jun 2011 22:42:33 +0530
Subject: [PATCH] Fix bug 651723, don't look for libnotify.h if not requested

---
 src/notification.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/notification.c b/src/notification.c
index 41d64e2..3f4c454 100644
--- a/src/notification.c
+++ b/src/notification.c
@@ -29,12 +29,12 @@
 #include <gtk/gtk.h>
 #include <time.h>
 #include <string.h>
+#ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
 
 #include "zenity.h"
 #include "util.h"
 
-#ifdef HAVE_LIBNOTIFY
 static char *icon_file;
 
 static void
-- 
1.7.3.4

