--- src/terminal-screen.c	2006-02-04 19:35:24.000000000 -0600
+++ src/terminal-screen.c	2006-02-04 19:35:33.000000000 -0600
@@ -1111,7 +1111,7 @@
   retval[i] = g_strdup ("COLORTERM="EXECUTABLE_NAME);
   ++i;
 
-  retval[i] = g_strdup ("TERM=xterm"); /* FIXME configurable later? */
+  retval[i] = g_strdup ("TERM=gnome"); /* FIXME configurable later? */
   ++i;
 
   retval[i] = g_strdup_printf ("WINDOWID=%ld",
