diff -ur gnome-terminal-2.0.2/src/terminal-screen.c gnome-terminal-2.0.2-spaced/src/terminal-screen.c
--- gnome-terminal-2.0.2/src/terminal-screen.c	2002-10-19 06:24:26.000000000 +1300
+++ gnome-terminal-2.0.2-spaced/src/terminal-screen.c	2003-01-13 10:42:10.000000000 +1300
@@ -1820,6 +1820,9 @@
                                                str->str,
                                                str->len);
 
+        terminal_widget_write_data_to_child (screen->priv->term,
+                                             " ", 1);
+
         g_free (filename);        
         g_string_free (str, TRUE);
       }
@@ -1872,6 +1875,8 @@
             flat = g_strjoinv (" ", uris);
             terminal_widget_write_data_to_child (screen->priv->term,
                                                  flat, strlen (flat));
+            terminal_widget_write_data_to_child (screen->priv->term,
+                                                 " ", 1);
             g_free (flat);
           }
 
