diff --git a/ChangeLog b/ChangeLog
index 17d8ce3..43d1599 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2008-03-19  Christian Persch  <chpe@gnome.org>
+
+	* src/terminal.c: (terminal_app_new_terminal): Fix DetachTab action.
+	Bug #521193, regression from bug #353775.
+
 2008-03-10  Behdad Esfahbod  <behdad@gnome.org>
 
 	* NEWS:
diff --git a/src/terminal.c b/src/terminal.c
index 664c322..c06936a 100644
--- a/src/terminal.c
+++ b/src/terminal.c
@@ -1551,6 +1551,22 @@ terminal_app_new_terminal (TerminalApp     *app,
       terminal_window_set_active (window, screen);
       gtk_widget_grab_focus (GTK_WIDGET (screen));
     }
+  else
+    {
+      TerminalWindow *source_window;
+
+      source_window = terminal_screen_get_window (screen);
+      if (source_window)
+        {
+          g_object_ref_sink (screen);
+          terminal_window_remove_screen (source_window, screen);
+          terminal_window_add_screen (window, screen, -1);
+          g_object_unref (screen);
+
+          terminal_window_set_active (window, screen);
+          gtk_widget_grab_focus (GTK_WIDGET (screen));
+        }
+    }
 
   if (geometry)
     {
