diff --git a/stickynotes/Makefile.am b/stickynotes/Makefile.am
index d12d1bf..f5dc634 100644
--- a/stickynotes/Makefile.am
+++ b/stickynotes/Makefile.am
@@ -59,7 +59,6 @@ if GCONF_SCHEMAS_INSTALL
 install-data-local:
 	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) \
 	$(GCONFTOOL) --makefile-install-rule $(schemas_DATA)
-	$(GCONFTOOL) --shutdown
 endif
 endif
 
diff --git a/stickynotes/Makefile.in b/stickynotes/Makefile.in
index e2b72be..bfae0ae 100644
--- a/stickynotes/Makefile.in
+++ b/stickynotes/Makefile.in
@@ -875,7 +875,6 @@ uninstall-am: uninstall-builderDATA uninstall-libexecPROGRAMS \
 @BUILD_STICKYNOTES_APPLET_TRUE@@GCONF_SCHEMAS_INSTALL_TRUE@install-data-local:
 @BUILD_STICKYNOTES_APPLET_TRUE@@GCONF_SCHEMAS_INSTALL_TRUE@	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) \
 @BUILD_STICKYNOTES_APPLET_TRUE@@GCONF_SCHEMAS_INSTALL_TRUE@	$(GCONFTOOL) --makefile-install-rule $(schemas_DATA)
-@BUILD_STICKYNOTES_APPLET_TRUE@@GCONF_SCHEMAS_INSTALL_TRUE@	$(GCONFTOOL) --shutdown
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
