Index: plugins/background/gsd-background-manager.c
===================================================================
--- plugins/background/gsd-background-manager.c	(revision 639)
+++ plugins/background/gsd-background-manager.c	(working copy)
@@ -161,7 +161,10 @@
         int         n_screens;
         int         i;
 
-        if (nautilus_is_running ()) {
+        if (nautilus_is_running () ||
+            !gconf_client_get_bool (manager->priv->client,
+                                    "/desktop/gnome/background/draw_background",
+                                    NULL)) {
                 return;
         }
 
