From 9b534448f26d51ce1ac80b49483a621ac3a9540f Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Tue, 06 Sep 2011 10:19:08 +0000
Subject: gconf: Plug a memory leak

---
diff --git a/plugins/gconf/gsd-gconf-manager.c b/plugins/gconf/gsd-gconf-manager.c
index 7723e35..20c94b1 100644
--- a/plugins/gconf/gsd-gconf-manager.c
+++ b/plugins/gconf/gsd-gconf-manager.c
@@ -138,6 +138,7 @@ gsd_gconf_manager_start (GsdGconfManager *manager, GError **error)
                         /* Free all memory */
                         g_free (path);
                         g_strfreev (groups);
+                        g_key_file_free (key_file);
 
                         result = TRUE;
                 }
--
cgit v0.9.0.2
