Index: plugins/a11y-keyboard/gsd-a11y-keyboard-plugin.c
===================================================================
--- plugins/a11y-keyboard/gsd-a11y-keyboard-plugin.c	(revision 114)
+++ plugins/a11y-keyboard/gsd-a11y-keyboard-plugin.c	(working copy)
@@ -70,15 +70,12 @@
 impl_activate (GnomeSettingsPlugin *plugin)
 {
         gboolean res;
-        GError  *error;
 
         g_debug ("Activating a11y_keyboard plugin");
 
-        error = NULL;
-        res = gsd_a11y_keyboard_manager_start (GSD_A11Y_KEYBOARD_PLUGIN (plugin)->priv->manager, &error);
+        res = gsd_a11y_keyboard_manager_start (GSD_A11Y_KEYBOARD_PLUGIN (plugin)->priv->manager);
         if (! res) {
-                g_warning ("Unable to start a11y_keyboard manager: %s", error->message);
-                g_error_free (error);
+                g_warning ("Unable to start a11y_keyboard manager");
         }
 }
 
Index: plugins/a11y-keyboard/gsd-a11y-keyboard-manager.h
===================================================================
--- plugins/a11y-keyboard/gsd-a11y-keyboard-manager.h	(revision 114)
+++ plugins/a11y-keyboard/gsd-a11y-keyboard-manager.h	(working copy)
@@ -48,8 +48,7 @@
 GType                   gsd_a11y_keyboard_manager_get_type            (void);
 
 GsdA11yKeyboardManager *gsd_a11y_keyboard_manager_new                 (void);
-gboolean                gsd_a11y_keyboard_manager_start               (GsdA11yKeyboardManager *manager,
-                                                                       GError                **error);
+gboolean                gsd_a11y_keyboard_manager_start               (GsdA11yKeyboardManager *manager);
 void                    gsd_a11y_keyboard_manager_stop                (GsdA11yKeyboardManager *manager);
 
 G_END_DECLS
Index: plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c
===================================================================
--- plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c	(revision 114)
+++ plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c	(working copy)
@@ -719,8 +719,7 @@
 }
 
 gboolean
-gsd_a11y_keyboard_manager_start (GsdA11yKeyboardManager *manager,
-                                 GError                **error)
+gsd_a11y_keyboard_manager_start (GsdA11yKeyboardManager *manager)
 {
         guint        event_mask;
         GConfClient *client;
