From 33efc3649aa75fc77e4f1521031992376131d507 Mon Sep 17 00:00:00 2001
From: Chris Coulson <chris.coulson@canonical.com>
Date: Wed, 13 Jul 2011 11:58:29 +0200
Subject: [PATCH 01/11] common: Fix grabbing of multimedia keys

---
 plugins/common/gsd-keygrab.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/plugins/common/gsd-keygrab.c b/plugins/common/gsd-keygrab.c
index 70fde7a..1d50a22 100644
--- a/plugins/common/gsd-keygrab.c
+++ b/plugins/common/gsd-keygrab.c
@@ -146,9 +146,9 @@ grab_key_unsafe (Key                 *key,
          * (which are useful to grab without a modifier).
          */
         if ((modifiers & gsd_used_mods) == 0 &&
-            (IN_RANGE(key->keysym, XF86KEYS_RANGE_MIN, XF86KEYS_RANGE_MAX) ||
-             IN_RANGE(key->keysym, FKEYS_RANGE_MIN, FKEYS_RANGE_MAX) ||
-             key->keysym == GDK_KEY_Pause)) {
+            !IN_RANGE(key->keysym, XF86KEYS_RANGE_MIN, XF86KEYS_RANGE_MAX) &&
+            !IN_RANGE(key->keysym, FKEYS_RANGE_MIN, FKEYS_RANGE_MAX) &&
+             key->keysym != GDK_KEY_Pause) {
                 GString *keycodes;
 
                 keycodes = g_string_new ("");
-- 
1.7.6

