Index: browser-plugin/Makefile.am
===================================================================
--- browser-plugin/Makefile.am	(revision 4827)
+++ browser-plugin/Makefile.am	(working copy)
@@ -109,6 +109,7 @@
 	$(WARN_CFLAGS)		\
 	$(DBUS_CFLAGS)		\
 	$(NVTV_CFLAGS)		\
+	$(SN_CFLAGS)		\
 	$(AM_CFLAGS)
 
 totem_plugin_viewer_LDADD =						\
@@ -122,6 +123,7 @@
 	$(DBUS_LIBS)		\
 	$(NVTV_LIBS) 		\
 	$(XTEST_LIBS)		\
+	$(SN_LIBS)		\
 	$(X_LIBS)
 
 BUILT_SOURCES += \
Index: configure.in
===================================================================
--- configure.in	(revision 4827)
+++ configure.in	(working copy)
@@ -685,6 +686,18 @@
 AM_CONDITIONAL([ENABLE_MULLY_PLUGIN],[test "$enable_mully_plugin" = "yes"])
 AM_CONDITIONAL([ENABLE_CONE_PLUGIN],[test "$enable_cone_plugin" = "yes"])
 
+if test "$enable_basic_plugin" = "yes" ||
+   test "$enable_gmp_plugin" = "yes" ||
+   test "$enable_complex_plugin" = "yes" ||
+   test "$enable_narrowspace_plugin" = "yes" ||
+   test "$enable_mully_plugin" = "yes" ||
+   test "$enable_cone_plugin" = "yes" ; then
+
+   PKG_CHECK_MODULES([SN], [ libstartup-notification-1.0 >= 0.8 ])
+   AC_SUBST([SN_CFLAGS])
+   AC_SUBST([SN_LIBS])
+fi
+
 dnl ********
 dnl Nautilus
 dnl ********
