Index: configure.in
===================================================================
--- configure.in	(revision 4827)
+++ configure.in	(working copy)
@@ -585,6 +588,7 @@
 	if test "$have_libxpcomglue_s" = "yes"; then
 		LIBXPCOMGLUE_S="-L$MOZILLA_LIBDIR -lxpcomglue_s"
 	else
+      LIBXPCOMGLUE_S="-L$MOZILLA_LIBDIR -lxpcom"
 		AC_MSG_WARN([libxpcomglue_s not available; plugins may not be portable])
 	fi
 fi
