--- totem-2.23.4/configure.in	2008-08-13 23:30:23.000000000 +0200
+++ totem-2.23.4/configure.in	2008-08-13 23:34:57.000000000 +0200
@@ -641,7 +641,7 @@
 if test "$enable_browser_plugins" = "yes" ; then
 	AC_DEFINE([ENABLE_BROWSER_PLUGINS],[1],[Define to build the browser plugin])
 
-	BROWSER_PLUGIN_DIR="\${libdir}/mozilla/plugins"
+	BROWSER_PLUGIN_DIR="${BROWSER_PLUGIN_DIR:-"\${libdir}/mozilla/plugins"}"
 	AC_ARG_VAR([BROWSER_PLUGIN_DIR],[Where to install the plugin to])
 
 	PKG_CHECK_MODULES([BROWSER_PLUGIN],
