--- totem-2.90.0.orig/src/backend/bacon-resize.c
+++ totem-2.90.0/src/backend/bacon-resize.c
@@ -227,7 +227,7 @@ bacon_resize_resize (BaconResize *resize
 	g_return_if_fail (GTK_IS_WIDGET (resize->priv->video_widget));
 	g_return_if_fail (gtk_widget_get_realized (resize->priv->video_widget));
 
-	xdisplay = GDK_DRAWABLE_XDISPLAY (resize->priv->video_widget->window);
+	xdisplay = GDK_DRAWABLE_XDISPLAY (gtk_widget_get_window (resize->priv->video_widget));
 	if (xdisplay == NULL)
 		return;
 
@@ -308,7 +308,7 @@ bacon_resize_restore (BaconResize *resiz
 	if (resize->priv->xr_screen_conf == NULL)
 		return;
 
-	xdisplay = GDK_DRAWABLE_XDISPLAY (resize->priv->video_widget->window);
+	xdisplay = GDK_DRAWABLE_XDISPLAY (gtk_widget_get_window (resize->priv->video_widget));
 	if (xdisplay == NULL)
 		return;
 
--- totem-2.90.0.orig/src/totem-object.c
+++ totem-2.90.0/src/totem-object.c
@@ -2308,22 +2308,22 @@ drop_video_cb (GtkWidget     *widget,
 
 	/* Drop of video on itself */
 	if (source_widget && widget == source_widget &&
-	    context->action == GDK_ACTION_MOVE) {
+	    gdk_drag_context_get_selected_action (context) == GDK_ACTION_MOVE) {
 		gtk_drag_finish (context, FALSE, FALSE, _time);
 		return;
 	}
 
-	if (context->action == GDK_ACTION_ASK) {
-		context->action = totem_drag_ask (totem_get_playlist_length (totem) > 0);
+	if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_ASK) {
+		gdk_drag_status (context, totem_drag_ask (totem_get_playlist_length (totem) > 0), _time);
 	}
 
 	/* User selected cancel */
-	if (context->action == GDK_ACTION_DEFAULT) {
+	if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_DEFAULT) {
 		gtk_drag_finish (context, FALSE, FALSE, _time);
 		return;
 	}
 
-	empty_pl = (context->action == GDK_ACTION_MOVE);
+	empty_pl = (gdk_drag_context_get_selected_action (context) == GDK_ACTION_MOVE);
 	totem_action_drop_files (totem, data, info, empty_pl);
 	gtk_drag_finish (context, TRUE, FALSE, _time);
 	return;
@@ -2342,7 +2342,7 @@ drag_motion_video_cb (GtkWidget      *wi
 	gdk_window_get_pointer (gtk_widget_get_window (widget), NULL, NULL, &mask);
 	if (mask & GDK_CONTROL_MASK) {
 		gdk_drag_status (context, GDK_ACTION_COPY, _time);
-	} else if (mask & GDK_MOD1_MASK || context->suggested_action == GDK_ACTION_ASK) {
+	} else if (mask & GDK_MOD1_MASK || gdk_drag_context_get_suggested_action (context) == GDK_ACTION_ASK) {
 		gdk_drag_status (context, GDK_ACTION_ASK, _time);
 	} else {
 		gdk_drag_status (context, GDK_ACTION_MOVE, _time);
@@ -2361,15 +2361,15 @@ drop_playlist_cb (GtkWidget     *widget,
 {
 	gboolean empty_pl;
 
-	if (context->action == GDK_ACTION_ASK)
-		context->action = totem_drag_ask (totem_get_playlist_length (totem) > 0);
+	if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_ASK)
+		gdk_drag_status (context, totem_drag_ask (totem_get_playlist_length (totem) > 0), _time);
 
-	if (context->action == GDK_ACTION_DEFAULT) {
+	if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_DEFAULT) {
 		gtk_drag_finish (context, FALSE, FALSE, _time);
 		return;
 	}
 
-	empty_pl = (context->action == GDK_ACTION_MOVE);
+	empty_pl = (gdk_drag_context_get_selected_action (context) == GDK_ACTION_MOVE);
 
 	totem_action_drop_files (totem, data, info, empty_pl);
 	gtk_drag_finish (context, TRUE, FALSE, _time);
@@ -2387,7 +2387,7 @@ drag_motion_playlist_cb (GtkWidget
 
 	gdk_window_get_pointer (gtk_widget_get_window (widget), NULL, NULL, &mask);
 
-	if (mask & GDK_MOD1_MASK || context->suggested_action == GDK_ACTION_ASK) {
+	if (mask & GDK_MOD1_MASK || gdk_drag_context_get_suggested_action (context) == GDK_ACTION_ASK) {
 		gdk_drag_status (context, GDK_ACTION_ASK, _time);
 	}
 }
--- totem-2.90.0.orig/src/totem-playlist.c
+++ totem-2.90.0/src/totem-playlist.c
@@ -507,15 +507,15 @@ drop_cb (GtkWidget        *widget,
 	GList *p, *file_list;
 	guint i;
 
-	if (context->suggested_action == GDK_ACTION_ASK) {
-		context->action = totem_drag_ask (PL_LEN != 0);
-		if (context->action == GDK_ACTION_DEFAULT) {
+	if (gdk_drag_context_get_suggested_action (context) == GDK_ACTION_ASK) {
+		gdk_drag_status (context, totem_drag_ask (PL_LEN != 0), _time);
+		if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_DEFAULT) {
 			gtk_drag_finish (context, FALSE, FALSE, _time);
 			return;
 		}
 	}
 
-	if (context->action == GDK_ACTION_MOVE)
+	if (gdk_drag_context_get_selected_action (context) == GDK_ACTION_MOVE)
 		totem_playlist_clear (playlist);
 
 	list = g_uri_list_extract_uris ((char *) gtk_selection_data_get_data (data));
