diff --git a/src/hdy-combo-row.c b/src/hdy-combo-row.c
index 633dd7d00e16fc8581823ecc501e53b0908c2ca6..b221c220a95ea0ff5d365b0e9df396962d2ffc74 100644
--- a/src/hdy-combo-row.c
+++ b/src/hdy-combo-row.c
@@ -479,7 +479,7 @@ hdy_combo_row_set_for_enum (HdyComboRow                     *self,
                             GDestroyNotify                   user_data_free_func)
 {
   g_autoptr (GListStore) store = g_list_store_new (HDY_TYPE_ENUM_VALUE_OBJECT);
-  g_autoptr (GEnumClass) enum_class = g_type_class_ref (enum_type);
+  GEnumClass *enum_class = g_type_class_ref (enum_type);
   gsize i;
 
   g_return_if_fail (HDY_IS_COMBO_ROW (self));
@@ -488,6 +488,7 @@ hdy_combo_row_set_for_enum (HdyComboRow                     *self,
     g_list_store_append (store, hdy_enum_value_object_new (&enum_class->values[i]));
 
   hdy_combo_row_bind_name_model (self, G_LIST_MODEL (store), (HdyComboRowGetNameFunc) get_name_func, user_data, user_data_free_func);
+  g_type_class_unref (enum_class);
 }
 
 /**
