--- Makefile.am.orig	2013-08-30 15:04:11.011724801 +0400
+++ Makefile.am	2013-08-30 15:04:32.931449233 +0400
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk libgail-util modules  tests uite examples
+SRC_SUBDIRS = gdk gtk libgail-util modules testsuite examples
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
