From ec26a909f69e12dee1d151c8795f62d794fe7b24 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Wed, 21 Aug 2013 02:30:05 +0000
Subject: clipboard: fix build without wayland

Signed-off-by: Saleem Abdulrasool <compnerd@compnerd.org>
---
diff --git a/gtk/gtkclipboard-wayland.c b/gtk/gtkclipboard-wayland.c
index 8ab1730..d3eb906 100644
--- a/gtk/gtkclipboard-wayland.c
+++ b/gtk/gtkclipboard-wayland.c
@@ -23,6 +23,8 @@
 
 #include "gtkclipboard-waylandprivate.h"
 
+#ifdef GDK_WINDOWING_WAYLAND
+
 #include <string.h>
 
 #include "gtkmain.h"
@@ -30,8 +32,6 @@
 #include "gtkintl.h"
 #include "gtkselectionprivate.h"
 
-#ifdef GDK_WINDOWING_WAYLAND
-
 static void gtk_clipboard_wayland_owner_change     (GtkClipboard             *clipboard,
                                                     GdkEventOwnerChange      *event);
 static gboolean gtk_clipboard_wayland_set_contents (GtkClipboard             *clipboard,
diff --git a/gtk/gtkclipboard-waylandprivate.h b/gtk/gtkclipboard-waylandprivate.h
index 7a5b58e..d2d1f22 100644
--- a/gtk/gtkclipboard-waylandprivate.h
+++ b/gtk/gtkclipboard-waylandprivate.h
@@ -18,11 +18,13 @@
 #ifndef __GTK_CLIPBOARD_WAYLAND_WAYLAND_PRIVATE_H__
 #define __GTK_CLIPBOARD_WAYLAND_WAYLAND_PRIVATE_H__
 
-#include <gdk/wayland/gdkwayland.h>
-#include <gtk/gtkclipboardprivate.h>
+#include <gdk/gdk.h>
 
 #ifdef GDK_WINDOWING_WAYLAND
 
+#include <gdk/wayland/gdkwayland.h>
+#include <gtk/gtkclipboardprivate.h>
+
 #define GTK_TYPE_CLIPBOARD_WAYLAND            (gtk_clipboard_wayland_get_type ())
 #define GTK_CLIPBOARD_WAYLAND(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_CLIPBOARD_WAYLAND, GtkClipboardWayland))
 #define GTK_IS_CLIPBOARD_WAYLAND(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_TYPE_CLIPBOARD_WAYLAND))
--
cgit v0.9.2
