Index: gtk+2.0-2.11.6/gtk/gtkclipboard.c
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtkclipboard.c	2007-08-31 16:52:48.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtkclipboard.c	2007-08-31 16:53:16.000000000 +0200
@@ -287,6 +287,7 @@
 gtk_clipboard_get_for_display (GdkDisplay *display, 
 			       GdkAtom     selection)
 {
+  g_return_val_if_fail (display != NULL, NULL);
   g_return_val_if_fail (GDK_IS_DISPLAY (display), NULL);
   g_return_val_if_fail (!display->closed, NULL);
 
