From 9ddc11fbb5a2e4bb1f9d71c7278feab874add1d6 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Wed, 14 Sep 2011 14:09:36 +0000
Subject: iconview: Don't rturn TRUE from draw handler

---
diff --git a/gtk/gtkiconview.c b/gtk/gtkiconview.c
index 8b4a184..2324889 100644
--- a/gtk/gtkiconview.c
+++ b/gtk/gtkiconview.c
@@ -1683,9 +1683,7 @@ gtk_icon_view_draw (GtkWidget *widget,
 
   cairo_restore (cr);
 
-  GTK_WIDGET_CLASS (gtk_icon_view_parent_class)->draw (widget, cr);
-
-  return TRUE;
+  return GTK_WIDGET_CLASS (gtk_icon_view_parent_class)->draw (widget, cr);
 }
 
 static gboolean
--
cgit v0.9.0.2
