From f86351c05f88d3d7911615fc27ddd8091260caf3 Mon Sep 17 00:00:00 2001
From: Ryan Lortie <desrt@desrt.ca>
Date: Wed, 14 Sep 2011 18:02:51 +0000
Subject: GtkApplication: chain up ::startup

---
diff --git a/gtk/gtkapplication.c b/gtk/gtkapplication.c
index c6db4e8..c192d75 100644
--- a/gtk/gtkapplication.c
+++ b/gtk/gtkapplication.c
@@ -97,6 +97,9 @@ gtk_application_focus_in_event_cb (GtkWindow      *window,
 static void
 gtk_application_startup (GApplication *application)
 {
+  G_APPLICATION_CLASS (gtk_application_parent_class)
+    ->startup (application);
+
   gtk_init (0, 0);
 }
 
--
cgit v0.9.0.2
