diff --git a/Makefile.am b/Makefile.am
new file mode 100644
index 0000000..dff484a
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,31 @@
+DISTCHECK_CONFIGURE_FLAGS = --enable-introspection --enable-gtk-doc
+
+-include $(INTROSPECTION_MAKEFILE)
+INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir)
+INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
+
+SUBDIRS = po dbind xml atspi bus registryd doc test
+
+ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
+
+EXTRA_DIST = \
+	atspi-2.pc.in \
+	meson.build \
+	meson_options.txt
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = atspi-2.pc
+
+DISTCLEANFILES = \
+	atspi-2.pc
+
+MAINTAINERCLEANFILES = \
+	$(GITIGNORE_MAINTAINERCLEANFILES_TOPLEVEL) \
+	$(GITIGNORE_MAINTAINERCLEANFILES_MAKEFILE_IN) \
+	$(GITIGNORE_MAINTAINERCLEANFILES_M4_LIBTOOL) \
+	gtk-doc.make \
+	m4/gtk-doc.m4 \
+	m4/intltool.m4
+
+-include $(top_srcdir)/git.mk
diff --git a/atspi/Makefile.am b/atspi/Makefile.am
new file mode 100644
index 0000000..636cc76
--- /dev/null
+++ b/atspi/Makefile.am
@@ -0,0 +1,160 @@
+lib_LTLIBRARIES =
+
+BUILT_SOURCES =
+
+CLEANFILES =
+DISTCLEANFILES =
+EXTRA_DIST = \
+	meson.build
+
+lib_LTLIBRARIES += libatspi.la
+
+libatspi_la_LDFLAGS = \
+	@LDFLAGS@ @LT_VERSION_INFO@ @LIBTOOL_EXPORT_OPTIONS@ \
+	-no-undefined
+
+libatspi_la_CFLAGS = \
+	$(DBUS_CFLAGS) \
+	$(DBIND_CFLAGS) \
+	$(GLIB_CFLAGS) \
+	-I$(top_srcdir)/registryd \
+	-I$(top_builddir)/registryd \
+	-I$(top_builddir) \
+	-I$(top_srcdir)
+
+libatspi_la_LIBADD = \
+	$(DBUS_LIBS) \
+	$(GOBJ_LIBS) \
+	$(X11_LIBS) \
+	$(top_builddir)/dbind/libdbind.la
+
+libatspi_la_DEPENDENCIES = $(top_builddir)/dbind/libdbind.la
+
+atspi_headers = \
+	atspi.h	\
+	atspi-accessible.h \
+	atspi-action.h \
+	atspi-application.h \
+	atspi-collection.h \
+	atspi-component.h \
+	atspi-constants.h \
+	atspi-device-listener.h \
+	atspi-device-listener-private.h \
+	atspi-document.h \
+	atspi-editabletext.h \
+	atspi-event-listener.h \
+	atspi-event-listener-private.h \
+	atspi-gmain.h \
+	atspi-hyperlink.h \
+	atspi-hypertext.h \
+	atspi-image.h \
+	atspi-matchrule.h \
+	atspi-misc.h \
+	atspi-object.h \
+	atspi-private.h \
+	atspi-registry.h \
+	atspi-relation.h \
+	atspi-selection.h \
+	atspi-stateset.h \
+	atspi-table.h \
+	atspi-table-cell.h \
+	atspi-text.h \
+	atspi-types.h \
+	atspi-value.h
+
+libatspi_la_SOURCES = \
+	$(BUILT_SOURCES) \
+	atspi-accessible.c \
+	atspi-accessible-private.h \
+	atspi-action.c \
+	atspi-application.c \
+	atspi-collection.c \
+	atspi-component.c \
+	atspi-device-listener.c \
+	atspi-device-listener-private.h \
+	atspi-document.c \
+	atspi-editabletext.c \
+	atspi-event-listener.c \
+	atspi-event-listener-private.h \
+	atspi-gmain.c \
+	atspi-hyperlink.c \
+	atspi-hypertext.c \
+	atspi-image.c \
+	atspi-matchrule.c \
+	atspi-matchrule-private.h \
+	atspi-misc.c \
+	atspi-misc-private.h \
+	atspi-object.c \
+	atspi-private.h \
+	atspi-registry.c \
+	atspi-relation.c \
+	atspi-selection.c \
+	atspi-stateset.c \
+	atspi-table.c \
+	atspi-table-cell.c \
+	atspi-text.c \
+	atspi-value.c \
+	$()
+
+BUILT_SOURCES += \
+	atspi-enum-types.c \
+	atspi-enum-types.h
+
+libatspiincludedir = $(includedir)/at-spi-2.0/atspi
+libatspiinclude_HEADERS = $(atspi_headers) atspi-enum-types.h
+
+#CLEANFILES = atspi-constants.h
+
+#atspi-constants.h: $(top_srcdir)/xml/spec.xml $(top_srcdir)/tools/c-constants-gen.py
+#	python $(top_srcdir)/tools/c-constants-gen.py Atspi $(top_srcdir)/xml/spec.xml atspi-constants
+
+ENUM_TYPES = \
+	atspi-constants.h \
+	atspi-types.h
+
+atspi-enum-types.h: atspi-enum-types.h.template $(ENUM_TYPES)
+	$(AM_V_GEN) $(GLIB_MKENUMS) --template $(srcdir)/atspi-enum-types.h.template $(addprefix $(srcdir)/, $(ENUM_TYPES)) > $@.tmp \
+	&& (cmp -s $@.tmp $@ || cp -f $@.tmp $@) \
+	&& rm -f $@.tmp
+
+atspi-enum-types.c: atspi-enum-types.c.template $(ENUM_TYPES)
+	$(AM_V_GEN) $(GLIB_MKENUMS) --template $(srcdir)/atspi-enum-types.c.template $(addprefix $(srcdir)/, $(ENUM_TYPES)) > $@.tmp \
+	&& (cmp -s $@.tmp $@ || cp -f $@.tmp $@) \
+	&& rm -f $@.tmp
+
+CLEANFILES += \
+	atspi-enum-types.c \
+	atspi-enum-types.h
+
+EXTRA_DIST += \
+	atspi-enum-types.c.template \
+	atspi-enum-types.h.template
+
+-include $(INTROSPECTION_MAKEFILE)
+INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir) --warn-all
+INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
+
+if HAVE_INTROSPECTION
+Atspi-2.0.gir: libatspi.la
+Atspi_2_0_gir_INCLUDES = DBus-1.0 GLib-2.0 GObject-2.0
+Atspi_2_0_gir_EXPORT_PACKAGES = atspi-2
+Atspi_2_0_gir_CFLAGS = -I$(top_builddir) -I$(top_srcdir)
+Atspi_2_0_gir_LIBS = libatspi.la $(top_builddir)/dbind/libdbind.la
+Atspi_2_0_gir_FILES = $(libatspi_la_SOURCES) $(atspi_headers) atspi-enum-types.c atspi-enum-types.h
+Atspi_2_0_gir_SCANNERFLAGS = --namespace Atspi --nsversion=2.0
+INTROSPECTION_SCANNER_ARGS += --c-include='atspi/atspi.h'
+INTROSPECTION_GIRS += Atspi-2.0.gir
+
+girdir = $(datadir)/gir-1.0
+gir_DATA = $(INTROSPECTION_GIRS)
+
+typelibdir = $(libdir)/girepository-1.0
+typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
+
+CLEANFILES += \
+    $(gir_DATA) \
+    $(typelib_DATA)
+endif
+
+-include $(top_srcdir)/git.mk
diff --git a/autogen.sh b/autogen.sh
new file mode 100755
index 0000000..4576400
--- /dev/null
+++ b/autogen.sh
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+test -n "$srcdir" || srcdir=`dirname "$0"`
+test -n "$srcdir" || srcdir=.
+
+olddir=`pwd`
+cd "$srcdir"
+
+# run gtkdocize
+gtkdocize || exit 1
+
+# gnome-autogen.sh runs configure, so do likewise.
+autoreconf -vif
+
+cd "$olddir"
+
+test -n "$NOCONFIGURE" || "$srcdir/configure" "$@"
+
diff --git a/bus/Makefile.am b/bus/Makefile.am
new file mode 100644
index 0000000..688f82a
--- /dev/null
+++ b/bus/Makefile.am
@@ -0,0 +1,48 @@
+CLEANFILES =
+DISTCLEANFILES =
+EXTRA_DIST = \
+	meson.build
+
+libexec_PROGRAMS = at-spi-bus-launcher
+at_spi_bus_launcher_SOURCES = at-spi-bus-launcher.c
+at_spi_bus_launcher_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\" \
+                               -DDATADIR=\"$(datadir)\" \
+                               -DDBUS_DAEMON=\"$(DBUS_DAEMON)\"
+at_spi_bus_launcher_CFLAGS = $(GIO_CFLAGS)
+at_spi_bus_launcher_LDADD = $(GIO_LIBS) $(X11_LIBS)
+
+substitutions = "s,@libexecdir[@],$(libexecdir),"
+
+default_sessiondir = $(sysconfdir)/xdg/autostart
+default_session_DATA = at-spi-dbus-bus.desktop
+at-spi-dbus-bus.desktop: at-spi-dbus-bus.desktop.in
+	$(AM_V_GEN) $(SED) -e$ $(substitutions)  $< > $@.tmp && mv $@.tmp $@
+
+EXTRA_DIST += at-spi-dbus-bus.desktop.in
+CLEANFILES += at-spi-dbus-bus.desktop
+
+dbusservicedir = $(DBUS_SERVICES_DIR)
+dbusservice_DATA = org.a11y.Bus.service
+org.a11y.Bus.service: org.a11y.Bus.service.in
+	$(AM_V_GEN) $(SED) -e $(substitutions) $< > $@.tmp && mv $@.tmp $@
+
+EXTRA_DIST += org.a11y.Bus.service.in
+CLEANFILES += org.a11y.Bus.service
+
+systemd_userdir = $(prefix)/lib/systemd/user
+systemd_user_DATA = at-spi-dbus-bus.service
+at-spi-dbus-bus.service: at-spi-dbus-bus.service.in Makefile
+	$(AM_V_GEN) $(SED) -e $(substitutions) $< > $@.tmp && mv $@.tmp $@
+
+EXTRA_DIST += at-spi-dbus-bus.service.in
+CLEANFILES += at-spi-dbus-bus.service
+
+busconfigdir = $(datadir)/defaults/at-spi2
+busconfig_DATA = accessibility.conf
+accessibility.conf: accessibility.conf.in
+	$(AM_V_GEN) $(SED) -e "s|[@]DATADIR[@]|$(datadir)|" $(srcdir)/$@.in > $@
+
+EXTRA_DIST += accessibility.conf.in
+CLEANFILES += accessibility.conf
+
+-include $(top_srcdir)/git.mk
diff --git a/configure.ac b/configure.ac
new file mode 100644
index 0000000..a84dcbd
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,188 @@
+AC_INIT([at-spi2-core], [2.28.0], [accessibility-atspi@lists.linux-foundation.org])
+AC_PREREQ([2.59])
+AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_MACRO_DIR([m4])
+
+AT_SPI_CORE_MAJOR_VERSION=0
+AT_SPI_CORE_MINOR_VERSION=1
+AT_SPI_CORE_MICRO_VERSION=3
+AT_SPI_CORE_VERSION="$AT_SPI_CORE_MAJOR_VERSION.$AT_SPI_CORE_MINOR_VERSION"
+AC_SUBST(AT_SPI_CORE_VERSION)
+
+LT_RELEASE=$AT_SPI_CORE_VERSION
+LT_CURRENT=0
+LT_REVISION=1
+LT_AGE=0
+LT_VERSION_INFO='-version-info ${LT_CURRENT}:${LT_REVISION}:${LT_AGE}'
+AC_SUBST(LT_VERSION_INFO)
+AC_SUBST(LT_RELEASE)
+AC_SUBST(LT_CURRENT)
+AC_SUBST(LT_REVISION)
+AC_SUBST(LT_AGE)
+
+AM_INIT_AUTOMAKE([1.13 -Wall -Wno-portability foreign subdir-objects no-dist-gzip dist-xz])
+
+# Support silent build rules. Disable
+# by either passing --disable-silent-rules to configure or passing V=1
+# to make
+AM_SILENT_RULES([yes])
+
+AM_PROG_CC_C_O
+LT_INIT([disable-static])
+PKG_PROG_PKG_CONFIG
+
+AC_CONFIG_HEADERS([config.h])
+
+AC_PROG_SED
+
+AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT([external])
+
+GETTEXT_PACKAGE="${PACKAGE}"
+AC_SUBST(GETTEXT_PACKAGE)
+AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],
+                   ["$GETTEXT_PACKAGE"],
+                   [The prefix for our gettext translation domains.])
+
+PKG_CHECK_MODULES(DBUS, [dbus-1 >= 1.0])
+AC_SUBST(DBUS_LIBS)
+AC_SUBST(DBUS_CFLAGS)
+
+PKG_CHECK_MODULES(GLIB, [glib-2.0 >= 2.36.0])
+AC_SUBST(GLIB_LIBS)
+AC_SUBST(GLIB_CFLAGS)
+
+# Use pkg-config to find the glib-mkenums script
+GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
+AC_SUBST(GLIB_MKENUMS)
+
+PKG_CHECK_MODULES(GOBJ, [gobject-2.0 >= 2.0.0])
+AC_SUBST(GOBJ_LIBS)
+AC_SUBST(GOBJ_CFLAGS)
+
+PKG_CHECK_MODULES(GIO, [gio-2.0 >= 2.28])
+AC_SUBST(GIO_LIBS)
+AC_SUBST(GIO_CFLAGS)
+
+# --------------------------------------------------------------------
+# Find DL functionality
+
+AC_CHECK_LIB(c, dlopen, DL_LIBS="",
+        AC_CHECK_LIB(dl, dlopen, DL_LIBS="-ldl",
+                AC_MSG_ERROR([Could not find a library with the dlopen function])
+        )
+)
+
+AC_SUBST(DL_LIBS)
+
+AC_ARG_ENABLE(x11,
+              [AS_HELP_STRING([--disable-x11],
+                              [disable the X11 backend])],,
+			      [enable_x11=auto])
+
+AS_IF([test "x$enable_x11" = xno], [
+  AM_CONDITIONAL(USE_X11, false)
+], [
+  PKG_CHECK_MODULES(X11, [x11], [
+    AC_DEFINE(HAVE_X11, 1, [Define to use X11])
+    AC_SUBST(HAVE_X11)
+    have_x11=yes
+  ], [
+    AS_IF([test "x$enable_x11" = xyes], [
+      AC_MSG_ERROR([Unable to find X11 development files])
+    ])
+    have_x11=no
+  ])
+
+  # Check for additional X11 extensions
+  AS_IF([test "x$have_x11" = xyes], [
+    # XTest (mandatory)
+    PKG_CHECK_MODULES(XTST, [xtst])
+    X11_CFLAGS="$X11_CFLAGS $XTST_CFLAGS"
+    X11_LIBS="$X11_LIBS $XTST_LIBS"
+
+    # XInput (mandatory)
+    PKG_CHECK_MODULES(XINPUT, [xi])
+    X11_CFLAGS="$X11_CFLAGS $XINPUT_CFLAGS"
+    X11_LIBS="$X11_LIBS $XINPUT_LIBS"
+
+    # XKB (optional)
+    AC_CHECK_LIB(X11, XkbGetMap, [
+      AC_DEFINE(HAVE_XKB, 1, [Define to use XKB])
+    ], [:])
+  ])
+
+  AC_SUBST(X11_CFLAGS)
+  AC_SUBST(X11_LIBS)
+
+  AM_CONDITIONAL(USE_X11, test "x$have_x11" = xyes)
+])
+
+AC_ARG_WITH([dbus_daemondir],
+	[AS_HELP_STRING([--with-dbus-daemondir=<directory>],
+		[Directory where the DBUS daemon is installed])],
+    [DBUS_DAEMON="$with_dbus_daemondir/dbus-daemon"],
+    [AC_PATH_PROG([DBUS_DAEMON], [dbus-daemon], [XXXDBUSDAEMONNOTFOUND],
+		[$PATH$PATH_SEPARATOR/sbin$PATH_SEPARATOR/usr/sbin$PATH_SEPARATOR]dnl
+[/libexec$PATH_SEPARATOR/usr/libexec$PATH_SEPARATOR/usr/local/bin]dnl
+[$PATH_SEPARATOR/usr/pkg/bin])]
+	)
+if test "$DBUS_DAEMON" = "XXXDBUSDAEMONNOTFOUND"; then
+	DBUS_DAEMON = "$bindir/dbus-daemon"
+	AC_MSG_WARN([at-spi2 relies on dbus, and the dbus-daemon was not found])
+	AC_MSG_WARN([we assume that it will be installed in "$bindir"])
+fi
+AC_SUBST(DBUS_DAEMON)
+
+AC_ARG_WITH(dbus-services,
+           [AS_HELP_STRING([--with-dbus-services=<directory>],
+           [where D-BUS services directory is])])
+if ! test -z "$with_dbus_services" ; then
+           DBUS_SERVICES_DIR="$with_dbus_services"
+else
+           DBUS_SERVICES_DIR="$datadir/dbus-1/services"
+fi
+AC_SUBST(DBUS_SERVICES_DIR)
+
+GOBJECT_INTROSPECTION_CHECK([1.32.0])
+
+AC_SUBST(LIBTOOL_EXPORT_OPTIONS)
+
+dnl find sizes & alignments
+orig_CPPFLAGS=$CPPFLAGS
+CPPFLAGS="$CPPFLAGS $DBUS_CFLAGS"
+AC_CHECK_ALIGNOF(char)
+AC_CHECK_ALIGNOF(dbus_bool_t, [#include <dbus/dbus.h>])
+AC_CHECK_ALIGNOF(dbus_int16_t, [#include <dbus/dbus.h>])
+AC_CHECK_ALIGNOF(dbus_int32_t, [#include <dbus/dbus.h>])
+AC_CHECK_ALIGNOF(dbus_int64_t, [#include <dbus/dbus.h>])
+AC_CHECK_ALIGNOF(double, [#include <dbus/dbus.h>])
+AC_CHECK_ALIGNOF(dbind_pointer, [AC_INCLUDES_DEFAULT
+typedef void *dbind_pointer;])
+AC_CHECK_ALIGNOF(dbind_struct, [AC_INCLUDES_DEFAULT
+typedef struct {char s1;} dbind_struct;])
+
+GTK_DOC_CHECK([1.25])
+
+if test "x$GCC" = xyes; then
+	CFLAGS="$CFLAGS -Werror-implicit-function-declaration"
+fi
+
+CPPFLAGS=$orig_CPPFLAGS 
+
+AC_CONFIG_FILES([
+        Makefile
+        po/Makefile.in
+	xml/Makefile
+	dbind/Makefile
+	atspi/Makefile
+	registryd/Makefile
+	bus/Makefile
+        doc/Makefile
+        doc/libatspi/Makefile
+        doc/libatspi/version.xml
+        test/Makefile
+        atspi-2.pc
+])
+
+AC_OUTPUT
diff --git a/dbind/Makefile.am b/dbind/Makefile.am
new file mode 100644
index 0000000..6594e3e
--- /dev/null
+++ b/dbind/Makefile.am
@@ -0,0 +1,27 @@
+noinst_LTLIBRARIES = libdbind.la
+
+AM_CPPFLAGS = \
+	-DG_LOG_DOMAIN=\"dbind\" \
+	-I$(top_srcdir) \
+	$(WARN_CFLAGS) \
+	$(DBUS_CFLAGS) \
+	$(GLIB_CFLAGS)
+
+libdbind_la_SOURCES = \
+        dbind.h \
+	dbind.c \
+        dbind-any.h \
+	dbind-any.c
+libdbind_la_LIBADD = $(DBUS_LIBS) $(GLIB_LIBS)
+
+TESTS = dbtest
+
+check_PROGRAMS = dbtest
+dbtest_SOURCES = dbtest.c ../atspi/atspi-gmain.c
+dbtest_LDFLAGS = libdbind.la
+dbtest_LDADD = $(DBUS_LIBS)
+
+EXTRA_DIST = \
+	meson.build
+
+-include $(top_srcdir)/git.mk
diff --git a/doc/Makefile.am b/doc/Makefile.am
new file mode 100644
index 0000000..b478520
--- /dev/null
+++ b/doc/Makefile.am
@@ -0,0 +1,4 @@
+
+SUBDIRS=libatspi
+
+-include $(top_srcdir)/git.mk
diff --git a/po/Makevars b/po/Makevars
new file mode 100644
index 0000000..b40107f
--- /dev/null
+++ b/po/Makevars
@@ -0,0 +1,78 @@
+# Makefile variables for PO directory in any package using GNU gettext.
+
+# Usually the message domain is the same as the package name.
+DOMAIN = $(PACKAGE)
+
+# These two variables depend on the location of this directory.
+subdir = po
+top_builddir = ..
+
+# These options get passed to xgettext.
+XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --add-comments
+
+# This is the copyright holder that gets inserted into the header of the
+# $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
+# package.  (Note that the msgstr strings, extracted from the package's
+# sources, belong to the copyright holder of the package.)  Translators are
+# expected to transfer the copyright for their translations to this person
+# or entity, or to disclaim their copyright.  The empty string stands for
+# the public domain; in this case the translators are expected to disclaim
+# their copyright.
+COPYRIGHT_HOLDER = at-spi2-core authors
+
+# This tells whether or not to prepend "GNU " prefix to the package
+# name that gets inserted into the header of the $(DOMAIN).pot file.
+# Possible values are "yes", "no", or empty.  If it is empty, try to
+# detect it automatically by scanning the files in $(top_srcdir) for
+# "GNU packagename" string.
+PACKAGE_GNU = no
+
+# This is the email address or URL to which the translators shall report
+# bugs in the untranslated strings:
+# - Strings which are not entire sentences, see the maintainer guidelines
+#   in the GNU gettext documentation, section 'Preparing Strings'.
+# - Strings which use unclear terms or require additional context to be
+#   understood.
+# - Strings which make invalid assumptions about notation of date, time or
+#   money.
+# - Pluralisation problems.
+# - Incorrect English spelling.
+# - Incorrect formatting.
+# It can be your email address, or a mailing list address where translators
+# can write to without being subscribed, or the URL of a web page through
+# which the translators can contact you.
+MSGID_BUGS_ADDRESS = https://bugzilla.gnome.org/enter_bug.cgi?product=at-spi&keywords=I18N+L10N&component=at-spi2-core
+
+# This is the list of locale categories, beyond LC_MESSAGES, for which the
+# message catalogs shall be used.  It is usually empty.
+EXTRA_LOCALE_CATEGORIES =
+
+# This tells whether the $(DOMAIN).pot file contains messages with an 'msgctxt'
+# context.  Possible values are "yes" and "no".  Set this to yes if the
+# package uses functions taking also a message context, like pgettext(), or
+# if in $(XGETTEXT_OPTIONS) you define keywords with a context argument.
+USE_MSGCTXT = no
+
+# These options get passed to msgmerge.
+# Useful options are in particular:
+#   --previous            to keep previous msgids of translated messages,
+#   --quiet               to reduce the verbosity.
+MSGMERGE_OPTIONS =
+
+# These options get passed to msginit.
+# If you want to disable line wrapping when writing PO files, add
+# --no-wrap to MSGMERGE_OPTIONS, XGETTEXT_OPTIONS, and
+# MSGINIT_OPTIONS.
+MSGINIT_OPTIONS =
+
+# This tells whether or not to regenerate a PO file when $(DOMAIN).pot
+# has changed.  Possible values are "yes" and "no".  Set this to no if
+# the POT file is checked in the repository and the version control
+# program ignores timestamps.
+PO_DEPENDS_ON_POT = no
+
+# This tells whether or not to forcibly update $(DOMAIN).pot and
+# regenerate PO files on "make dist".  Possible values are "yes" and
+# "no".  Set this to no if the POT file and PO files are maintained
+# externally.
+DIST_DEPENDS_ON_UPDATE_PO = no
diff --git a/registryd/Makefile.am b/registryd/Makefile.am
new file mode 100644
index 0000000..6a4b1d8
--- /dev/null
+++ b/registryd/Makefile.am
@@ -0,0 +1,68 @@
+EXTRA_DIST = \
+	meson.build
+
+libexec_PROGRAMS = at-spi2-registryd
+
+at_spi2_registryd_CFLAGS =	\
+	$(GLIB_CFLAGS)		\
+	$(GIO_CFLAGS)		\
+	$(DBUS_CFLAGS)		\
+	$(GOBJ_CFLAGS)		\
+	$(ATK_CFLAGS)		\
+	$(X11_CFLAGS)		\
+	-I$(top_srcdir)		\
+	-I$(top_builddir)	\
+	-I$(top_builddir)/atspi	\
+	-DATSPI_INTROSPECTION_PATH=\"$(pkgdatadir)/$(DEFAULT_ATSPI_INTROSPECTION_PATH)\"
+
+at_spi2_registryd_LDADD =	\
+	../atspi/libatspi.la \
+	$(GLIB_LIBS)		\
+	$(GIO_LIBS)		\
+	$(DBUS_LIBS)	\
+	$(GOBJ_CFLAGS)		\
+	$(ATK_LIBS)		\
+	$(X11_LIBS)		\
+	$(DL_LIBS)
+
+at_spi2_registryd_SOURCES =	\
+	de-marshaller.h         \
+	de-marshaller.c         \
+	de-types.h              \
+	keymasks.h 		\
+	paths.h                 \
+	registry-main.c		\
+	registry.c		\
+	registry.h		\
+	introspection.h         \
+	introspection.c         \
+	deviceeventcontroller.c	\
+	deviceeventcontroller.h	\
+	reentrant-list.c	\
+	reentrant-list.h
+
+X11_SOURCES = \
+	deviceeventcontroller-x11.c \
+	display.h \
+	display.c \
+	event-source.c \
+	event-source.h \
+	ucs2keysym.c
+
+if USE_X11
+at_spi2_registryd_SOURCES += $(X11_SOURCES)
+else
+EXTRA_DIST += $(X11_SOURCES)
+endif
+
+servicedir=$(datadir)/dbus-1/accessibility-services
+service_in_files = org.a11y.atspi.Registry.service.in
+service_DATA     = $(service_in_files:.service.in=.service)
+
+$(service_DATA): $(service_in_files) Makefile
+	$(AM_V_GEN)$(SED) -e "s|[@]libexecdir[@]|$(libexecdir)|" $(srcdir)/$@.in > $@
+
+DISTCLEANFILES = org.a11y.atspi.Registry.service
+EXTRA_DIST += org.a11y.atspi.Registry.service.in
+
+-include $(top_srcdir)/git.mk
diff --git a/test/Makefile.am b/test/Makefile.am
new file mode 100644
index 0000000..b76db0a
--- /dev/null
+++ b/test/Makefile.am
@@ -0,0 +1,10 @@
+LDADD = $(top_builddir)/atspi/libatspi.la
+noinst_PROGRAMS = memory
+memory_SOURCES = memory.c
+memory_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir) -I$(top_builddir)/atspi
+memory_CFLAGS = $(GLIB_CFLAGS) 	$(GOBJ_LIBS) $(DBUS_CFLAGS)
+memory_LDFLAGS = 
+
+-include $(top_srcdir)/git.mk
+
+EXTRA_DIST = meson.build
diff --git a/xml/Makefile.am b/xml/Makefile.am
new file mode 100644
index 0000000..edb71d3
--- /dev/null
+++ b/xml/Makefile.am
@@ -0,0 +1,41 @@
+EXTRA_DIST = \
+	Accessibility.xml \
+	Accessible.xml \
+	Action.xml \
+	Application.xml \
+	Cache.xml \
+	Collection.xml \
+	Component.xml \
+	create-introspection.sh \
+	DeviceEventController.xml \
+	DeviceEventListener.xml \
+	Document.xml \
+	EditableText.xml \
+	Event.xml \
+	Hyperlink.xml \
+	Hypertext.xml \
+	Image.xml \
+	Registry.xml \
+	remove-annotations.xsl \
+	Selection.xml \
+	Socket.xml \
+	Table.xml \
+	Text.xml \
+	Value.xml \
+	versioned-introspection.py
+
+CLEANFILES = \
+	introspection.c \
+	introspection.h \
+	Processed.xml
+
+#BUILT_SOURCES = spec.xml
+#CLEANFILES += spec.xml
+
+XML_SPEC = $(filter %.xml,$(EXTRA_DIST))
+
+spec.xml: $(XML_SPEC)
+	xsltproc --xinclude $(top_srcdir)/tools/identity.xsl Accessibility.xml >spec.xml
+
+
+-include $(top_srcdir)/git.mk
