From a1fcea21a879596976e5bae444894a98468a207f Mon Sep 17 00:00:00 2001
From: Alex Acevedo <alex-acevedo@protonmail.com>
Date: Tue, 25 Apr 2017 16:33:50 -0400
Subject: [PATCH] remove post-install script from meson build

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index da4b159..c1c336e 100644
--- a/meson.build
+++ b/meson.build
@@ -37,7 +37,6 @@ am_cflags = [
 ]
 
 add_global_arguments(am_cflags, language: 'c')
-meson.add_install_script('meson_post_install.sh')
 
 # Budgie needs a minimum 3.18 GNOME stack
 gnome_minimum_version = '>= 3.18.0'
-- 
2.10.2
