From faadc51fb554b90cfd07d108262383e1772f279b Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 14 Jul 2010 00:47:18 +0000
Subject: Complete the GTK+ 3 port by using unique-3.0

---
diff --git a/configure.ac b/configure.ac
index 7622885..637c4d4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -115,7 +115,7 @@ if test "x$enable_pulseaudio" != "xno"; then
             gio-2.0
             gconf-2.0 >= $GCONF_REQUIRED_VERSION
             libcanberra-gtk3 >= $CANBERRA_REQUIRED_VERSION
-            unique-1.0
+            unique-3.0
             libxml-2.0
     )
     AC_SUBST(VOLUME_CONTROL_CFLAGS)
--
cgit v0.8.3.1
