diff -Naur gnome-media-2.25.5/configure.ac gnome-media-2.25.5.new/configure.ac
--- gnome-media-2.25.5/configure.ac	2009-01-21 01:32:53.000000000 +0530
+++ gnome-media-2.25.5.new/configure.ac	2009-02-12 11:04:13.336933629 +0530
@@ -117,6 +117,8 @@
   if test "x$have_pulseaudio" = "xyes" ; then
     AC_DEFINE(HAVE_PULSEAUDIO, [], [Define if we have pulseaudio])
   fi
+else
+  have_pulseaudio=no
 fi
 AM_CONDITIONAL(HAVE_PULSEAUDIO, test x$have_pulseaudio = xyes)
 AC_SUBST(HAVE_PULSEAUDIO)
