Index: colord/data/org.freedesktop.ColorManager.conf.in
===================================================================
--- colord.orig/data/org.freedesktop.ColorManager.conf.in	2011-09-01 17:59:58.330626765 +1000
+++ colord/data/org.freedesktop.ColorManager.conf.in	2011-09-01 18:02:25.410628315 +1000
@@ -16,8 +16,8 @@
     <allow own="org.freedesktop.ColorManager"/>
   </policy>
 
- <!-- Allow anyone to call into the service - we'll reject callers using PolicyKit -->
-  <policy context="default">
+ <!-- Use at_console policy rather than PolicyKit to allow running as colord user -->
+  <policy at_console="true">
     <allow send_destination="org.freedesktop.ColorManager"
            send_interface="org.freedesktop.ColorManager"/>
     <allow send_destination="org.freedesktop.ColorManager"
@@ -32,5 +32,20 @@
            send_interface="org.freedesktop.DBus.Introspectable"/>
   </policy>
 
+  <!-- Also allow root to access the API; cupsd needs to talk to colord -->
+  <policy user="root">
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.ColorManager"/>
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.ColorManager.Profile"/>
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.ColorManager.Device"/>
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.ColorManager.Sensor"/>
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.DBus.Properties"/>
+    <allow send_destination="org.freedesktop.ColorManager"
+           send_interface="org.freedesktop.DBus.Introspectable"/>
+  </policy>
 </busconfig>
 
