diff -ur a/util/rarian-sk-update.in b/util/rarian-sk-update.in
--- a/util/rarian-sk-update.in	2007-09-01 18:22:35 +0100
+++ b/util/rarian-sk-update.in	2007-10-18 15:27:42 +0100
@@ -258,10 +258,7 @@
 # They are inherited from scrollkeeper :(
 
 # We use TEMP as set -- seems to nuke the return value of getopt
-TEMP=`getopt -u -n$(basename $0) -o "o:r:p:vqnhV" \
-    --longoptions="clean-index,omf-dir:,resulting-dir:,help,version" -- "$@"` \
-    || print_usage
-
+TEMP=`getopt "o:r:p:cvqnhV" "$@"`
 if [ $? != 0 ] ; then
 	print_usage
 	exit 0
@@ -271,12 +268,12 @@
 
 while true; do
     case "$1" in
-		-o | --omf-dir )
+		-o )
 			convert_dir=$2
 			update_output_dir=1
 			shift 2
 			;;
-		-r | --result-dir )
+		-r )
 			output_dir=$2
 			overload_update=1
 			shift 2
@@ -288,7 +285,7 @@
 		-q )
 			shift
 			;;
-        --clean-index )
+        -c )
         	clean_index=1
 			shift
             ;;	      
@@ -299,9 +296,9 @@
 			# Scrollkeeper compat.  Actually do nothing
 			shift
 			;;
-	        -p )
-	                statedir=$2
-	                shift 2
+	    -p )
+	        statedir=$2
+	        shift 2
 			;;
 		-V | --version )
 			print_version
@@ -416,4 +413,4 @@
 fi
 rm -rf $tmpdir
 
-fi # ENABLE_OMF_READ
\ No newline at end of file
+fi # ENABLE_OMF_READ
