Index: totem-plparser.pc.in
===================================================================
--- totem-plparser.pc.in	(revision 127)
+++ totem-plparser.pc.in	(working copy)
@@ -9,6 +9,6 @@
 Description: Totem Playlist Parser library
 Version: @VERSION@
 Requires: gtk+-2.0
-Requires.private: libxml-2.0 gnome-vfs-2.0 gnome-vfs-module-2.0 camel-1.2
+Requires.private: libxml-2.0 camel-1.2
 Libs: -L${libdir} -ltotem-plparser
 Cflags: -I${includedir}/totem-pl-parser/1/plparser
Index: configure.in
===================================================================
--- configure.in	(revision 127)
+++ configure.in	(working copy)
@@ -76,9 +76,9 @@
 AM_CONDITIONAL(HAVE_HAL, test x"$with_hal" = xyes)
 
 dnl Check for packages for building libtotem-plparser.la
-PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gtk+-2.0 libxml-2.0 gnome-vfs-module-2.0 camel-1.2 gio-2.0])
+PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gtk+-2.0 libxml-2.0 camel-1.2 gio-2.0])
 
-PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI], [gnome-vfs-2.0 gnome-vfs-module-2.0 gio-2.0])
+PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI], [gio-2.0])
 
 GNOME_COMMON_INIT
 GNOME_DEBUG_CHECK
Index: totem-plparser-mini-uninstalled.pc.in
===================================================================
--- totem-plparser-mini-uninstalled.pc.in	(revision 127)
+++ totem-plparser-mini-uninstalled.pc.in	(working copy)
@@ -8,6 +8,5 @@
 Name: totem-plparser
 Description: Totem Playlist Parser library
 Version: @VERSION@
-Requires.private: gnome-vfs-2.0 gnome-vfs-module-2.0
 Libs: ${pc_top_builddir}/${pcfiledir}/plparse/libtotem-plparser-mini.la
 Cflags: -I${pc_top_builddir}/${pcfiledir}/plparse
Index: totem-plparser-uninstalled.pc.in
===================================================================
--- totem-plparser-uninstalled.pc.in	(revision 127)
+++ totem-plparser-uninstalled.pc.in	(working copy)
@@ -9,6 +9,6 @@
 Description: Totem Playlist Parser library
 Version: @VERSION@
 Requires: gtk+-2.0
-Requires.private: libxml-2.0 gnome-vfs-2.0 gnome-vfs-module-2.0 camel-1.2
+Requires.private: libxml-2.0 camel-1.2
 Libs: ${pc_top_builddir}/${pcfiledir}/plparse/libtotem-plparser.la
 Cflags: -I${pc_top_builddir}/${pcfiledir}/plparse
Index: plparse/totem-pl-parser.c
===================================================================
--- plparse/totem-pl-parser.c	(revision 127)
+++ plparse/totem-pl-parser.c	(working copy)
@@ -754,7 +754,7 @@
 
 /**
  * totem_pl_parser_write_buffer:
- * @handle: a #GnomeVFSHandle to an open file
+ * @stream: a #GFileOutputStream to an open file
  * @buf: the string buffer to write out
  * @len: the length of the string to write out
  * @error: return location for a #GError, or %NULL
Index: totem-plparser-mini.pc.in
===================================================================
--- totem-plparser-mini.pc.in	(revision 127)
+++ totem-plparser-mini.pc.in	(working copy)
@@ -8,6 +8,5 @@
 Name: totem-plparser-mini
 Description: Totem Playlist Parser library, mini version
 Version: @VERSION@
-Requires.private: gnome-vfs-2.0 gnome-vfs-module-2.0
 Libs: -L${libdir} -ltotem-plparser-mini
 Cflags: -I${includedir}/totem-pl-parser/1/plparser
