diff -Naur totem-pl-parser-2.26.1/docs/reference/Makefile.am totem-pl-parser-2.26.1.new/docs/reference/Makefile.am
--- totem-pl-parser-2.26.1/docs/reference/Makefile.am	2009-03-23 20:03:17.000000000 +0530
+++ totem-pl-parser-2.26.1.new/docs/reference/Makefile.am	2009-05-05 17:23:56.184865429 +0530
@@ -96,5 +96,7 @@
 # e.g. EXTRA_DIST += version.xml.in
 EXTRA_DIST += version.xml.in
 
+if ENABLE_GTK_DOC
 TESTS_ENVIRONMENT = cd $(srcdir) && 
 TESTS = $(GTKDOC_CHECK)
+endif
