diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN swfdec-gnome-2.22.2.orig/configure.ac swfdec-gnome-2.22.2/configure.ac
--- swfdec-gnome-2.22.2.orig/configure.ac	2008-04-07 15:10:42.000000000 -0400
+++ swfdec-gnome-2.22.2/configure.ac	2008-07-25 11:36:01.000000000 -0400
@@ -42,9 +42,9 @@ AM_GLIB_GNU_GETTEXT
 
 dnl Required versions
 
-SWFDEC_MAJORMINOR=0.6
+SWFDEC_MAJORMINOR=0.7
 AC_SUBST(SWFDEC_MAJORMINOR)
-SWFDEC_REQUIRED=0.6.0
+SWFDEC_REQUIRED=0.7.0
 
 GTK_REQUIRED=2.12.0
 
