From af0022fc388a587726f1186bbe131c666e4d50a3 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Fri, 02 Sep 2011 16:21:58 +0000
Subject: previewer: Fix leaked GError

---
diff --git a/src/nautilus-previewer.c b/src/nautilus-previewer.c
index 742e083..3f53fce 100644
--- a/src/nautilus-previewer.c
+++ b/src/nautilus-previewer.c
@@ -84,6 +84,7 @@ nautilus_previewer_init (NautilusPreviewer *self)
 
   if (error != NULL) {
     g_printerr ("Unable to initialize DBus connection: %s", error->message);
+    g_error_free (error);
     return;
   }
 }
--
cgit v0.9.0.2
