diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN nautilus-1.21.2.orig/configure.in nautilus-2.21.2/configure.in
--- configure.in	2008-01-08 04:29:18.000000000 -0500
+++ configure.in	2008-01-08 12:06:58.000000000 -0500
@@ -5,7 +5,6 @@ dnl ====================================
 m4_define(bonobo_activation_minver,    2.1.0)
 m4_define(bonobo_minver,               2.1.0)
 m4_define(eel_minver,                  2.21.5)
-m4_define(esound_minver,               0.2.27)
 m4_define(glib_minver,                 2.15.2)
 m4_define(gnome_desktop_minver,        2.9.91)
 m4_define(gnome_minver,                2.14.0)
@@ -43,7 +42,6 @@ AC_SUBST([ACLOCAL_AMFLAGS], ["\${ACLOCAL
 AC_SUBST(BONOBO_ACTIVATION_REQUIRED, [bonobo_activation_minver])
 AC_SUBST(BONOBO_REQUIRED, [bonobo_minver])
 AC_SUBST(EEL_REQUIRED, [eel_minver])
-AC_SUBST(ESOUND_REQUIRED, [esound_minver])
 AC_SUBST(GLIB_REQUIRED [glib_minver])
 AC_SUBST(GNOME_DESKTOP_REQUIRED, [gnome_desktop_minver])
 AC_SUBST(GNOME_REQUIRED, [gnome_minver])
@@ -87,7 +85,6 @@ fi
 AC_SUBST(STARTUP_NOTIFICATION_PACKAGE)
 
 PKG_CHECK_MODULES(ALL, [
-	esound			>= esound_minver
 	bonobo-activation-2.0	>= bonobo_activation_minver
 	eel-2.0			>= eel_minver
 	glib-2.0		>= glib_minver
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN nautilus-2.21.2.orig/src/file-manager/fm-icon-view.c nautilus-2.21.2/src/file-manager/fm-icon-view.c
--- src/file-manager/fm-icon-view.c	2007-12-14 05:23:13.000000000 -0500
+++ src/file-manager/fm-icon-view.c	2008-01-08 12:06:32.000000000 -0500
@@ -68,7 +68,6 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
-#include <esd.h>
 
 #include "nautilus-audio-mime-types.h"
 
