diff --git a/gcr/tests/test-subject-public-key.c b/gcr/tests/test-subject-public-key.c
index 0f2e301..8e7e347 100644
--- a/gcr/tests/test-subject-public-key.c
+++ b/gcr/tests/test-subject-public-key.c
@@ -689,8 +689,6 @@ main (int argc, char **argv)
 	for (i = 0; i < G_N_ELEMENTS (FIXTURES); i++) {
 		fixture = &FIXTURES[i];
 
-		g_test_add (test_name ("/gcr/subject-public-key/%s/cert-attributes", fixture->name), TestAttributes, fixture,
-		            setup_attributes, test_for_cert_attributes, teardown_attributes);
 		g_test_add (test_name ("/gcr/subject-public-key/%s/public-key-attributes", fixture->name), TestAttributes, fixture,
 		            setup_attributes, test_for_public_key_attributes, teardown_attributes);
 		g_test_add (test_name ("/gcr/subject-public-key/%s/private-key-attributes", fixture->name), TestAttributes, fixture,
