--- webkitgtk-2.3.4.old/Source/WebCore/GNUmakefile.am	2014-01-13 09:59:57.000000000 +0100
+++ webkitgtk-2.3.4/Source/WebCore/GNUmakefile.am	2014-02-02 17:10:42.680013927 +0100
@@ -384,8 +384,7 @@
 if ENABLE_WEB_AUDIO
 # Installing HRTF database wav files
 audiodir = ${datadir}/webkitgtk-@WEBKITGTK_API_VERSION@/resources/audio/
-audio_DATA = $(WebCore)/platform/audio/resources/Composite.wav
-dist_audio_DATA = $(audio_DATA)
+dist_audio_DATA = $(WebCore)/platform/audio/resources/Composite.wav
 endif  # END ENABLE_WEB_AUDIO
 
 # ANGLE tokenizer & parser
