diff -ur webkit-1.7.2.orig//Source/WebKit/gtk/tests/testatk.c webkit-1.7.2//Source/WebKit/gtk/tests/testatk.c
--- webkit-1.7.2.orig//Source/WebKit/gtk/tests/testatk.c	2011-11-28 01:06:19.868230689 +0100
+++ webkit-1.7.2//Source/WebKit/gtk/tests/testatk.c	2011-11-28 01:45:56.214331937 +0100
@@ -1686,7 +1686,9 @@
     g_test_init(&argc, &argv, 0);
     gtk_disable_setlocale();
     setlocale(LC_ALL, "C");
+#ifndef GTK_API_VERSION_2
     gdk_disable_multidevice();
+#endif
     gtk_init(&argc, &argv);
 }
 
