--- GStreamerGWorld.cpp.orig	2011-01-03 11:30:54.000000000 +0200
+++ WebCore/platform/graphics/gstreamer/GStreamerGWorld.cpp	2011-01-03 11:34:25.000000000 +0200
@@ -210,7 +210,8 @@
 
     if (m_videoWindow) {
         m_videoWindow->prepareForOverlay(message);
-#if GST_CHECK_VERSION(0, 10, 31) || GST_VERSION_NANO
+// New since gst-plugins-base-0.10.31
+#ifdef GST_CHECK_PLUGINS_BASE_VERSION
         gst_x_overlay_set_window_handle(GST_X_OVERLAY(sink), m_videoWindow->videoWindowId());
 #else
         gst_x_overlay_set_xwindow_id(GST_X_OVERLAY(sink), m_videoWindow->videoWindowId());
