diff -rupN work-old/libxml2-2.9.1/configure.in work/libxml2-2.9.1/configure.in
--- work-old/libxml2-2.9.1/configure.in	2013-08-09 19:35:26.309193912 +0000
+++ work/libxml2-2.9.1/configure.in	2013-08-09 19:51:16.669173139 +0000
@@ -1389,7 +1389,10 @@ ICU_LIBS=""
 if test "$with_icu" != "yes" ; then
     echo Disabling ICU support
 else
-    ICU_CONFIG=icu-config
+    if test "x$ICU_CONFIG" = x; then
+        ICU_CONFIG=icu-config
+    fi
+
     if ${ICU_CONFIG} --cflags >/dev/null 2>&1
     then
         ICU_LIBS=`${ICU_CONFIG} --ldflags`
