From 5097c671f199cb108d999fed80156b8a9a506b55 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@gentoo.org>
Date: Tue, 3 May 2011 11:35:01 +0530
Subject: [PATCH] Don't error out if no cursor themes were found

---
 gtweak/widgets.py |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gtweak/widgets.py b/gtweak/widgets.py
index f50ae29..7414836 100644
--- a/gtweak/widgets.py
+++ b/gtweak/widgets.py
@@ -137,8 +137,9 @@ class GSettingsComboTweak(_GSettingsTweak):
     def __init__(self, schema_name, key_name, key_options, **options):
         _GSettingsTweak.__init__(self, schema_name, key_name, **options)
 
-        assert len(key_options) > 0
-        assert len(key_options[0]) == 2
+        # Don't error out if no cursor themes were found
+        #assert len(key_options) > 0
+        #assert len(key_options[0]) == 2
 
         combo = build_combo_box_text(
                     self.settings.get_value(self.key_name),
-- 
1.7.3.4

