From fa5688ef9d73c2155bec38827dea466d028bda0e Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Tue, 13 Sep 2011 15:31:23 +0000
Subject: gnome-rr: fix crash in update_brightness_limits

XFree() isn't as forgiving as free() about NULL pointers.
---
diff --git a/libgnome-desktop/gnome-rr.c b/libgnome-desktop/gnome-rr.c
index 3cab1ed..8c5e9c9 100644
--- a/libgnome-desktop/gnome-rr.c
+++ b/libgnome-desktop/gnome-rr.c
@@ -1558,7 +1558,10 @@ update_brightness_limits (GnomeRROutput *output)
     output->backlight_min = info->values[0];
     output->backlight_max = info->values[1];
 out:
-    XFree (info);
+    if (info != NULL)
+    {
+        XFree (info);
+    }
 #endif
 }
 
--
cgit v0.9.0.2
