--- gtk-doc.make.orig	2008-01-14 22:15:45.000000000 -0800
+++ gtk-doc.make	2008-01-14 22:16:26.000000000 -0800
@@ -139,8 +139,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 	
 
--- docs/reference/gnome-desktop/Makefile.in.orig	2008-01-14 22:18:03.000000000 -0800
+++ docs/reference/gnome-desktop/Makefile.in	2008-01-14 22:18:33.000000000 -0800
@@ -620,8 +620,9 @@
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR) || :; \
-	  which gtkdoc-rebase >/dev/null && \
+	  if test $(which gtkdoc-rebase 2> /dev/null) ; then \
 	    gtkdoc-rebase --relative --dest-dir=$(DESTDIR) --html-dir=$(DESTDIR)$(TARGET_DIR) ; \
+	  fi \
 	fi
 
 uninstall-local:
