diff --git a/Makefile.am b/Makefile.am
index e121e68..53bb8b1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,10 +2,10 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper --disable-schemas-install
 
-SUBDIRS = po src help icons
+SUBDIRS = po src
 
-schemadir = $(GCONF_SCHEMA_FILE_DIR)
-schema_DATA = ghex2.schemas
+#schemadir = $(GCONF_SCHEMA_FILE_DIR)
+#schema_DATA = ghex2.schemas
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtkhex.pc
@@ -25,22 +25,22 @@ EXTRA_DIST =			\
 dist-hook: ghex.spec
 	cp ghex.spec $(distdir)
 
-desktopdir = $(datadir)/applications
-desktop_in_files = ghex.desktop.in
-desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
+#desktopdir = $(datadir)/applications
+#desktop_in_files = ghex.desktop.in
+#desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
-@INTLTOOL_DESKTOP_RULE@
+##@INTLTOOL_DESKTOP_RULE@
 
 #Install the ghex2 schema file
-if GCONF_SCHEMAS_INSTALL
-install-data-local:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
-else
-install-data-local:
-endif
-
-CLEANFILES =			\
-	$(desktop_DATA)
+#if GCONF_SCHEMAS_INSTALL
+#install-data-local:
+#	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
+#else
+#install-data-local:
+#endif
+
+#CLEANFILES =			\
+#	$(desktop_DATA)
 
 DISTCLEANFILES =		\
 	intltool-extract	\
diff --git a/src/Makefile.am b/src/Makefile.am
index 28b460c..00c0b9b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -13,13 +13,13 @@ BUILT_SOURCES = \
 	ghex-marshal.c	\
 	ghex-marshal.h
 
-bin_PROGRAMS = ghex2
+#bin_PROGRAMS = ghex2
 
 lib_LTLIBRARIES = libgtkhex.la
 
 libgtkhexdir = $(includedir)/gtkhex/
 
-uidir = $(datadir)/gnome-2.0/ui/
+#uidir = $(datadir)/gnome-2.0/ui/
 
 libgtkhex_la_SOURCES = \
 	hex-document.c			\
@@ -41,15 +41,15 @@ libgtkhex_HEADERS = \
 	hex-document.h 			\
 	gtkhex.h
 
-ghex2_SOURCES = \
-	hex-document-ui.c preferences.c findreplace.c converter.c	\
-	config.c main.c ui.c chartable.c session.c		\
-	print.c ghex-window.c ghex-window.h				\
-	gnome-print-font-dialog.c gnome-print-font-dialog.h		\
-	gnome-print-font-picker.c gnome-print-font-picker.h		\
-	factory.c factory.h hex-dialog.c hex-dialog.h			\
-	configuration.h preferences.h findreplace.h print.h		\
-	converter.h chartable.h ui.h session.h
+#ghex2_SOURCES = \
+#	hex-document-ui.c preferences.c findreplace.c converter.c	\
+#	config.c main.c ui.c chartable.c session.c		\
+#	print.c ghex-window.c ghex-window.h				\
+#	gnome-print-font-dialog.c gnome-print-font-dialog.h		\
+#	gnome-print-font-picker.c gnome-print-font-picker.h		\
+#	factory.c factory.h hex-dialog.c hex-dialog.h			\
+#	configuration.h preferences.h findreplace.h print.h		\
+#	converter.h chartable.h ui.h session.h
 
 ghex-marshal.h: ghex-marshal.list $(GLIB_GENMARSHAL)
 	$(GLIB_GENMARSHAL) $< --header --prefix=ghex_marshal > $@
@@ -57,11 +57,12 @@ ghex-marshal.c: ghex-marshal.list $(GLIB_GENMARSHAL)
 	echo "#include \"ghex-marshal.h\"" > $@ && \
 	$(GLIB_GENMARSHAL) $< --body --prefix=ghex_marshal >> $@
 
-ui_DATA = ghex-ui.xml
+#ui_DATA = ghex-ui.xml
 
-EXTRA_DIST = \
-	$(ui_DATA)		\
-	ghex-marshal.list
+#EXTRA_DIST = \
+#	$(ui_DATA)		\
+#	ghex-marshal.list
+EXTRA_DIST = ghex-marshal.list
 
 LDADD =	\
 	$(POPT_LIBS)	\
